---
title: "Gale 2024 script"
output: html_document
---

# get started
```{r get started}

library(multcomp)#
library(dunn.test)#
library(rcompanion)#
library(tidyr)#
library(ggplot2)#
library(gridExtra)#
library(reshape)#
library(grid)
library(readxl)
library(reshape2)
library(maps)
library(dplyr)
library(vegan)
library(cowplot)
library(ggtext)
library(gridtext)
library(VennDiagram)
library(ggh4x)
library(minpack.lm)
library(Hmisc)
library(stats4)
library(ggpubr)
library(stringr)
library(survival)
library(survminer)

source(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/ancom_tests.R")
source(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/g_legend.R")
source(file = '~/Dropbox/sequencing/16sscripts/FrederickHuangLin-ANCOM-c3d6f98/scripts/ancom_v2.0.R') ## NOTE USER must get ANCOM from independent access to run ANCOM scripts
source(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/pcoa_0.0.3.R")
source(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/make_taxon_plot_0.0.5.R")
source(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/calculate_correlations.R")
source(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/make_permanova_tables.R")
source(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/96_well_scripts.R")
source(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/Gale_functions.R")
source(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/other_taxonomy_files_0.0.1.R")
source(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/make_distance_plot_0.0.3.R")

predefined_taxa <- c("k__Bacteria_ p__Proteobacteria_ c__Gammaproteobacteria_ o___ f___ g___",
                     "k__Bacteria_ p__Proteobacteria_ c__Gammaproteobacteria_ o__Enterobacteriales_ f__Enterobacteriaceae_NA",
                     "k__Bacteria_ p__Proteobacteria_ c__Gammaproteobacteria_ o__Pasteurellales_ f___ g__", 
                     "k__Bacteria_ p__Proteobacteria_ c__Gammaproteobacteria_ o__Enterobacteriales_ f__Enterobacteriaceae_",
                     "k__Bacteria_ p__Firmicutes_ c__Bacilli_ o__Lactobacillales_ f___ g__",
                     "k__Bacteria_ p__Firmicutes_ c__Bacilli_ o__Lactobacillales_ f__Lactobacillaceae_",
                     "k__Bacteria_ p__Firmicutes_ c__Bacilli_ o__Lactobacillales_ f__Enterococcaceae_ g__Enterococcus",
                     "k__Bacteria_ p__Firmicutes_ c__Bacilli_ o__Lactobacillales_ f__Leuconostocaceae_ g__Leuconostoc",
                     "k__Bacteria_ p__Firmicutes_ c__Bacilli_ o__Lactobacillales_ f__Leuconostocaceae_ g__Weissella",
                     "k__Bacteria_ p__Firmicutes_ c__Bacilli_ o__Lactobacillales_ f__Lactobacillaceae_ g__Lactobacillus",
                     "k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_NA",
                     "k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_",
                     "k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Commensalibacter",
                     "k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Gluconobacter",
                     "k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Gluconacetobacter",
                     "k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Acetobacter"
                     )

predefined_shortname <- c("Gammaproteobacteria","GammaproteobacteriaNA",
                     "Pasteurellales", 
                     "Enterobacteriaceae","EnterobacteriaceaeNA",
                     "Lactobacillales",
                     "Lactobacillaceae",
                     "Enterococcus",
                     "Leuconostoc",
                     "Weissella",
                     "Lactobacillus",
                     "Acetobacteraceae","AcetobacteraceaeNA",
                     "Commensalibacter",
                     "Gluconobacter",
                     "Gluconacetobacter",
                     "Acetobacter"
                     )

predefined_colors <- c("green4","green4",
                     "lightgreen", 
                     "darkgreen","darkgreen",
                     "purple3",
                     "blue4",
                     "purple4",
                     "purple1",
                     "lavender",
                     "blue",
                     "pink","pink",
                     "yellow",
                     "orange",
                     "magenta",
                     "red"
                     )




```

# Figure 1
## make the east coast map
```{r}

## making plots for the tradeoffs manuscript

# used this link: http://eriqande.github.io/rep-res-web/lectures/making-maps-with-R.html

map <- map_data("state")

east_coast <- subset(map, region %in% c("maine","connecticut","delaware","district of columbia","maryland","massachusetts","new hampshire","new jersey","new york","north carolina","pennsylvania","rhode island","south carolina","vermont","virginia"))

east_coast$group2 <- east_coast$group
length(table(list(east_coast$group2)))
droplevels(subset(east_coast, east_coast$region=="maine"))[1,]
droplevels(subset(east_coast, east_coast$region=="florida"))[1,]
east_coast$color <- "white"

labs2 <- readr::read_csv(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/lat_long.csv", show_col_types = F) %>% 
  data.frame() %>% 
  rbind(data.frame(site = as.character(c("Homestead, FL","ME3")),Geography = as.character(c("FL","ME3")), latitude = as.numeric(c(25.4687,45.5)), longitude = as.numeric(c(-80.4776, -68.5)))) %>% 
  mutate(mergecol = c("ME2", "VA","MD", "NH","ME1","MA","PA","CT","FL", "ME3"))
labs2$mergecol <- labs2$mergecol[c(5,2,3,4,1,6,7,8,9,10)]

# sampling_sites <- read_xlsx('~/Dropbox/east_coast_flies_manuscript/megamap_latlong.xlsx', sheet = "Sheet5") # original readin
sampling_sites <- readr::read_csv(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/megamap_latlong_sheet5.csv", show_col_types = F) 
  
ecmap <- ggplot() + 
  geom_polygon(data=east_coast, aes(x=long, y=lat, group=group2), color="grey",fill=east_coast$color) + 
  coord_fixed(1.3) + 
  theme_void() + 
  geom_point(data = sampling_sites, aes(y=latitude, x=longitude * -1, shape = location), size = 2.5, color = "black") + 
  scale_shape_manual(values = c(1:10,18,12:17,0)) +
  #geom_line(data = sampling_sites %>% filter(location %in% c("Grandaddys","Brysons")), aes(y=latitude, x=longitude * -1), linewidth = 1, color = "steelblue") +
  geom_line(data = sampling_sites %>% filter(location %in% c("Durham, NH","Carter Hill Orchard","Champlain, VT")), aes(y=latitude, x=longitude * -1), linewidth = 1, color = "steelblue") + 
  geom_line(data = sampling_sites %>% filter(location %in% c("Terhune","Solebury 9-28","Unionville")), aes(y=latitude, x=longitude * -1), linewidth = 1, color = "steelblue") + 
  geom_line(data = sampling_sites %>% filter(location %in% c("middlefield","Taylor Brooke")), aes(y=latitude, x=longitude * -1), linewidth = 1, color = "steelblue") + 
  geom_line(data = sampling_sites %>% filter(location %in% c("Walpole, ME","Rocky Ridge")), aes(y=latitude, x=longitude * -1), linewidth = 1, color = "steelblue") +
  geom_line(data = sampling_sites %>% filter(location %in% c("etna, me","North Chester")), aes(y=latitude, x=longitude * -1), linewidth = 1, color = "steelblue") +
  theme(legend.position = "none")

ecmap

# Harvard, MA	42.50867	-71.57071	Westward_F
# Bowdoin, ME	44.0253	-69.94379	RockyRidge_I
# Middlefield, CT	41.49383	-72.71145	Lyman_N
# Media, PA	39.8845	-75.4125	Linvilla_D
# Durham, NH	43.08745	-71.04666	DemerritHillFarm_D
# Churchville, MD	39.539	-76.2399	Lahr_G
# Etna, ME	44.82082	-69.11144	Conant_H
# Charlottesville, VA	38.07474	-78.38155	CarterMountain_H


```

## make the utah map
```{r}
east_coast <- subset(map, region %in% c("utah"))

east_coast$group2 <- east_coast$group
length(table(list(east_coast$group2)))
droplevels(subset(east_coast, east_coast$region=="maine"))[1,]
droplevels(subset(east_coast, east_coast$region=="florida"))[1,]
east_coast$color <- "white"

utmap <- ggplot() + 
  geom_polygon(data=east_coast, aes(x=long, y=lat, group=group2), color="grey",fill=east_coast$color) + 
  coord_fixed(1.3) + 
  theme_void() + 
  geom_point(aes(y=c(41.72573, 40.3327707, 41.326252, 39.966865, 40.4457589), x=c(-111.809883, -111.7219739, -112.011375, -111.791562, -111.7818936), shape = c(letters[6:10])), size = 2.5, color = "black") + 
  scale_shape_manual(values = c(7,2,6,1,5)) +
  theme(legend.position = "none")

utmap
# logan 41.72573	--111.809883
# martells 40.3327707	-111.7219739
# Ogden, UT	41.326252	-112.011375
# Santaquin, UT	39.966865	-111.791562
# alpine 40.4457589	-111.7818936
```

## Walters flies (2009)
### file downloads
```{sh}
pwd
mkdir core-metrics-results-redo2009m2
wget -O core-metrics-results-redo2009m2/rarefied_table.txt https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-redo2009m2/rarefied_table.txt
wget https://raw.githubusercontent.com/johnchaston/Gale2024/main/amber_mapper_7_JMC.tsv
mkdir taxonomy-Amber
wget -O taxonomy-Amber/taxonomy_forR.csv https://raw.githubusercontent.com/johnchaston/Gale2024/main/taxonomy-Amber/taxonomy_forR.csv
mkdir taxonomy-megaC
wget -O taxonomy-megaC/taxonomy_forR.csv https://raw.githubusercontent.com/johnchaston/Gale2024/main/taxonomy-megaC/taxonomy_forR.csv
wget https://raw.githubusercontent.com/johnchaston/Gale2024/main/amber_mapper_7_JMC_forcor.txt
pwd
```

### taxon plot
```{r}
setwd('~/Dropbox/Working Trade-offs/Gale/sequence_results/')
file_path <- "redo2009m2"
mapper_file <- "amber_mapper_7_JMC.tsv"
map_path = "-Amber" # taxonomy folder extension
plotvarnames = c("latitude") 
mapper_file = mapper_file 
taxonomic_level="genus"
rpa_in_chart = .03
read_depth = 475
legend_position = "bottom"
sort_x_axis = "latitude"
x_cluster = "X.SampleID" #

ptp <- prep_taxon_plot_numbersfirst(file_path = file_path, map_path = map_path, plotvarnames = plotvarnames, mapper_file = mapper_file, taxonomic_level=taxonomic_level, rpa_in_chart = rpa_in_chart, read_depth = read_depth, legend_position=legend_position,  sort_x_axis = sort_x_axis,  x_cluster = x_cluster, predefined_taxa = predefined_taxa, predefined_shortname = predefined_shortname)

x_cluster <- "X.SampleID"
plot_color <- c("black")
plot_order <- ptp[[3]]$shortname[c(7,2,1,4,6,5,3)] 
facet_formula <- ". ~ latitude"
sort_x_axis <- "latitude"
legend_position = "bottom"
xorder = "latitude"

table(ptp[[1]]$latitude)
ptp[[1]] <- ptp[[1]] %>% rbind(ptp[[1]][1:5,])
dim(ptp[[1]])
ptp[[1]][443,c("Count","twovar","latitude")] = c(0,rep(34.8,2))
ptp[[1]][442,c("Count","twovar","latitude")] = c(0,rep(35.3,2))
ptp[[1]][441,c("Count","twovar","latitude")] = c(0,rep(37.9,2))
ptp[[1]][440,c("Count","twovar","latitude")] = c(0,rep(39.5,2))
ptp[[1]][439,c("Count","twovar","latitude")] = c(0,rep(44.8,2))
table(ptp[[1]]$latitude)
ptp[[1]] <- ptp[[1]] %>%
  rowwise() %>%
  mutate(latitude = ifelse(latitude == "40.3836", "40.3319", ifelse(latitude == "43.9559", "44.02",ifelse(latitude == "43.8569", "43.8569",latitude)))) %>%
  ungroup()

x_cluster = "latitude"
base_plot <- make_taxon_plot_condensed(ptp = ptp, plot_color = plot_color, facet_formula = facet_formula, sort_x_axis = sort_x_axis, x_cluster = x_cluster, plotvarnames = plotvarnames, legend_position = legend_position, read_depth = read_depth, plot_order = plot_order, xorder = xorder, predefined_taxa = predefined_taxa, predefined_colors = predefined_colors, predefined_shortname = predefined_shortname, default_colors = T)
base_plot
amberflies <- base_plot +
  facet_grid(rows = vars(latitude), drop = T, as.table = FALSE, scales = "free", space = "free") + 
  theme(strip.text.y = element_blank(), axis.text.y = element_blank(), legend.position = "none") + 
  coord_flip() + 
  labs(x = NULL)

amberflies+ theme(strip.text.y = element_text(size=10))

amber_table <- calculate_correlations(file_path = file_path,taxonomy_extension = "megaC", mapper_file = "amber_mapper_7_JMC_forcor.txt", use_fdr = F, defined_hierarchy = "genus", return_table = T)
cortest_amber <- cor.test(amber_table$`k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Acetobacter`, amber_table$`k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Commensalibacter`, method = "spearman")
corplot_amber <- ggplot(amber_table, aes(x = `k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Acetobacter`, y = `k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Commensalibacter`)) + geom_jitter(height = 0.1, width = 0.1) + stat_smooth(method = "lm", alpha = 0)
```

## Henry flies
###  bash
```{bash}
pwd
mkdir core-metrics-results-flies_mantel2o2
wget -O core-metrics-results-flies_mantel2o2/rarefied_table.txt https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-flies_mantel2o2/rarefied_table.txt
wget https://raw.githubusercontent.com/johnchaston/Gale2024/main/henrymapper2_JMC.txt
mkdir taxonomy-henry
wget -O taxonomy-henry/taxonomy_forR.csv https://raw.githubusercontent.com/johnchaston/Gale2024/main/taxonomy-henry/taxonomy_forR.csv
mkdir core-metrics-results-flies_mantel2v2
wget -O core-metrics-results-flies_mantel2v2/rarefied_table.txt https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-flies_mantel2v2/rarefied_table.txt
pwd
```

### orchard taxon plot
```{r}

file_path <- "flies_mantel2o2"
mapper_file <- "henrymapper2_JMC.txt"
map_path = "-henry" # taxonomy folder extension
plotvarnames = c("sample_type","location","sexplot","species","wolbplot","fnd","fndl","latitude") 
mapper_file = mapper_file 
taxonomic_level="genus"
rpa_in_chart = .03
read_depth = 500
legend_position = "bottom"
sort_x_axis = "sample_type"
x_cluster = "X.SampleID" #

ptp <- prep_taxon_plot(file_path = file_path, map_path = map_path, plotvarnames = plotvarnames, mapper_file = mapper_file, taxonomic_level=taxonomic_level, rpa_in_chart = rpa_in_chart, read_depth = read_depth, legend_position=legend_position,  sort_x_axis = sort_x_axis,  x_cluster = x_cluster, predefined_taxa = predefined_taxa, predefined_shortname = predefined_shortname)

x_cluster <- "latitude"
plot_color <- c("black","lightgreen","darkgreen","purple4","purple1","blue","pink","yellow","orange","magenta","red")
plot_order <- ptp[[3]]$shortname[c(10,9,1,3,2,8,5,7,6,4)]
order_numbers <- c(10,9,1,3,2,8,5,7,6,4)

facet_formula <- ". ~ latitude"
sort_x_axis <- "latitude"
legend_position = "bottom"
xorder = "latitude"

table(ptp[[1]]$latitude)
ptp[[1]] <- ptp[[1]] %>% rbind(ptp[[1]][1:2,])
dim(ptp[[1]])
ptp[[1]][1832,c("Count","twovar","latitude")] = c(0,rep(39.88,2))
ptp[[1]][1831,c("Count","twovar","latitude")] = c(0,rep(42.49,2))
table(ptp[[1]]$latitude)

ptp[[1]] <- ptp[[1]] %>%
  rowwise() %>%
  mutate(latitude = ifelse(latitude == 40.32, 40.38,ifelse(latitude == 42.23, 43.23,latitude))) %>%
  ungroup()

table(ptp[[1]]$latitude)

base_plot <- make_taxon_plot_condensed(ptp = ptp, plot_color = plot_color, facet_formula = facet_formula, sort_x_axis = sort_x_axis, x_cluster = x_cluster, plotvarnames = plotvarnames, legend_position = legend_position, read_depth = read_depth, plot_order = plot_order, predefined_colors = predefined_colors, predefined_shortname = predefined_shortname, predefined_taxa = predefined_taxa, default_colors = T, xorder = xorder, default_order = F, order_numbers = order_numbers)
base_plot
henryorchard <- base_plot + 
  facet_grid(rows = vars(latitude), drop = FALSE, as.table = FALSE, scales = "free", space = "free") + 
  theme(strip.text.x = element_blank(), axis.text.y = element_blank(), axis.text.x = element_blank(), strip.text.y = element_blank(), legend.position = "none") + 
  coord_flip() + 
  labs(x = "Sample") 

henryorchard+ theme(strip.text.y = element_text())

orchard_table <- calculate_correlations(file_path = file_path,taxonomy_extension = gsub(pattern = "-", replacement = "", map_path), mapper_file = mapper_file, frac_samples = .10, use_fdr = F, defined_hierarchy = "genus", return_table = T)
cortest_orchard <- cor.test(orchard_table$`k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Acetobacter`, orchard_table$`k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Commensalibacter`, method = "spearman")
corplot_orchard <- ggplot(orchard_table, aes(x = `k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Acetobacter`, y = `k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Commensalibacter`)) + geom_jitter(height = 0.1, width = 0.1) + stat_smooth(method = "lm", alpha = 0)

```

### vineyard taxon plot
```{r}
file_path <- "flies_mantel2v2"
mapper_file <- "henrymapper2_JMC.txt"
map_path = "-henry" # taxonomy folder extension
plotvarnames = c("sample_type","location","sexplot","species","wolbplot","fnd","fndl","latitude") 
mapper_file = mapper_file 
taxonomic_level="genus"
rpa_in_chart = .02
read_depth = 500
legend_position = "bottom"
sort_x_axis = "sample_type"
x_cluster = "X.SampleID" #

ptp <- prep_taxon_plot(file_path = file_path, map_path = map_path, plotvarnames = plotvarnames, mapper_file = mapper_file, taxonomic_level=taxonomic_level, rpa_in_chart = rpa_in_chart, read_depth = read_depth, legend_position=legend_position,  sort_x_axis = sort_x_axis,  x_cluster = x_cluster, predefined_taxa = predefined_taxa, predefined_shortname = predefined_shortname)

x_cluster <- "latitude"
plot_color <- c("black","lightgreen","green4","purple4","purple1","blue","pink","yellow","orange","magenta","red")
plot_order <- ptp[[3]]$shortname[c(10,9,1,3,2,8,5,7,6,4)]
order_numbers <- c(10,9,1,3,2,8,5,7,6,4)
facet_formula <- ". ~ latitude"
sort_x_axis <- "latitude"
legend_position = "bottom"


table(ptp[[1]]$location)
ptp[[1]] <- ptp[[1]] %>% rbind(ptp[[1]][1:5,])
dim(ptp[[1]])
ptp[[1]][2409,c("Count","twovar","latitude")] = c(0,rep(39.88,2))
ptp[[1]][2408,c("Count","twovar","latitude")] = c(0,rep(42.49,2))
ptp[[1]][2407,c("Count","twovar","latitude")] = c(0,rep(43.23,2))
ptp[[1]][2406,c("Count","twovar","latitude")] = c(0,rep(43.95,2))
ptp[[1]][2405,c("Count","twovar","latitude")] = c(0,rep(44.02,2))
table(ptp[[1]]$latitude)

ptp[[1]] <- ptp[[1]] %>%
  rowwise() %>%
  mutate(latitude = ifelse(latitude == 35.34, 35.36,latitude)) %>%
  ungroup()

table(ptp[[1]]$latitude)

base_plot <- make_taxon_plot_condensed(ptp = ptp, plot_color = plot_color, facet_formula = facet_formula, sort_x_axis = sort_x_axis, x_cluster = x_cluster, plotvarnames = plotvarnames, legend_position = legend_position, read_depth = read_depth, plot_order = plot_order, xorder = xorder, default_colors = T, default_order = F, predefined_colors = predefined_colors, predefined_taxa = predefined_taxa, predefined_shortname = predefined_shortname, order_numbers = order_numbers)
vineyard <- base_plot + 
  facet_grid(rows = vars(latitude), drop = FALSE, as.table = FALSE, scales = "free", space = "free") + 
  theme(strip.text.x = element_blank(), axis.text.x = element_blank(), axis.text.y = element_blank(), strip.text.y = element_blank(), legend.position = "none") + 
  coord_flip() + 
  labs(x = "Sample")

vineyard + theme(strip.text.y = element_text())


vineyard_table <- calculate_correlations(file_path = file_path,taxonomy_extension = gsub(pattern = "-", replacement = "", map_path), mapper_file = mapper_file, frac_samples = .10, use_fdr = F, defined_hierarchy = "genus", return_table = T)
cortest_vineyard <- cor.test(vineyard_table$`k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Acetobacter`, vineyard_table$`k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Commensalibacter`, method = "spearman")
corplot_vineyard <- ggplot(vineyard_table, aes(x = `k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Acetobacter`, y = `k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Commensalibacter`)) + geom_jitter(height = 0.1, width = 0.1) + stat_smooth(method = "lm", alpha = 0)
```

## East coast flies
### bash
```{sh}
# bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh ec1and2""_noW ec1and2 east_coast_overall3 500 mapper_1and2.tsv "sample_type = 'flies' AND species = 'melanogaster' AND starved = 'starved' AND subsite NOT IN ('pile')"
pwd
mkdir core-metrics-results-east_coast_overall3
wget -O core-metrics-results-east_coast_overall3/rarefied_table.txt https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-east_coast_overall3/rarefied_table.txt
wget https://raw.githubusercontent.com/johnchaston/Gale2024/main/mapper_1and2.tsv
mkdir taxonomy-ec1and2
wget -O taxonomy-ec1and2/taxonomy_forR.csv https://raw.githubusercontent.com/johnchaston/Gale2024/main/taxonomy-ec1and2/taxonomy_forR.csv
pwd
```

### taxon plot
```{r}
file_path <- "east_coast_overall3"
mapper_file <- "mapper_1and2.tsv"
map_path = "-ec1and2" # taxonomy folder extension
plotvarnames = c("site") 
mapper_file = mapper_file 
taxonomic_level="genus"
rpa_in_chart = .02
read_depth = 500
legend_position = "bottom"
sort_x_axis = "site"
x_cluster = "X.SampleID" #

ptp <- prep_taxon_plot(file_path = file_path, map_path = map_path, plotvarnames = plotvarnames, mapper_file = mapper_file, taxonomic_level=taxonomic_level, rpa_in_chart = rpa_in_chart, read_depth = read_depth, legend_position=legend_position,  sort_x_axis = sort_x_axis,  x_cluster = x_cluster, predefined_taxa = predefined_taxa, predefined_shortname = predefined_shortname)

table(ptp[[1]]$site)
x_cluster <- "site"
plot_color = c("black","lightgreen", "darkgreen","purple3","blue4","purple4","purple1","lavender","blue","pink","yellow","orange","magenta","red")
plot_order <- ptp[[3]]$shortname[c(13,12,1,3,2,5,6,4,7,9,11,10,8)]
order_numbers <- c(13,12,1,3,2,5,6,4,7,9,11,10,8)
facet_formula <- ". ~ site"
sort_x_axis <- "site"
legend_position = "bottom"


ptp[[1]] <- ptp[[1]] %>% rbind(ptp[[1]][1:4,])
ptp[[1]][1508,c("Count","twovar","site")] = c(0,rep(4,2))
ptp[[1]][1507,c("Count","twovar","site")] = c(0,rep(1,2))
ptp[[1]][1506,c("Count","twovar","site")] = c(0,rep(2,2))
ptp[[1]][1505,c("Count","twovar","site")] = c(0,rep(3,2))

factor(ptp[[1]]$site, levels = rev(c("Etna, ME","Bowdoin, ME","Durham, NH","Harvard, MA","Middlefield, CT","1","Media, PA", "Churchville, MD","4","2","3")))

base_plot <- make_taxon_plot_condensed(ptp = ptp, plot_color = plot_color, facet_formula = facet_formula, sort_x_axis = sort_x_axis, x_cluster = x_cluster, plotvarnames = plotvarnames, legend_position = legend_position, read_depth = read_depth, plot_order = plot_order, relevel_col = "site", relevel_vec = rev(c("Etna, ME","Bowdoin, ME","Durham, NH","Harvard, MA","Middlefield, CT","1","Media, PA", "Churchville, MD","4","2","3")), xorder = xorder, default_colors = T, default_order = F, predefined_taxa = predefined_taxa, predefined_shortname = predefined_shortname, predefined_colors = predefined_colors, order_numbers = order_numbers)
ec2021 <- base_plot + 
  facet_grid(rows = vars(site), drop = FALSE, as.table = FALSE, scales = "free", space = "free") + 
  theme(strip.text.y = element_blank(), axis.text.y = element_blank(), axis.title.y = element_blank(), legend.position = "none") + 
  coord_flip() + 
  labs(x = "Sample")


ec2021 + theme(strip.text.y = element_text())

ec_table <- calculate_correlations(file_path = file_path,taxonomy_extension = gsub(pattern = "-", replacement = "", map_path), mapper_file = mapper_file, use_fdr = F, defined_hierarchy = "genus", return_table = T)
cortest_ec <- cor.test(ec_table$`k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Acetobacter`, ec_table$`k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Commensalibacter`, method = "spearman")
corplot_ec <- ggplot(ec_table, aes(x = `k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Acetobacter`, y = `k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Commensalibacter`)) + geom_jitter(height = 0.1, width = 0.1) + stat_smooth(method = "lm", alpha = 0)
```

## UT
### bash
```{bash}
# name=utflies2021
# bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name utah_mantel500S 500 mapper_file_JMCv3_edit.tsv "SampleID2 NOT IN ('utf1008', 'utf1022', 'utf1036', 'utf1360', 'utf1401', 'utf385', 'utf415', 'utf427', 'utf432', 'utf433', 'utf438', 'utf444',  'utf445', 'utf449', 'utf476', 'utf478', 'utf484', 'utf534', 'utf535', 'utf536', 'utf542', 'utf549', 'utf559', 'utf572', 'utf577', 'utf665', 'utf676', 'utf678', 'utf968', 'utf972', 'utf974', 'utf980', 'utf996', 'utf998') AND species='MEL' AND sex='M' AND starved='Y'"
pwd
mkdir core-metrics-results-utah_mantel500S
wget -O core-metrics-results-utah_mantel500S/rarefied_table.txt https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-utah_mantel500S/rarefied_table.txt
wget https://raw.githubusercontent.com/johnchaston/Gale2024/main/mapper_file_JMCv3_edit.tsv
mkdir taxonomy-utflies2021
wget -O taxonomy-utflies2021/taxonomy_forR.csv https://raw.githubusercontent.com/johnchaston/Gale2024/main/taxonomy-utflies2021/taxonomy_forR.csv
pwd
```

### taxon plot
```{r}
file_path <- "utah_mantel500S"
mapper_file <- "mapper_file_JMCv3_edit.tsv"
map_path = "-utflies2021" # taxonomy folder extension
plotvarnames = c("site") 
mapper_file = mapper_file 
taxonomic_level="genus"
rpa_in_chart = .02
read_depth = 500
legend_position = "bottom"
sort_x_axis = "site"
x_cluster = "X.SampleID" #

ptp <- prep_taxon_plot(file_path = file_path, map_path = map_path, plotvarnames = plotvarnames, mapper_file = mapper_file, taxonomic_level=taxonomic_level, rpa_in_chart = rpa_in_chart, read_depth = read_depth, legend_position=legend_position,  sort_x_axis = sort_x_axis,  x_cluster = x_cluster, predefined_taxa = predefined_taxa, predefined_shortname = predefined_shortname)

x_cluster <- "site"
plot_color = c("black","lightgreen", "darkgreen","blue4","purple4","purple1","blue","pink","yellow","orange","magenta","red")
plot_order <- ptp[[3]]$shortname[c(11,10,2,1,4,3,5,7,9,8,6)]
order_numbers <- c(11,10,2,1,4,3,5,7,9,8,6)
facet_formula <- ". ~ site"
sort_x_axis <- "site"
legend_position = "bottom"

base_plot <- make_taxon_plot_condensed(ptp = ptp, plot_color = plot_color, facet_formula = facet_formula, sort_x_axis = sort_x_axis, x_cluster = x_cluster, plotvarnames = plotvarnames, legend_position = legend_position, read_depth = read_depth, plot_order = plot_order, relevel_col = "site", relevel_vec = (c("SANTAQUIN","MARTELLS","ALPINE","OGDEN", "LOGAN")))
ut2020 <- base_plot + 
  facet_grid(rows = vars(site), drop = FALSE, as.table = FALSE, scales = "free", space = "free") + 
  theme(strip.text.y = element_blank(), axis.text.y = element_blank(), axis.title.y = element_blank(), legend.position = "none") + 
  coord_flip() + 
  labs(x = "Sample")

ut2020 + theme(strip.text.y = element_text())

utah_table <- calculate_correlations(file_path = file_path,taxonomy_extension = gsub(pattern = "-", replacement = "", map_path), mapper_file = mapper_file, use_fdr = F, defined_hierarchy = "genus", return_table = T)
cortest_UT <- cor.test(utah_table$`k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Acetobacter`, utah_table$`k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Commensalibacter`, method = "spearman")
corplot_UT <- ggplot(utah_table, aes(x = `k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Acetobacter`, y = `k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Commensalibacter`)) + geom_jitter(height = 0.1, width = 0.1) + stat_smooth(method = "lm", alpha = 0)
```

## Mantel tests
### download files
```{bash}
pwd
for i in flies_mantel2o2 flies_mantel2v2 redo2009m2 east_coast_overall3 utah_mantel500S; do
  mkdir core-metrics-results-$i""/bray_curtis_distance_matrix
  wget -O core-metrics-results-$i""/bray_curtis_distance_matrix/distance-matrix.tsv https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-$i""/bray_curtis_distance_matrix/distance-matrix.tsv
  mkdir core-metrics-results-$i""/weighted_unifrac_distance_matrix
  wget -O core-metrics-results-$i""/weighted_unifrac_distance_matrix/distance-matrix.tsv https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-$i""/weighted_unifrac_distance_matrix/distance-matrix.tsv
  mkdir core-metrics-results-$i""/unweighted_unifrac_distance_matrix
  wget -O core-metrics-results-$i""/unweighted_unifrac_distance_matrix/distance-matrix.tsv https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-$i""/unweighted_unifrac_distance_matrix/distance-matrix.tsv
done

wget https://raw.githubusercontent.com/johnchaston/Gale2024/main/Amber_mapper_9.txt
pwd

```
### script
```{r}
point_size = 0.1
####
## henry
####
henryorchard3 <- mantel_EC18("henrymapper2_JMC.txt", "flies_mantel2o2", "metadata_2018.txt", cols_to_test = list(c("MinRH","latitude","MaxTemp")),"bray_curtis") 
henryorchardlat <- mantel_EC18("henrymapper2_JMC.txt", "flies_mantel2o2", "metadata_2018.txt", cols_to_test = list(c("latitude")),"bray_curtis") 
phenryorchard3 <- mantel_distance_plot(henryorchard3, y_title = "environmental distance", point_size = point_size) + labs(title = bquote(rho~" = 0.13, p = 0.02"))
phenryorchardlat <- mantel_distance_plot(henryorchardlat, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = -0.04, p = 0.77"))

henryorchard3_WU <- mantel_EC18("henrymapper2_JMC.txt", "flies_mantel2o2", "metadata_2018.txt", cols_to_test = list(c("MinRH","latitude","MaxTemp")),"weighted_unifrac") 
henryorchardlat_WU <- mantel_EC18("henrymapper2_JMC.txt", "flies_mantel2o2", "metadata_2018.txt", cols_to_test = list(c("latitude")),"weighted_unifrac") 
phenryorchard3_WU <- mantel_distance_plot(henryorchard3, y_title = "environmental distance", point_size = point_size) + labs(title = bquote(rho~" = 0.17, p = 0.02"))
phenryorchardlat_WU <- mantel_distance_plot(henryorchardlat, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = 0.03, p = 0.30"))
henryorchard3_UU <- mantel_EC18("henrymapper2_JMC.txt", "flies_mantel2o2", "metadata_2018.txt", cols_to_test = list(c("MinRH","latitude","MaxTemp")),"unweighted_unifrac") 
henryorchardlat_UU <- mantel_EC18("henrymapper2_JMC.txt", "flies_mantel2o2", "metadata_2018.txt", cols_to_test = list(c("latitude")),"unweighted_unifrac") 
phenryorchard3_UU <- mantel_distance_plot(henryorchard3, y_title = "environmental distance", point_size = point_size) + labs(title = bquote(rho~" = 0.07, p = 0.10"))
phenryorchardlat_UU <- mantel_distance_plot(henryorchardlat, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = 0.13, p = 0.006"))

# sig sig not
henryv3 <- mantel_EC18("henrymapper2_JMC.txt", "flies_mantel2v2", "metadata_2018.txt", cols_to_test = list(c("MinRH","latitude","MaxTemp")),"bray_curtis") 
henryvlat <- mantel_EC18("henrymapper2_JMC.txt", "flies_mantel2v2", "metadata_2018.txt", cols_to_test = list(c("latitude")),"bray_curtis") 
phenryv3 <- mantel_distance_plot(henryv3, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = 0.32, p = 0.001"))
phenryvlat <- mantel_distance_plot(henryvlat, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = -0.01, p = 0.56"))

henryv3_WU <- mantel_EC18("henrymapper2_JMC.txt", "flies_mantel2v2", "metadata_2018.txt", cols_to_test = list(c("MinRH","latitude","MaxTemp")),"weighted_unifrac") 
henryvlat_WU <- mantel_EC18("henrymapper2_JMC.txt", "flies_mantel2v2", "metadata_2018.txt", cols_to_test = list(c("latitude")),"weighted_unifrac") 
phenryv3_WU <- mantel_distance_plot(henryv3, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = 0.21, p = 0.001"))
phenryvlat_WU <- mantel_distance_plot(henryvlat, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = -0.01, p = 0.59"))
henryv3_UU <- mantel_EC18("henrymapper2_JMC.txt", "flies_mantel2v2", "metadata_2018.txt", cols_to_test = list(c("MinRH","latitude","MaxTemp")),"unweighted_unifrac") 
henryvlat_UU <- mantel_EC18("henrymapper2_JMC.txt", "flies_mantel2v2", "metadata_2018.txt", cols_to_test = list(c("latitude")),"unweighted_unifrac") 
phenryv3_UU <- mantel_distance_plot(henryv3, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = 0.23, p = 0.001"))
phenryvlat_UU <- mantel_distance_plot(henryvlat, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = 0.03, p = 0.20"))

####
## amber
####
amber3 <- mantel_EC18("Amber_mapper_9.txt", "redo2009m2", "metadata_2009_to_12.txt", cols_to_test = list(c("MinRH","latitude","MaxTemp")),"bray_curtis") 
amberlat <- mantel_EC18("Amber_mapper_9.txt", "redo2009m2", "metadata_2009_to_12.txt", cols_to_test = list(c("latitude")),"bray_curtis") 
pamber3 <- mantel_distance_plot(amber3, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = 0.28, p = 0.007"))
pamberlat <- mantel_distance_plot(amberlat, y_title = "environmental distance (latitude)", point_size = point_size)+ labs(title = bquote(rho~" = 0.31, p = 0.004"))

amber3_WU <- mantel_EC18("Amber_mapper_9.txt", "redo2009m2", "metadata_2009_to_12.txt", cols_to_test = list(c("MinRH","latitude","MaxTemp")),"weighted_unifrac") 
amberlat_WU <- mantel_EC18("Amber_mapper_9.txt", "redo2009m2", "metadata_2009_to_12.txt", cols_to_test = list(c("latitude")),"weighted_unifrac") 
pamber3_WU <- mantel_distance_plot(amber3, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = 0.16, p = 0.07"))
pamberlat_WU <- mantel_distance_plot(amberlat, y_title = "environmental distance (latitude)", point_size = point_size)+ labs(title = bquote(rho~" = 0.45, p = 0.003"))
amber3_UU <- mantel_EC18("Amber_mapper_9.txt", "redo2009m2", "metadata_2009_to_12.txt", cols_to_test = list(c("MinRH","latitude","MaxTemp")),"unweighted_unifrac") 
amberlat_UU <- mantel_EC18("Amber_mapper_9.txt", "redo2009m2", "metadata_2009_to_12.txt", cols_to_test = list(c("latitude")),"unweighted_unifrac") 
pamber3_UU <- mantel_distance_plot(amber3, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = 0.11, p = 0.21"))
pamberlat_UU <- mantel_distance_plot(amberlat, y_title = "environmental distance (latitude)", point_size = point_size)+ labs(title = bquote(rho~" = -0.03, p = 0.57"))

####
## EC - all sig
####
## basically sig all
ec213 <- mantel_EC21seq("mapper_1and2.tsv","east_coast_overall3","metadata_2021_JMC.txt",list(c("MinRH","latitude","MaxTemp")),"bray_curtis")
ec21lat <- mantel_EC21seq("mapper_1and2.tsv","east_coast_overall3","metadata_2021_JMC.txt",list(c("latitude")),"bray_curtis")
pec213 <- mantel_distance_plot(ec213, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = 0.21, p = 0.001"))
pec21lat <- mantel_distance_plot(ec21lat, y_title = "environmental distance (latitude)", point_size = point_size)+ labs(title = bquote(rho~" = 0.35, p = 0.001"))

ec213_WU <- mantel_EC21seq("mapper_1and2.tsv","east_coast_overall3","metadata_2021_JMC.txt",list(c("MinRH","latitude","MaxTemp")),"weighted_unifrac")
ec21lat_WU <- mantel_EC21seq("mapper_1and2.tsv","east_coast_overall3","metadata_2021_JMC.txt",list(c("latitude")),"weighted_unifrac")
pec213_WU <- mantel_distance_plot(ec213, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = 0.13, p = 0.01"))
pec21lat_WU <- mantel_distance_plot(ec21lat, y_title = "environmental distance (latitude)", point_size = point_size)+ labs(title = bquote(rho~" = 0.38, p = 0.001"))
ec213_UU <- mantel_EC21seq("mapper_1and2.tsv","east_coast_overall3","metadata_2021_JMC.txt",list(c("MinRH","latitude","MaxTemp")),"unweighted_unifrac")
ec21lat_UU <- mantel_EC21seq("mapper_1and2.tsv","east_coast_overall3","metadata_2021_JMC.txt",list(c("latitude")),"unweighted_unifrac")
pec213_UU <- mantel_distance_plot(ec213, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = 0.10, p = 0.04"))
pec21lat_UU <- mantel_distance_plot(ec21lat, y_title = "environmental distance (latitude)", point_size = point_size)+ labs(title = bquote(rho~" = 0.10, p = 0.10"))

####
## Utah
####
utah3 <- mantel_UT20("mapper_file_JMCv3_edit.tsv", "utah_mantel500S", "metadata_2020.txt", list(c("MinRH","latitude","MaxTemp")), "bray_curtis")
utah3b <- mantel_UT20("mapper_file_JMCv3_edit.tsv", "utah_mantel500S", "metadata_2020.txt", list(c("latitude","MaxTemp")), "bray_curtis")
utahlat <- mantel_UT20("mapper_file_JMCv3_edit.tsv", "utah_mantel500S", "metadata_2020.txt", list(c("latitude")), "bray_curtis")
putah3 <- mantel_distance_plot(utah3, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = 0.13, p = 0.06"))
putah3b <- mantel_distance_plot(utah3b, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = 0.21, p = 0.001"))
putahlat <- mantel_distance_plot(utahlat, y_title = "environmental distance (latitude)", point_size = point_size)+ labs(title = bquote(rho~" = 0.21, p = 0.02"))

utah3_WU <- mantel_UT20("mapper_file_JMCv3_edit.tsv", "utah_mantel500S", "metadata_2020.txt", list(c("MinRH","latitude","MaxTemp")), "weighted_unifrac")
utah3b_WU <- mantel_UT20("mapper_file_JMCv3_edit.tsv", "utah_mantel500S", "metadata_2020.txt", list(c("latitude","MaxTemp")), "weighted_unifrac")
utahlat_WU <- mantel_UT20("mapper_file_JMCv3_edit.tsv", "utah_mantel500S", "metadata_2020.txt", list(c("latitude")), "weighted_unifrac")
putah3_WU <- mantel_distance_plot(utah3, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = 0.05, p = 0.21"))
putah3b_WU <- mantel_distance_plot(utah3b, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = 0.13, p = 0.03"))
putahlat_WU <- mantel_distance_plot(utahlat, y_title = "environmental distance (latitude)", point_size = point_size)+ labs(title = bquote(rho~" = 0.16, p = 0.047"))
utah3_UU <- mantel_UT20("mapper_file_JMCv3_edit.tsv", "utah_mantel500S", "metadata_2020.txt", list(c("MinRH","latitude","MaxTemp")), "unweighted_unifrac")
utah3b_UU <- mantel_UT20("mapper_file_JMCv3_edit.tsv", "utah_mantel500S", "metadata_2020.txt", list(c("latitude","MaxTemp")), "unweighted_unifrac")
utahlat_UU <- mantel_UT20("mapper_file_JMCv3_edit.tsv", "utah_mantel500S", "metadata_2020.txt", list(c("latitude")), "unweighted_unifrac")
putah3_UU <- mantel_distance_plot(utah3, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = -0.03, p = 0.63"))
putah3b_UU <- mantel_distance_plot(utah3b, y_title = "environmental distance", point_size = point_size)+ labs(title = bquote(rho~" = 0.04, p = 0.27"))
putahlat_UU <- mantel_distance_plot(utahlat, y_title = "environmental distance (latitude)", point_size = point_size)+ labs(title = bquote(rho~" = -0.08, p = 0.73"))


```

## Temperature
```{r}
temp_data <- readr::read_tsv(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/readin5_temperature.txt", show_col_types = F) %>%
  dplyr::select(-experiment, sample_type) %>%
  mutate(Genotype = as.factor(Genotype),
         Treatment = as.factor(Treatment),
         Vial = as.factor(Vial),
         Sex = as.factor(Sex),
         plate = as.factor(plate), 
         exp = as.factor(exp)
  )

temp_relative <- make_chart_stats_96_bar_jitter_violin(bac = temp_data, column = "Treatment", plotout = 2, yaxis = NULL, aceto_bottom = F,logplot = F, baradjust = 10, loglabeladjust = 10,baradjust_bottom = -0, yaxis_perc = c(0,1), perc_errorbars = NULL, add_jitter = T, add_violin = T) +   labs(x = "Temperature (°C)", y = "fractional abundance") +
  theme(legend.position = "none") +
  scale_x_discrete(labels = (c("15","25","32"))) 

temp_absolute <- make_chart_stats_96_bar_jitter_violin(bac = temp_data, column = "Treatment", plotout = 1, yaxis = NULL, aceto_bottom = F,logplot = F, baradjust = 50000, loglabeladjust = 10,baradjust_bottom = -10000, yaxis_perc = c(0,1), perc_errorbars = NULL, add_jitter = T, add_violin = T) +   labs(x = "Temperature (°C)", y = bquote(CFU~fly^-1)) +
  theme(legend.position = "none") +
  scale_x_discrete(labels = (c("15","25","32")))

for(sex in c("M","F")) {
  for (treatment in c("A","B","C")) {
    assign(x = paste0("tempplot_",treatment,"_",sex), 
           make_chart_stats_96_bar_jitter_violin(bac = temp_data %>% filter(Sex==sex,Treatment == treatment), column = "Genotype", plotout = 2, yaxis = NULL, aceto_bottom = F,logplot = F, baradjust = 10, loglabeladjust = 10,baradjust_bottom = -0, yaxis_perc = c(0,1), perc_errorbars = NULL, add_jitter = T, add_violin = T) +   coord_flip() + 
  labs(x = "Source location", y = "fractional abundance") +
  theme(legend.position = "none", axis.title = element_text(face = "bold")) + 
  scale_x_discrete(labels = rev(c("ME2","ME1","NH","MA","CT","PA","MD","VA")))
    )
        print(sex)
    cat(treatment)
    calculate_figuremeans(temp_data %>% mutate(Humidity = Treatment, Geography=Genotype), c(treatment), c(sex))

  }
}

i=1.3
#jpeg(h=9*i, w=6*i, "temp_figureS_relative.jpg", units = "in", res=300)
grid.arrange(tempplot_A_M + labs(tag="A"),
             tempplot_A_F + labs(tag="B"),
             tempplot_B_M + labs(tag="C"),
             tempplot_B_F + labs(tag="D"),
             tempplot_C_M + labs(tag="E"),
             tempplot_C_F + labs(tag="F"),
             widths = c(1,1),
             heights = c(1,1,1),
             layout_matrix=rbind(		
                c(1,2),
                c(3,4),
                c(5,6)
             )
)
#dev.off()

for(sex in c("M","F")) {
  for (treatment in c("A","B","C")) {
    assign(x = paste0("tempplotabun_",treatment,"_",sex), 
           make_chart_stats_96_bar_jitter_violin(bac = temp_data %>% filter(Sex==sex,Treatment == treatment), column = "Genotype", plotout = 1, yaxis = NULL, aceto_bottom = F,logplot = F, baradjust = 80000, loglabeladjust = 10,baradjust_bottom = -50000, yaxis_perc = c(0,1), perc_errorbars = NULL, add_jitter = T, add_violin = T) +   coord_flip() + 
  labs(x = "Source location", y = bquote(CFU~fly^-1)) +
  theme(legend.position = "none", axis.title = element_text(face = "bold")) + 
  scale_x_discrete(labels = rev(c("ME2","ME1","NH","MA","CT","PA","MD","VA"))) +
    ylim(c(-50000,1450000))
    )

  }
}
i=1.3
#jpeg(h=9*i, w=6*i, "temp_figureS_abundance.jpg", units = "in", res=300)
grid.arrange(tempplotabun_A_M + labs(tag="A"),
             tempplotabun_A_F + labs(tag="B"),
             tempplotabun_B_M + labs(tag="C"),
             tempplotabun_B_F + labs(tag="D"),
             tempplotabun_C_M + labs(tag="E"),
             tempplotabun_C_F + labs(tag="F"),
             widths = c(1,1),
             heights = c(1,1,1),
             layout_matrix=rbind(		
                c(1,2),
                c(3,4),
                c(5,6)
             )
)
#dev.off()

getwd()




#jpeg(h=9*i, w=6*i*2, "temp_figureS.jpg", units = "in", res=300)
grid.arrange(tempplot_A_M + labs(tag="A"),
             tempplot_A_F + labs(tag="B"),
             tempplot_B_M + labs(tag="C"),
             tempplot_B_F + labs(tag="D"),
             tempplot_C_M + labs(tag="E"),
             tempplot_C_F + labs(tag="F"),
             tempplotabun_A_M + labs(tag="G"),
             tempplotabun_A_F + labs(tag="H"),
             tempplotabun_B_M + labs(tag="I"),
             tempplotabun_B_F + labs(tag="J"),
             tempplotabun_C_M + labs(tag="K"),
             tempplotabun_C_F + labs(tag="L"),
             widths = c(1,1,1,1),
             heights = c(1,1,1),
             layout_matrix=rbind(		
                c(1,2,7,8),
                c(3,4,9,10),
                c(5,6,11,12)
             )
)
#dev.off()




```

## Table S4 top
```{r, eval = F}

filename = "temp24"
readin2_rename <- temp_data %>%
  mutate(codename = poscod) %>%
  filter(cfuA!=0 | cfuL!=0) %>%
  filter(!is.na(cfuA), !is.na(cfuL)) %>%
  mutate(total = cfuA + cfuL) %>%
  mutate(filtercol = "Y") %>% 
  mutate(repexp = paste(Genotype, Treatment,Vial,exp),codename = paste0(codename,"_"))

cfu_stats_for_qiime <- readin2_rename %>% dplyr::select(cfuA, cfuL) %>% t()
colnames(cfu_stats_for_qiime) <- readin2_rename$codename

cfu_stats <- readin2_rename

cfu_stats_for_qiime <- cbind(`#OTU ID`= c("aab","lb"), cfu_stats_for_qiime)
df_csfq <- data.frame(t(cfu_stats_for_qiime))[-1,] %>% mutate(cfuA = as.numeric(cfuA), cfuL = as.numeric(cfuL))

## write the new OTU table
write.table("# Constructed from BIOM file", paste0("otu_out_manual_",filename,".txt"), row.names = F, col.names = F, quote = F)
write.table(cfu_stats_for_qiime, paste0("otu_out_manual_",filename,".txt"), row.names = F, col.names = T, quote = F, append=T, sep = "\t")

 ## write the cfumapper
write.table(cfu_stats %>% dplyr::select(-cfuA,-cfuL) %>% dplyr::rename(`#SampleID` = codename) %>% relocate(`#SampleID`), paste0("cfu_mapper_",filename,".tsv"), row.names = F, col.names = T, quote = F, sep = "\t")

hist(log10(cfu_stats %>% dplyr::pull(total) %>% as.numeric(as.character())+1), breaks=50)
```
```{sh, eval = F}
cd ~/Dropbox/
filename=temp24
conda activate qiime2-2021.11

## convert manual table to biom
biom convert -i otu_out_manual_$filename"".txt -o cfu_asv_table_$filename"".biom --to-hdf5 --table-type="OTU table"

## convert biom table to qza
qiime tools import --input-path cfu_asv_table_$filename"".biom --type 'FeatureTable[Frequency]' --input-format BIOMV210Format --output-path cfu_asv_table_$filename"".qza

## run the data
bash ~/Dropbox/sequencing/16sscripts/bray_curtis.sh cfu_asv_table_$filename"" temp24 10000 cfu_mapper_$filename"".tsv "filtercol = 'Y'" 
```
```{r, eval = F}
## Table S9
bc_analysis("temp24", paste0("cfu_mapper_",filename,".tsv"), "~ Genotype * Treatment * Sex + exp/plate/repexp", permutations = 1000)
vegan::adonis2(formula = as.formula(paste0("flies_prerarefied_dm ", formula)), data = flies_unifrac_table, permutations = permutations)
#                         Df SumOfSqs      R2       F   Pr(>F)    
# Genotype                 7   20.937 0.09706 19.0055 0.000999 ***
# Treatment                2   15.453 0.07164 49.0972 0.000999 ***
# Sex                      1    0.461 0.00214  2.9298 0.032967 *  
# exp                      2    9.927 0.04602 31.5387 0.000999 ***
# Genotype:Treatment      13    6.358 0.02948  3.1078 0.000999 ***
# Genotype:Sex             7    1.001 0.00464  0.9084 0.597403    
# Treatment:Sex            2    1.390 0.00645  4.4175 0.001998 ** 
# exp:plate                3    2.836 0.01315  6.0075 0.000999 ***
# Genotype:Treatment:Sex  13    2.047 0.00949  1.0005 0.479520    
# exp:plate:repexp        81   35.374 0.16399  2.7750 0.000999 ***
# Residual               762  119.921 0.55595                     
# Total                  893  215.705 1.00000                     
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 1000
# 
# adonis2(formula = as.formula(paste0("flies_rarefied_dm ", formula)), data = flies_unifrac_table, permutations = permutations)
#                         Df SumOfSqs      R2       F   Pr(>F)    
# Genotype                 7    8.495 0.14318 29.7535 0.000999 ***
# Treatment                2    4.805 0.08098 58.9001 0.000999 ***
# Sex                      1    0.122 0.00206  2.9899 0.072927 .  
# exp                      2    1.567 0.02641 19.2100 0.000999 ***
# Genotype:Treatment      13    1.555 0.02621  2.9331 0.000999 ***
# Genotype:Sex             7    0.388 0.00655  1.3607 0.203796    
# Treatment:Sex            2    0.186 0.00313  2.2755 0.100899    
# exp:plate                3    0.287 0.00484  2.3476 0.073926 .  
# Genotype:Treatment:Sex  13    0.849 0.01431  1.6013 0.065934 .  
# exp:plate:repexp        81   10.976 0.18500  3.3225 0.000999 ***
# Residual               738   30.100 0.50733                     
# Total                  869   59.331 1.00000                     
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

```

## Photoperiod
```{r}

phot_data <- readr::read_csv(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/readin5_photoperiod.csv", show_col_types = F) %>% 
 dplyr::select(-`...1`, experiment, sample_type) %>%
  mutate(Genotype = as.factor(Genotype),
         Treatment = as.factor(Treatment),
         Sex = as.factor(Sex),
         plate = as.factor(plate), 
         exp = as.factor(exp)
  )

photoperiod_relative <- make_chart_stats_96_bar_jitter_violin(bac = phot_data, column = "Treatment", plotout = 2, yaxis = NULL, aceto_bottom = F,logplot = F, baradjust = 10, loglabeladjust = 10,baradjust_bottom = -0, yaxis_perc = c(0,1), perc_errorbars = NULL, add_jitter = T, add_violin = T) +   labs(x = "Light:dark cycle (h)", y = "fractional abundance") +
  theme(legend.position = "none") + 
  scale_x_discrete(labels = (c("1:23","12:12", "23:1"))) 

photoperiod_abundance <- make_chart_stats_96_bar_jitter_violin(bac = phot_data, column = "Treatment", plotout = 1, yaxis = NULL, aceto_bottom = F,logplot = F, baradjust = 6000, loglabeladjust = 10,baradjust_bottom = -2000, yaxis_perc = c(0,1), perc_errorbars = NULL, add_jitter = T, add_violin = T) +     labs(x = "Light:dark cycle (h)", y = bquote(CFU~fly^-1)) +
  theme(legend.position = "none") + 
  scale_x_discrete(labels = (c("1:23","12:12", "23:1"))) 


for(sex in c("M","F")) {
  for (treatment in c("A","B","C")) {
    assign(x = paste0("photoplot_",treatment,"_",sex), 
           make_chart_stats_96_bar_jitter_violin(bac = phot_data %>% filter(Sex==sex,Treatment == treatment), column = "Genotype", plotout = 2, yaxis = NULL, aceto_bottom = F,logplot = F, baradjust = 10, loglabeladjust = 10,baradjust_bottom = -0, yaxis_perc = c(0,1), perc_errorbars = NULL, add_jitter = T, add_violin = T) +   coord_flip() + 
  labs(x = "Source location", y = "fractional abundance") +
  theme(legend.position = "none") + 
  scale_x_discrete(labels = rev(c("ME2","ME1","NH","MA","CT","PA","MD","VA")))
    )
  }
}

for(sex in c("M","F")) {
  for (treatment in c("A","B","C")) {
    assign(x = paste0("photoplotabun_",treatment,"_",sex), 
           make_chart_stats_96_bar_jitter_violin(bac = phot_data %>% filter(Sex==sex,Treatment == treatment), column = "Genotype", plotout = 1, yaxis = NULL, aceto_bottom = F,logplot = F, baradjust = 20000, loglabeladjust = 10,baradjust_bottom = -20000, yaxis_perc = c(0,1), perc_errorbars = NULL, add_jitter = T, add_violin = T) +   coord_flip() + 
  labs(x = "Source location", y = bquote(CFU~fly^-1)) +
  theme(legend.position = "none", axis.title = element_text(face = "bold")) + 
  scale_x_discrete(labels = rev(c("ME2","ME1","NH","MA","CT","PA","MD","VA"))) +
    ylim(c(-25000,300000))
    )
        print(sex)
    cat(treatment)
    calculate_figuremeans(phot_data %>% mutate(Humidity = Treatment, Geography=Genotype), c(treatment), c(sex))

  }
}

i=1.3
#jpeg(h=9*i, w=6*i, "photoperiod_figureS.jpg", units = "in", res=300)
grid.arrange(photoplot_A_M + labs(tag="A"),
             photoplot_A_F + labs(tag="B"),
             photoplot_B_M + labs(tag="C"),
             photoplot_B_F + labs(tag="D"),
             photoplot_C_M + labs(tag="E"),
             photoplot_C_F + labs(tag="F"),
             widths = c(1,1),
             heights = c(1,1,1),
             layout_matrix=rbind(		
                c(1,2),
                c(3,4),
                c(5,6)
             )
)
#dev.off()

i=1.3
#jpeg(h=9*i, w=6*i, "photoperiod_figureS_abundance.jpg", units = "in", res=300)
grid.arrange(photoplotabun_A_M + labs(tag="A"),
             photoplotabun_A_F + labs(tag="B"),
             photoplotabun_B_M + labs(tag="C"),
             photoplotabun_B_F + labs(tag="D"),
             photoplotabun_C_M + labs(tag="E"),
             photoplotabun_C_F + labs(tag="F"),
             widths = c(1,1),
             heights = c(1,1,1),
             layout_matrix=rbind(		
                c(1,2),
                c(3,4),
                c(5,6)
             )
)
#dev.off()

i=1.3
#jpeg(h=9*i, w=6*i*2, "photoperiod_figureS.jpg", units = "in", res=300)
grid.arrange(photoplot_A_M + labs(tag="A"),
             photoplot_A_F + labs(tag="B"),
             photoplot_B_M + labs(tag="C"),
             photoplot_B_F + labs(tag="D"),
             photoplot_C_M + labs(tag="E"),
             photoplot_C_F + labs(tag="F"),
             photoplotabun_A_M + labs(tag="G"),
             photoplotabun_A_F + labs(tag="H"),
             photoplotabun_B_M + labs(tag="I"),
             photoplotabun_B_F + labs(tag="J"),
             photoplotabun_C_M + labs(tag="K"),
             photoplotabun_C_F + labs(tag="L"),
             widths = c(1,1,1,1),
             heights = c(1,1,1),
             layout_matrix=rbind(		
                c(1,2,7,8),
                c(3,4,9,10),
                c(5,6,11,12)
                )
)
#dev.off()

```

## Table S4 bottom
```{r, eval = F}

filename = "phot24"
readin2_rename <- temp_data %>%
  mutate(codename = poscod) %>%
  filter(cfuA!=0 | cfuL!=0) %>%
  filter(!is.na(cfuA), !is.na(cfuL)) %>%
  mutate(total = cfuA + cfuL) %>%
  mutate(filtercol = "Y") %>% 
  mutate(repexp = paste(Genotype, Treatment,Vial,exp),codename = paste0(codename,"_"))

cfu_stats_for_qiime <- readin2_rename %>% dplyr::select(cfuA, cfuL) %>% t()
colnames(cfu_stats_for_qiime) <- readin2_rename$codename

cfu_stats <- readin2_rename

cfu_stats_for_qiime <- cbind(`#OTU ID`= c("aab","lb"), cfu_stats_for_qiime)
df_csfq <- data.frame(t(cfu_stats_for_qiime))[-1,] %>% mutate(cfuA = as.numeric(cfuA), cfuL = as.numeric(cfuL))

## write the new OTU table
write.table("# Constructed from BIOM file", paste0("otu_out_manual_",filename,".txt"), row.names = F, col.names = F, quote = F)
write.table(cfu_stats_for_qiime, paste0("otu_out_manual_",filename,".txt"), row.names = F, col.names = T, quote = F, append=T, sep = "\t")

 ## write the cfumapper
write.table(cfu_stats %>% dplyr::select(-cfuA,-cfuL) %>% dplyr::rename(`#SampleID` = codename) %>% relocate(`#SampleID`), paste0("cfu_mapper_",filename,".tsv"), row.names = F, col.names = T, quote = F, sep = "\t")

hist(log10(cfu_stats %>% dplyr::pull(total) %>% as.numeric(as.character())+1), breaks=50)
table(cfu_stats %>% dplyr::pull(total) %>% as.numeric(as.character()))

```
```{sh, eval = F}
cd ~/Dropbox/
filename=phot24
conda activate qiime2-2021.11

## convert manual table to biom
biom convert -i otu_out_manual_$filename"".txt -o cfu_asv_table_$filename"".biom --to-hdf5 --table-type="OTU table"

## convert biom table to qza
qiime tools import --input-path cfu_asv_table_$filename"".biom --type 'FeatureTable[Frequency]' --input-format BIOMV210Format --output-path cfu_asv_table_$filename"".qza

## run the data
bash ~/Dropbox/sequencing/16sscripts/bray_curtis.sh cfu_asv_table_$filename"" phot24 4000 cfu_mapper_$filename"".tsv "filtercol = 'Y'" 

bash ~/Dropbox/sequencing/16sscripts/bray_curtis.sh cfu_asv_table_$filename"" phot24m 4000 cfu_mapper_$filename"".tsv "filtercol = 'Y' AND Sex='M'" 

bash ~/Dropbox/sequencing/16sscripts/bray_curtis.sh cfu_asv_table_$filename"" phot24f 4000 cfu_mapper_$filename"".tsv "filtercol = 'Y' AND Sex='F'" 

```
```{r, eval = F}
## Table S9
bc_analysis("phot24", paste0("cfu_mapper_",filename,".tsv"), "~ Genotype * Treatment * Sex + exp/plate/repexp", permutations = 1000)
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 1000
# 
# vegan::adonis2(formula = as.formula(paste0("flies_prerarefied_dm ", formula)), data = flies_unifrac_table, permutations = permutations)
#                          Df SumOfSqs      R2       F   Pr(>F)    
# Genotype                  7   13.110 0.04981 14.4451 0.000999 ***
# Treatment                 2    1.497 0.00569  5.7742 0.000999 ***
# Sex                       1    0.198 0.00075  1.5250 0.180819    
# exp                       2   17.347 0.06591 66.8961 0.000999 ***
# Genotype:Treatment       14    6.155 0.02339  3.3910 0.000999 ***
# Genotype:Sex              7    1.148 0.00436  1.2647 0.186813    
# Treatment:Sex             2    0.573 0.00218  2.2113 0.032967 *  
# exp:plate                 3    5.068 0.01926 13.0293 0.000999 ***
# Genotype:Treatment:Sex   14    2.289 0.00870  1.2610 0.118881    
# exp:plate:repexp        141   59.314 0.22536  3.2444 0.000999 ***
# Residual               1207  156.497 0.59460                     
# Total                  1400  263.197 1.00000                     
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1
# Permutation test for adonis under reduced model
# Terms added sequentially (first to last)
# Permutation: free
# Number of permutations: 1000
# 
# adonis2(formula = as.formula(paste0("flies_rarefied_dm ", formula)), data = flies_unifrac_table, permutations = permutations)
#                          Df SumOfSqs      R2        F   Pr(>F)    
# Genotype                  7    1.761 0.02602   7.4237 0.000999 ***
# Treatment                 2    0.677 0.01000   9.9853 0.000999 ***
# Sex                       1    0.043 0.00064   1.2771 0.252747    
# exp                       2   10.691 0.15790 157.7005 0.000999 ***
# Genotype:Treatment       14    1.497 0.02210   3.1537 0.000999 ***
# Genotype:Sex              7    0.602 0.00889   2.5359 0.012987 *  
# Treatment:Sex             2    0.030 0.00045   0.4491 0.639361    
# exp:plate                 3    2.579 0.03809  25.3585 0.000999 ***
# Genotype:Treatment:Sex   14    0.737 0.01089   1.5533 0.085914 .  
# exp:plate:repexp        141   10.313 0.15231   2.1577 0.000999 ***
# Residual               1144   38.778 0.57272                      
# Total                  1337   67.707 1.00000                      
# ---
# Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1


bc_analysis("phot24f", paste0("cfu_mapper_",filename,".tsv"), "~ Genotype * Treatment + exp/plate/repexp", permutations = 1000)
bc_analysis("phot24m", paste0("cfu_mapper_",filename,".tsv"), "~ Genotype * Treatment + exp/plate/repexp", permutations = 1000)

```

## Overall figure
```{r}
#points_data <- read_xlsx('~/Dropbox/east_coast_flies_manuscript/megamap_latlong.xlsx', sheet = "Sheet7")
points_data <- readr::read_csv(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/megamap_latlong_sheet7.csv", show_col_types = F) 

amberpoints <- ggplot(points_data %>% filter(plot=="amber"), aes(x = column, y = row, shape = symbol)) +   geom_point(size = 2) + scale_shape_identity() +   theme_cowplot() +  theme(legend.position = "none")+   xlim(c(0.88,1.12)) + ylim(c(1,11.1))
vpoints <- ggplot(points_data %>% filter(plot=="vineyard"), aes(x = column, y = row, shape = symbol)) +   geom_point(size = 2) + scale_shape_identity() +   theme_cowplot() + theme(legend.position = "none")+   xlim(c(0.88,1.12)) + ylim(c(1,11))
opoints <- ggplot(points_data %>% filter(plot=="orchard"), aes(x = column, y = row, shape = symbol)) +   geom_point(size = 2) + scale_shape_identity() +   theme_cowplot() + theme(legend.position = "none")+   xlim(c(0.88,1.12)) + ylim(c(1,11))
ec21points <- ggplot(points_data %>% filter(plot=="ec"), aes(x = column, y = row, shape = symbol)) +   geom_point(size = 2) + scale_shape_identity() +   theme_cowplot() +   theme(legend.position = "none")+   xlim(c(0.88,1.12)) + ylim(c(1,11))

plot_legend <- g_legend(ec2021 + theme(legend.position = "bottom", legend.text = element_text(size = 12), legend.title = element_blank()) + guides(fill=guide_legend(nrow=6,byrow=TRUE)))

plot_legend <- g_legend(ec2021 + theme(legend.position = "bottom", legend.text = element_text(size = 12), legend.title = element_blank()) + guides(fill=guide_legend(ncol=3,byrow=F)) + scale_fill_manual(name = "", values = c("black",predefined_colors[-c(1,2,4,12)]), labels = c("other", bquote(italic(Pasteurellales)), bquote(italic(Enterobacteriaceae)),bquote(italic(Lactobacillales)), bquote(italic(Lactobacillaceae)), bquote(italic(Enterococcus)), bquote(italic(Leuconostoc)), bquote(italic(Weissella)), bquote(italic(Lactobacillus)), bquote(italic(Acetobacteraceae)), bquote(italic(Commensalibacter)), bquote(italic(Gluconobacter)), bquote(italic(Komagataeibacter)), bquote(italic(Acetobacter)))))
plot_legend_ec <- g_legend(ec2021 + theme(legend.position = "bottom", legend.text = element_text(size = 12)) + guides(fill=guide_legend(ncol=2,byrow=F)) + scale_fill_manual(name = "           ", values = c("black",predefined_colors[-c(1,2,4,12)]), labels = c("other", bquote(italic(Pasteurellales)), bquote(italic(Enterobacteriaceae)),bquote(italic(Lactobacillales)), bquote(italic(Lactobacillaceae)), bquote(italic(Enterococcus)), bquote(italic(Leuconostoc)), bquote(italic(Weissella)), bquote(italic(Lactobacillus)), bquote(italic(Acetobacteraceae)), bquote(italic(Commensalibacter)), bquote(italic(Gluconobacter)), bquote(italic(Komagataeibacter)), bquote(italic(Acetobacter)))))

plot(plot_legend_ec)
fig1tagsize=25
mantel_size = 12

#jpeg(h=8*.875*1.5*1.3*.9, w=28*.875*.75*.9, "~/Dropbox/east_coast_flies_manuscript/Figure1_aug2024b.jpg", units = "in", res=300)
grid.arrange(ecmap + labs(tag = "A") + theme(plot.tag = element_text(face = "bold", size = fig1tagsize)), 
             amberpoints + theme(legend.position = "none", strip.text = element_blank(), panel.spacing = unit(-.1, "lines"), axis.title.y = element_blank(), axis.text = element_blank(), axis.line = element_line(color = "white"), axis.text.x = element_text(size = 12, angle = 90, color = "white"), axis.ticks = element_line(color = "white"), axis.title.x = element_text(color = "white"), axis.line.y = element_blank(), axis.ticks.y = element_blank()) + labs(tag = "B", y = " ")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize)),
             amberflies + theme(legend.position = "none", strip.text = element_blank(), panel.spacing = unit(-.1, "lines"), axis.text.x = element_text(size =12, angle = 90)),
             vpoints + theme(legend.position = "none", strip.text = element_blank(), panel.spacing = unit(-.1, "lines"), axis.title.y = element_blank(), axis.text = element_blank(), axis.line = element_line(color = "white"), axis.text.x = element_text(size = 12, angle = 90, color = "white"), axis.ticks = element_line(color = "white"), axis.title.x = element_text(color = "white"), axis.line.y = element_blank(), axis.ticks.y = element_blank()) + labs(tag = "C", y = " ")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize)),
             vineyard + theme(legend.position = "none", strip.text = element_blank(), panel.spacing = unit(-.1, "lines"), axis.text.x = element_text(size =12, angle = 90), axis.title.y = element_blank()),
             opoints+ theme(legend.position = "none", strip.text = element_blank(), panel.spacing = unit(-.1, "lines"), axis.title.y = element_blank(), axis.text = element_blank(), axis.line = element_line(color = "white"), axis.text.x = element_text(size = 12, angle = 90, color = "white"), axis.ticks = element_line(color = "white"), axis.title.x = element_text(color = "white"), axis.line.y = element_blank(), axis.ticks.y = element_blank()) + labs(tag = "D", y = " ")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize)),
             henryorchard+ theme(legend.position = "none", strip.text = element_blank(), panel.spacing = unit(-.1, "lines"), axis.text.x = element_text(size =12, angle = 90), axis.title.y = element_blank()),
             ec21points + theme(legend.position = "none", strip.text = element_blank(), panel.spacing = unit(-.1, "lines"), axis.title.y = element_blank(), axis.text = element_blank(), axis.line = element_line(color = "white"), axis.text.x = element_text(size = 12, angle = 90, color = "white"), axis.ticks = element_line(color = "white"), axis.title.x = element_text(color = "white"), axis.line.y = element_blank(), axis.ticks.y = element_blank()) + labs(tag = "E", y = " ")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize)),
             ec2021+ theme(legend.position = "none", strip.text = element_blank(), panel.spacing = unit(-.1, "lines"), axis.text.x = element_text(size =12, angle = 90)),
             pamberlat + theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (latitude)", y = "Microbiota distance", tag = "F")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "blue", size = mantel_size * 1.15)),   
             pamber3+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (lat., temp., hum.)", y = "Microbiota distance", tag = "J")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "blue", size = mantel_size * 1.15)),
             phenryvlat+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (latitude)", y = "Microbiota distance", tag = "G")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "black", size = mantel_size * 1.15)),
             phenryv3+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (lat., temp., hum.)", y = "Microbiota distance", tag = "K")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "blue", size = mantel_size * 1.15)),
             phenryorchardlat+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (latitude)", y = "Microbiota distance", tag = "H")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "black", size = mantel_size * 1.15)),
             phenryorchard3+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (lat., temp., hum.)", y = "Microbiota distance", tag = "L")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "blue", size = mantel_size * 1.15)),
             pec21lat+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (latitude)", y = "Microbiota distance", tag = "I")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "blue", size = mantel_size * 1.15)),
             pec213+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (lat., temp., hum.)", y = "Microbiota distance", tag = "M")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "blue", size = mantel_size * 1.15)),
             plot_legend_ec,
             temp_relative + labs(tag="N") + theme(plot.tag = element_text(face = "bold", size = fig1tagsize)),
             temp_absolute + labs(tag="O") + theme(plot.tag = element_text(face = "bold", size = fig1tagsize)),
             photoperiod_relative + labs(tag="P") + theme(plot.tag = element_text(face = "bold", size = fig1tagsize)),
             photoperiod_abundance + labs(tag="Q") + theme(plot.tag = element_text(face = "bold", size = fig1tagsize)),
            layout_matrix=rbind(		
                c(1,1,2,3,4,5,6,7,7,8,9),
                c(1,1,10,10,12,12,14,14,14,16,16),
                c(18,18,10,10,12,12,14,14,14,16,16),
                c(18,18,11,11,13,13,15,15,15,17,17),
                c(19,20,20,20,20,21,21,21,22,22,22)
             ),
             widths = c(.4625,.09,.07,.25,.07,.25,.09,.1225,.25-.1225,.07,.25),
             heights = c(2,.65,.65,1.3,2)
)
#dev.off()

```

## Figure SX - UU and WU mantel results
```{r}
#jpeg(h=8*.875*1.5, w=28*.875*.75, "~/Dropbox/east_coast_flies_manuscript/Figure1_july2024b_supp.jpg", units = "in", res=300)
grid.arrange(pamberlat_WU + theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (latitude)", y = "Microbiota distance", tag = "A")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "blue", size = mantel_size * 1.15)),   
             pamber3_WU+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (lat., temp., hum.)", y = "Microbiota distance", tag = "E")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "black", size = mantel_size * 1.15)),
             phenryvlat_WU+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (latitude)", y = "Microbiota distance", tag = "B")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "black", size = mantel_size * 1.15)),
             phenryv3_WU+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (lat., temp., hum.)", y = "Microbiota distance", tag = "F")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "blue", size = mantel_size * 1.15)),
             phenryorchardlat_WU+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (latitude)", y = "Microbiota distance", tag = "C")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "black", size = mantel_size * 1.15)),
             phenryorchard3_WU+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (lat., temp., hum.)", y = "Microbiota distance", tag = "G")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "blue", size = mantel_size * 1.15)),
             pec21lat_WU+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (latitude)", y = "Microbiota distance", tag = "D")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "blue", size = mantel_size * 1.15)),
             pec213_WU+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (lat., temp., hum.)", y = "Microbiota distance", tag = "H")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "blue", size = mantel_size * 1.15)),
             pamberlat_UU + theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (latitude)", y = "Microbiota distance", tag = "I")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "black", size = mantel_size * 1.15)),   
             pamber3_UU+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (lat., temp., hum.)", y = "Microbiota distance", tag = "M")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "black", size = mantel_size * 1.15)),
             phenryvlat_UU+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (latitude)", y = "Microbiota distance", tag = "J")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "black", size = mantel_size * 1.15)),
             phenryv3_UU+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (lat., temp., hum.)", y = "Microbiota distance", tag = "N")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "blue", size = mantel_size * 1.15)),
             phenryorchardlat_UU+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (latitude)", y = "Microbiota distance", tag = "K")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "blue", size = mantel_size * 1.15)),
             phenryorchard3_UU+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (lat., temp., hum.)", y = "Microbiota distance", tag = "O")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "black", size = mantel_size * 1.15)),
             pec21lat_UU+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (latitude)", y = "Microbiota distance", tag = "L")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "black", size = mantel_size * 1.15)),
             pec213_UU+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (lat., temp., hum.)", y = "Microbiota distance", tag = "P")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "blue", size = mantel_size * 1.15)),
            layout_matrix=rbind(
              c(1,3,5,7),
              c(2,4,6,8),
              c(9,11,13,15),
              c(10,12,14,16)
             ),
             widths = c(1,1,1,1),
             heights = c(1,1,1,1)
)
#dev.off()

```

## Figure SX - Utah flies figure
```{r}
plot_ut_legend <- g_legend(ut2020 + theme(legend.position = "bottom", legend.text = element_text(size = 12), legend.title = element_blank()) + guides(fill=guide_legend(ncol=3,byrow=TRUE)))
plot(plot_ut_legend)
utpoints <- ggplot(points_data %>% filter(plot=="utah"), aes(x = column, y = row, shape = symbol)) +   geom_point(size = 2) + scale_shape_identity() +   theme_cowplot() + theme(legend.position = "none")+   xlim(c(0.88,1.12)) + ylim(c(3.8,8.3))

#jpeg(h=8*.875, w=28*.875*.75*.75, "~/Dropbox/east_coast_flies_manuscript/FigureSX_utah_july2024.jpg", units = "in", res=300)
   grid.arrange(utmap + labs(tag = "A")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize)),
                utpoints + theme(legend.position = "none", strip.text = element_blank(), panel.spacing = unit(-.1, "lines"), axis.title.y = element_blank(), axis.text = element_blank(), axis.line = element_line(color = "white"), axis.text.x = element_text(size = 12, angle = 90, color = "white"), axis.ticks = element_line(color = "white"), axis.title.x = element_text(color = "white"), axis.line.y = element_blank(), axis.ticks.y = element_blank()) + labs(tag = "B", y = " ")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize)),
                putahlat + theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (latitude)", y = "microbiota distance", tag = "C")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "blue", size = mantel_size * 1.15)),
                putah3+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (lat., temp., hum.)", y = "microbiota distance", tag = "E")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "blue", size = mantel_size * 1.15)),
                ut2020 + theme(legend.position = "none", strip.text = element_blank(), panel.spacing = unit(-.1, "lines"), axis.text.x = element_text(size =12, angle = 90)),
                plot_ut_legend,
                putah3b+ theme(text = element_text(size=mantel_size*1.15), axis.text = element_text(size=mantel_size)) + labs(x = "Env. distance\n (lat., temp.)", y = "microbiota distance", tag = "D")+ theme(plot.tag = element_text(face = "bold", size = fig1tagsize), plot.title = element_text(color = "blue", size = mantel_size * 1.15)),

               layout_matrix=rbind(		
                c(1,2,5,7),
                c(1,3,3,4),
                c(6,3,3,4)
             ),
             widths = c(.3,.07,.18,.25),
             heights = c(1.5,.5,1)
   )
#dev.off()
```

## Figure SX - correlations figure
```{r}
annotationsize = 2
cortest_amber

#jpeg(h=8*.875*.5, w=28*.875*.75*.75, "~/Dropbox/east_coast_flies_manuscript/FigureScorX_utah_july2024.jpg", units = "in", res=300)
   grid.arrange(corplot_amber + labs(tag = "A", x = bquote(italic(Acetobacter)~"reads"), y = bquote(italic(Commensalibacter)~"reads")) + theme_cowplot() + ylim(c(-20,500)) + xlim(c(0,500)) +
                  ggplot2::annotate(geom = "text", x = 500, y = 500, hjust=1, vjust=1, label = paste0("S = ",unname(round(cortest_amber$estimate,2)),"\np =  ", unname(round(cortest_amber$p.value,3))), size = 5),
                corplot_vineyard + labs(tag = "B", x = bquote(italic(Acetobacter)~"reads"), y = bquote(italic(Commensalibacter)~"reads")) + theme_cowplot() + ylim(c(-20,500)) + xlim(c(0,500)) +
                  ggplot2::annotate(geom = "text", x = 500, y = 500, hjust=1, vjust=1, label = paste0("S = ",unname(round(cortest_vineyard$estimate,2)),"\np =  ", unname(round(cortest_vineyard$p.value,3))), size = 5),
                corplot_orchard + labs(tag = "C", x = bquote(italic(Acetobacter)~"reads"), y = bquote(italic(Commensalibacter)~"reads")) + theme_cowplot() + ylim(c(-20,500)) + xlim(c(0,500)) +
                  ggplot2::annotate(geom = "text", x = 500, y = 500, hjust=1, vjust=1, label = paste0("S = ",unname(round(cortest_orchard$estimate,2)),"\np =  ", unname(round(cortest_orchard$p.value,4))), size = 5),
                corplot_ec + labs(tag = "D", x = bquote(italic(Acetobacter)~"reads"), y = bquote(italic(Commensalibacter)~"reads")) + theme_cowplot() + ylim(c(-20,500)) + xlim(c(0,500)) +
                  ggplot2::annotate(geom = "text", x = 500, y = 500, hjust=1, vjust=1, label = paste0("S = ",unname(round(cortest_ec$estimate,2)),"\np =  ", unname(round(cortest_ec$p.value,2))), size = 5),
                corplot_UT + labs(tag = "E", x = bquote(italic(Acetobacter)~"reads"), y = bquote(italic(Commensalibacter)~"reads")) + theme_cowplot() + ylim(c(-20,500)) + xlim(c(0,500)) +
                  ggplot2::annotate(geom = "text", x = 500, y = 500, hjust=1, vjust=1, label = paste0("S = ",unname(round(cortest_UT$estimate,2)),"\np =  ", unname(round(cortest_UT$p.value,2))), size = 5),
               layout_matrix=rbind(		
                c(1,2,3,4,5)
             ),
             widths = c(1,1,1,1,1),
             heights = c(1)
   )
#dev.off()
```

# Figure 2
## Determine which samples go with each other
```{r}
#reference_table <- read.table('~/Dropbox/sequencing/2022-08-ECrun1and2/mapper_1and2_WendyJMC.tsv', fill = T, sep = "\t", comment.char = "", header=T) %>% filter(totalreads_noW > 500, numericsubsite!="<NA>")
reference_table <- readr::read_tsv(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/mapper_1and2_WendyJMC.tsv",skip = 0) %>% filter(totalreads_noW > 500, numericsubsite!="<NA>")
```

## bash 1
```{r, eval = F}
conda activate qiime2-2022.11

cd ~/Dropbox/sequencing/2022-08-ECrun1and2
name=ec1and2
mapper=mapper_1and2.tsv
seqsdir=Fastq

## starved
bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_starved_fruit_S 500 $mapper "filtercolumn IN ('ec1349','ec1356','ec1380','ec1396','ec1411','ec1420','ec1427','ec1435','ec1346','ec1361','ec1362','ec1378','ec1393','ec1409','ec1425','ec1301','ec1309','ec1317','ec1298','ec1314','ec1359','ec1390','ec1422','ec1437','ec1249','ec1257','ec1265','ec1368','ec1392','ec1400','ec1408','ec1424','eti111','eti121','eti125','ec1442','eti146','ec1489','ec1497','ec1449','ec1513','ec1457','eti145','ec1508','ec1468','ec1516','ec1524','eti149','ec1492','ec1500','eti151','eti152','ec1502','eti153','eti154','ec1530','ec1498','ec1514','eti156','ec1462','ec1470','eti162','ec1469','ec1493','ec1517','ec1525','ec1533','eti164','eti165','eti167') and sample_type IN ('flies')" no
bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_starved_fruit_F 500 $mapper "filtercolumn IN ('ec1349','ec1356','ec1380','ec1396','ec1411','ec1420','ec1427','ec1435','ec1346','ec1361','ec1362','ec1378','ec1393','ec1409','ec1425','ec1301','ec1309','ec1317','ec1298','ec1314','ec1359','ec1390','ec1422','ec1437','ec1249','ec1257','ec1265','ec1368','ec1392','ec1400','ec1408','ec1424','eti111','eti121','eti125','ec1442','eti146','ec1489','ec1497','ec1449','ec1513','ec1457','eti145','ec1508','ec1468','ec1516','ec1524','eti149','ec1492','ec1500','eti151','eti152','ec1502','eti153','eti154','ec1530','ec1498','ec1514','eti156','ec1462','ec1470','eti162','ec1469','ec1493','ec1517','ec1525','ec1533','eti164','eti165','eti167') and sample_type IN ('fruit')" no
bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_starved_fruit 500 $mapper "filtercolumn IN ('ec1349','ec1356','ec1380','ec1396','ec1411','ec1420','ec1427','ec1435','ec1346','ec1361','ec1362','ec1378','ec1393','ec1409','ec1425','ec1301','ec1309','ec1317','ec1298','ec1314','ec1359','ec1390','ec1422','ec1437','ec1249','ec1257','ec1265','ec1368','ec1392','ec1400','ec1408','ec1424','eti111','eti121','eti125','ec1442','eti146','ec1489','ec1497','ec1449','ec1513','ec1457','eti145','ec1508','ec1468','ec1516','ec1524','eti149','ec1492','ec1500','eti151','eti152','ec1502','eti153','eti154','ec1530','ec1498','ec1514','eti156','ec1462','ec1470','eti162','ec1469','ec1493','ec1517','ec1525','ec1533','eti164','eti165','eti167')" no

bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_starved_unstarved_S 500 $mapper "filtercolumn IN ('ec1356','ec1380','ec1396','ec1411','ec1346','ec1361','ec1377','ec1378','ec1393','ec1409','ec1425','ec1298','ec1314','ec1359','ec1390','ec1422','ec1437','ec1408','eti101','eti111','eti114','eti121','eti125','ec1364','ec1388','ec1404','ec1419','ec1354','ec1369','ec1385','ec1386','ec1401','ec1417','ec1433','ec1306','ec1322','ec1367','ec1398','ec1430','ec1350','ec1416','eti104','eti112','eti110','eti122','eti126') and starved IN ('starved')" no
bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_starved_unstarved_U 500 $mapper "filtercolumn IN ('ec1356','ec1380','ec1396','ec1411','ec1346','ec1361','ec1377','ec1378','ec1393','ec1409','ec1425','ec1298','ec1314','ec1359','ec1390','ec1422','ec1437','ec1408','eti101','eti111','eti114','eti121','eti125','ec1364','ec1388','ec1404','ec1419','ec1354','ec1369','ec1385','ec1386','ec1401','ec1417','ec1433','ec1306','ec1322','ec1367','ec1398','ec1430','ec1350','ec1416','eti104','eti112','eti110','eti122','eti126') and starved IN ('unstarved')" no
bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_starved_unstarved 500 $mapper "filtercolumn IN ('ec1356','ec1380','ec1396','ec1411','ec1346','ec1361','ec1377','ec1378','ec1393','ec1409','ec1425','ec1298','ec1314','ec1359','ec1390','ec1422','ec1437','ec1408','eti101','eti111','eti114','eti121','eti125','ec1364','ec1388','ec1404','ec1419','ec1354','ec1369','ec1385','ec1386','ec1401','ec1417','ec1433','ec1306','ec1322','ec1367','ec1398','ec1430','ec1350','ec1416','eti104','eti112','eti110','eti122','eti126')"

bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_starved_soil_St 500 $mapper "filtercolumn IN ('ec1349','ec1356','ec1380','ec1396','ec1411','ec1420','ec1427','ec1435','ec1346','ec1361','ec1362','ec1377','ec1378','ec1393','ec1409','ec1425','ec1301','ec1309','ec1317','ec1262','ec1298','ec1314','ec1330','ec1390','ec1422','ec1437','ec1249','ec1257','ec1265','ec1368','ec1392','ec1400','ec1408','ec1424','eti111','eti114','eti121','eti125','ec1058','ec1089','ec1105','ec1113','ec1065','ec1129','ec1073','ec1081','ec1124','ec1084','ec1132','ec1092','ec1140','ec1100','ec1108','ec1116','ec1102','ec1110','ec1118','ec1063','ec1083','ec1091','ec1139','ec1114','ec1130','ec1082','ec1078','ec1086','ec1094','ec1085','ec1109','ec1133','ec1141','ec1149','ec1122','ec1098','ec1138','ec1075') and sample_type IN ('flies')" no
bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_starved_soil_So 500 $mapper "filtercolumn IN ('ec1349','ec1356','ec1380','ec1396','ec1411','ec1420','ec1427','ec1435','ec1346','ec1361','ec1362','ec1377','ec1378','ec1393','ec1409','ec1425','ec1301','ec1309','ec1317','ec1262','ec1298','ec1314','ec1330','ec1390','ec1422','ec1437','ec1249','ec1257','ec1265','ec1368','ec1392','ec1400','ec1408','ec1424','eti111','eti114','eti121','eti125','ec1058','ec1089','ec1105','ec1113','ec1065','ec1129','ec1073','ec1081','ec1124','ec1084','ec1132','ec1092','ec1140','ec1100','ec1108','ec1116','ec1102','ec1110','ec1118','ec1063','ec1083','ec1091','ec1139','ec1114','ec1130','ec1082','ec1078','ec1086','ec1094','ec1085','ec1109','ec1133','ec1141','ec1149','ec1122','ec1098','ec1138','ec1075') and sample_type IN ('soil')" no
bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_starved_soil 500 $mapper "filtercolumn IN ('ec1349','ec1356','ec1380','ec1396','ec1411','ec1420','ec1427','ec1435','ec1346','ec1361','ec1362','ec1377','ec1378','ec1393','ec1409','ec1425','ec1301','ec1309','ec1317','ec1262','ec1298','ec1314','ec1330','ec1390','ec1422','ec1437','ec1249','ec1257','ec1265','ec1368','ec1392','ec1400','ec1408','ec1424','eti111','eti114','eti121','eti125','ec1058','ec1089','ec1105','ec1113','ec1065','ec1129','ec1073','ec1081','ec1124','ec1084','ec1132','ec1092','ec1140','ec1100','ec1108','ec1116','ec1102','ec1110','ec1118','ec1063','ec1083','ec1091','ec1139','ec1114','ec1130','ec1082','ec1078','ec1086','ec1094','ec1085','ec1109','ec1133','ec1141','ec1149','ec1122','ec1098','ec1138','ec1075')" no

## unstarved
bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_unstarved_fruit_U 500 $mapper "filtercolumn IN ('ec1254','ec1325','ec1341','ec1357','ec1364','ec1372','ec1388','ec1403','ec1404','ec1412','ec1419','ec1428','ec1436','ec1353','ec1354','ec1369','ec1386','ec1401','ec1417','ec1433','ec1270','ec1278','ec1306','ec1322','ec1350','ec1367','ec1382','ec1398','ec1430','ec1384','ec1416','eti112','eti122','eti124','eti126','ec1534','ec1510','ec1526','ec1450','eti146','eti147','ec1489','ec1441','ec1497','ec1505','ec1449','ec1521','ec1529','ec1460','ec1508','ec1468','ec1524','eti149','ec1492','ec1500','ec1503','ec1511','eti153','eti154','eti156','ec1530','ec1490','ec1498','ec1514','eti161','ec1525','eti164','eti165','eti166','eti167') and sample_type IN ('flies')" no
bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_unstarved_fruit_F 500 $mapper "filtercolumn IN ('ec1254','ec1325','ec1341','ec1357','ec1364','ec1372','ec1388','ec1403','ec1404','ec1412','ec1419','ec1428','ec1436','ec1353','ec1354','ec1369','ec1386','ec1401','ec1417','ec1433','ec1270','ec1278','ec1306','ec1322','ec1350','ec1367','ec1382','ec1398','ec1430','ec1384','ec1416','eti112','eti122','eti124','eti126','ec1534','ec1510','ec1526','ec1450','eti146','eti147','ec1489','ec1441','ec1497','ec1505','ec1449','ec1521','ec1529','ec1460','ec1508','ec1468','ec1524','eti149','ec1492','ec1500','ec1503','ec1511','eti153','eti154','eti156','ec1530','ec1490','ec1498','ec1514','eti161','ec1525','eti164','eti165','eti166','eti167') and sample_type IN ('fruit')" no
bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_unstarved_fruit 500 $mapper "filtercolumn IN ('ec1254','ec1325','ec1341','ec1357','ec1364','ec1372','ec1388','ec1403','ec1404','ec1412','ec1419','ec1428','ec1436','ec1353','ec1354','ec1369','ec1386','ec1401','ec1417','ec1433','ec1270','ec1278','ec1306','ec1322','ec1350','ec1367','ec1382','ec1398','ec1430','ec1384','ec1416','eti112','eti122','eti124','eti126','ec1534','ec1510','ec1526','ec1450','eti146','eti147','ec1489','ec1441','ec1497','ec1505','ec1449','ec1521','ec1529','ec1460','ec1508','ec1468','ec1524','eti149','ec1492','ec1500','ec1503','ec1511','eti153','eti154','eti156','ec1530','ec1490','ec1498','ec1514','eti161','ec1525','eti164','eti165','eti166','eti167')" no

bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_unstarved_soil_U 500 $mapper "filtercolumn IN ('ec1254','ec1325','ec1341','ec1357','ec1364','ec1372','ec1388','ec1403','ec1404','ec1412','ec1419','ec1428','ec1436','ec1353','ec1354','ec1369','ec1385','ec1386','ec1401','ec1417','ec1433','ec1270','ec1278','ec1306','ec1322','ec1350','ec1382','ec1398','ec1430','ec1384','ec1416','eti110','eti112','eti122','eti124','eti126','ec1150','ec1126','ec1142','ec1066','ec1089','ec1097','ec1105','ec1057','ec1113','ec1121','ec1065','ec1137','ec1145','ec1076','ec1124','ec1084','ec1092','ec1140','ec1100','ec1108','ec1116','ec1119','ec1127','ec1083','ec1091','ec1082','ec1106','ec1114','ec1130','ec1101','ec1141','ec1098','ec1122','ec1138','ec1059','ec1075') and sample_type IN ('flies')" no
bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_unstarved_soil_S 500 $mapper "filtercolumn IN ('ec1254','ec1325','ec1341','ec1357','ec1364','ec1372','ec1388','ec1403','ec1404','ec1412','ec1419','ec1428','ec1436','ec1353','ec1354','ec1369','ec1385','ec1386','ec1401','ec1417','ec1433','ec1270','ec1278','ec1306','ec1322','ec1350','ec1382','ec1398','ec1430','ec1384','ec1416','eti110','eti112','eti122','eti124','eti126','ec1150','ec1126','ec1142','ec1066','ec1089','ec1097','ec1105','ec1057','ec1113','ec1121','ec1065','ec1137','ec1145','ec1076','ec1124','ec1084','ec1092','ec1140','ec1100','ec1108','ec1116','ec1119','ec1127','ec1083','ec1091','ec1082','ec1106','ec1114','ec1130','ec1101','ec1141','ec1098','ec1122','ec1138','ec1059','ec1075') and sample_type IN ('soil')" no
bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_unstarved_soil 500 $mapper "filtercolumn IN ('ec1254','ec1325','ec1341','ec1357','ec1364','ec1372','ec1388','ec1403','ec1404','ec1412','ec1419','ec1428','ec1436','ec1353','ec1354','ec1369','ec1385','ec1386','ec1401','ec1417','ec1433','ec1270','ec1278','ec1306','ec1322','ec1350','ec1382','ec1398','ec1430','ec1384','ec1416','eti110','eti112','eti122','eti124','eti126','ec1150','ec1126','ec1142','ec1066','ec1089','ec1097','ec1105','ec1057','ec1113','ec1121','ec1065','ec1137','ec1145','ec1076','ec1124','ec1084','ec1092','ec1140','ec1100','ec1108','ec1116','ec1119','ec1127','ec1083','ec1091','ec1082','ec1106','ec1114','ec1130','ec1101','ec1141','ec1098','ec1122','ec1138','ec1059','ec1075')" no

#### fruit soil
bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_fruit_soil_F 500 $mapper "filtercolumn IN ('ec1510','ec1518','ec1526','ec1534','ec1441','ec1442','ec1449','ec1450','ec1457','ec1489','ec1497','ec1505','ec1513','ec1521','ec1529','ec1445','ec1453','ec1460','ec1461','ec1468','ec1492','ec1500','ec1508','ec1516','ec1524','ec1502','ec1455','ec1463','ec1471','ec1479','ec1495','ec1503','ec1511','ec1444','ec1452','ec1499','ec1507','ec1515','ec1490','ec1498','ec1514','ec1446','ec1454','ec1462','ec1469','ec1470','ec1493','ec1501','ec1509','ec1517','ec1525','ec1533','eti145','eti146','eti147','eti149','eti151','eti152','eti153','eti154','eti156','eti160','eti161','eti162','eti164','eti165','eti166','eti167','ec1126','ec1134','ec1142','ec1150','ec1057','ec1058','ec1065','ec1066','ec1073','ec1105','ec1113','ec1121','ec1129','ec1137','ec1145','ec1061','ec1069','ec1076','ec1077','ec1084','ec1108','ec1116','ec1124','ec1132','ec1140','ec1118','ec1071','ec1079','ec1087','ec1095','ec1111','ec1119','ec1127','ec1060','ec1068','ec1115','ec1123','ec1131','ec1106','ec1114','ec1130','ec1062','ec1070','ec1078','ec1085','ec1086','ec1109','ec1117','ec1125','ec1133','ec1141','ec1149','ec1081','ec1089','ec1097','ec1100','ec1102','ec1110','ec1083','ec1091','ec1082','ec1093','ec1101','ec1094','ec1122','ec1138','ec1059','ec1075') and sample_type IN ('fruit')" no
bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_fruit_soil_S 500 $mapper "filtercolumn IN ('ec1510','ec1518','ec1526','ec1534','ec1441','ec1442','ec1449','ec1450','ec1457','ec1489','ec1497','ec1505','ec1513','ec1521','ec1529','ec1445','ec1453','ec1460','ec1461','ec1468','ec1492','ec1500','ec1508','ec1516','ec1524','ec1502','ec1455','ec1463','ec1471','ec1479','ec1495','ec1503','ec1511','ec1444','ec1452','ec1499','ec1507','ec1515','ec1490','ec1498','ec1514','ec1446','ec1454','ec1462','ec1469','ec1470','ec1493','ec1501','ec1509','ec1517','ec1525','ec1533','eti145','eti146','eti147','eti149','eti151','eti152','eti153','eti154','eti156','eti160','eti161','eti162','eti164','eti165','eti166','eti167','ec1126','ec1134','ec1142','ec1150','ec1057','ec1058','ec1065','ec1066','ec1073','ec1105','ec1113','ec1121','ec1129','ec1137','ec1145','ec1061','ec1069','ec1076','ec1077','ec1084','ec1108','ec1116','ec1124','ec1132','ec1140','ec1118','ec1071','ec1079','ec1087','ec1095','ec1111','ec1119','ec1127','ec1060','ec1068','ec1115','ec1123','ec1131','ec1106','ec1114','ec1130','ec1062','ec1070','ec1078','ec1085','ec1086','ec1109','ec1117','ec1125','ec1133','ec1141','ec1149','ec1081','ec1089','ec1097','ec1100','ec1102','ec1110','ec1083','ec1091','ec1082','ec1093','ec1101','ec1094','ec1122','ec1138','ec1059','ec1075') and sample_type IN ('soil')" no
bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name mantel_fruit_soil 500 $mapper "filtercolumn IN ('ec1510','ec1518','ec1526','ec1534','ec1441','ec1442','ec1449','ec1450','ec1457','ec1489','ec1497','ec1505','ec1513','ec1521','ec1529','ec1445','ec1453','ec1460','ec1461','ec1468','ec1492','ec1500','ec1508','ec1516','ec1524','ec1502','ec1455','ec1463','ec1471','ec1479','ec1495','ec1503','ec1511','ec1444','ec1452','ec1499','ec1507','ec1515','ec1490','ec1498','ec1514','ec1446','ec1454','ec1462','ec1469','ec1470','ec1493','ec1501','ec1509','ec1517','ec1525','ec1533','eti145','eti146','eti147','eti149','eti151','eti152','eti153','eti154','eti156','eti160','eti161','eti162','eti164','eti165','eti166','eti167','ec1126','ec1134','ec1142','ec1150','ec1057','ec1058','ec1065','ec1066','ec1073','ec1105','ec1113','ec1121','ec1129','ec1137','ec1145','ec1061','ec1069','ec1076','ec1077','ec1084','ec1108','ec1116','ec1124','ec1132','ec1140','ec1118','ec1071','ec1079','ec1087','ec1095','ec1111','ec1119','ec1127','ec1060','ec1068','ec1115','ec1123','ec1131','ec1106','ec1114','ec1130','ec1062','ec1070','ec1078','ec1085','ec1086','ec1109','ec1117','ec1125','ec1133','ec1141','ec1149','ec1081','ec1089','ec1097','ec1100','ec1102','ec1110','ec1083','ec1091','ec1082','ec1093','ec1101','ec1094','ec1122','ec1138','ec1059','ec1075')" no

#### all samples together
bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name 4wayTaxon 500 $mapper "filtercolumn IN ('ec1349','ec1356','ec1380','ec1396','ec1411','ec1420','ec1427','ec1435','ec1346','ec1361','ec1362','ec1378','ec1393','ec1409','ec1425','ec1301','ec1309','ec1317','ec1298','ec1314','ec1359','ec1390','ec1422','ec1437','ec1249','ec1257','ec1265','ec1368','ec1392','ec1400','ec1408','ec1424','eti111','eti121','eti125','ec1442','eti146','ec1489','ec1497','ec1449','ec1513','ec1457','eti145','ec1508','ec1468','ec1516','ec1524','eti149','ec1492','ec1500','eti151','eti152','ec1502','eti153','eti154','ec1530','ec1498','ec1514','eti156','ec1462','ec1470','eti162','ec1469','ec1493','ec1517','ec1525','ec1533','eti164','eti165','eti167','ec1377','eti101','eti114','ec1364','ec1388','ec1404','ec1419','ec1354','ec1369','ec1385','ec1386','ec1401','ec1417','ec1433','ec1306','ec1322','ec1367','ec1398','ec1430','ec1350','ec1416','eti104','eti112','eti110','eti122','eti126','ec1262','ec1330','ec1058','ec1089','ec1105','ec1113','ec1065','ec1129','ec1073','ec1081','ec1124','ec1084','ec1132','ec1092','ec1140','ec1100','ec1108','ec1116','ec1102','ec1110','ec1118','ec1063','ec1083','ec1091','ec1139','ec1114','ec1130','ec1082','ec1078','ec1086','ec1094','ec1085','ec1109','ec1133','ec1141','ec1149','ec1122','ec1098','ec1138','ec1075','ec1254','ec1325','ec1341','ec1357','ec1372','ec1403','ec1412','ec1428','ec1436','ec1353','ec1270','ec1278','ec1382','ec1384','eti124','ec1534','ec1510','ec1526','ec1450','eti147','ec1441','ec1505','ec1521','ec1529','ec1460','ec1503','ec1511','ec1490','eti161','eti166','ec1150','ec1126','ec1142','ec1066','ec1097','ec1057','ec1121','ec1137','ec1145','ec1076','ec1119','ec1127','ec1106','ec1101','ec1059','ec1518','ec1445','ec1453','ec1461','ec1455','ec1463','ec1471','ec1479','ec1495','ec1444','ec1452','ec1499','ec1507','ec1515','ec1446','ec1454','ec1501','ec1509','eti160','ec1134','ec1061','ec1069','ec1077','ec1071','ec1079','ec1087','ec1095','ec1111','ec1060','ec1068','ec1115','ec1123','ec1131','ec1062','ec1070','ec1117','ec1125','ec1093')" no

```


### download files
```{bash}
pwd
core-metrics-results-mantel_starved_fruit_S/unweighted_unifrac_distance_matrix/distance-matrix.tsv

for i in mantel_starved_fruit_S mantel_starved_fruit_F mantel_starved_unstarved_S mantel_starved_unstarved_U mantel_starved_soil_St mantel_starved_soil_So mantel_unstarved_fruit_U mantel_unstarved_fruit_F mantel_unstarved_soil_U mantel_unstarved_soil_S mantel_fruit_soil_F mantel_fruit_soil_S; do
  mkdir core-metrics-results-$i
  for j in unweighted_unifrac_distance_matrix weighted_unifrac_distance_matrix bray_curtis_distance_matrix ; do
  mkdir core-metrics-results-$i""/$j
  wget -O core-metrics-results-$i""/$j/distance-matrix.tsv https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-$i""/$j/distance-matrix.tsv
  done;
done

wget https://raw.githubusercontent.com/johnchaston/Gale2024/main/mapper_1and2_WendyJMC.tsv

for i in mantel_starved_fruit mantel_unstarved_fruit mantel_starved_fruit_S mantel_starved_fruit_F mantel_unstarved_fruit_U mantel_unstarved_fruit_F; do
  mkdir core-metrics-results-$i
  wget -O core-metrics-results-$i""/rarefied_table.txt https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-$i""/rarefied_table.txt
done

for i in 4wayTaxon; do
  mkdir core-metrics-results-$i
  for j in unweighted_unifrac_distance_matrix weighted_unifrac_distance_matrix bray_curtis_distance_matrix ; do
  mkdir core-metrics-results-$i""/$j
  wget -O core-metrics-results-$i""/$j/distance-matrix.tsv https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-$i""/$j/distance-matrix.tsv
  done;
done

wget -O core-metrics-results-4wayTaxon/rarefied_table.txt https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-4wayTaxon/rarefied_table.txt
pwd
```


## Mantel tests
```{r}
mapper_file <- "mapper_1and2_WendyJMC.tsv"
mantel_2groups_all3("mantel_starved_fruit_S","mantel_starved_fruit_F",mapper_file) # nonew
mantel_2groups_all3("mantel_starved_unstarved_S","mantel_starved_unstarved_U",mapper_file) # UW = 0.046
mantel_2groups_all3("mantel_starved_soil_St","mantel_starved_soil_So",mapper_file) # nonew
mantel_2groups_all3("mantel_unstarved_fruit_U","mantel_unstarved_fruit_F",mapper_file) # UW = 0.02 BC = 0.013
mantel_2groups_all3("mantel_unstarved_soil_U","mantel_unstarved_soil_S",mapper_file) # W = 0.03
mantel_2groups_all3("mantel_fruit_soil_F","mantel_fruit_soil_S",mapper_file) # nothing....
```

## ancom
```{r}
taxonomic_level = "X.OTU.ID"
mapper_file <- "mapper_1and2_WendyJMC.tsv"
map_path <- "-ec1and2"
var1 = "plotorder"
var2 = "orchard"
var3 = "wolbachia"
var4 = "opo"
the_id = "X.SampleID"
newcol <- "newcol"
main.var="plotorder"
Group = "plotorder"
adj.formula= "orchard + wolbachia" 
random.formula = NULL 
multcorr="fdr"
sig=0.05
prev.cut=0.9
taxonomyrun <- c("phylum","class","order","family","genus","species","X.OTU.ID")

# for(file_path in c("mantel_starved_fruit","mantel_unstarved_fruit","mantel_starved_unstarved","mantel_starved_soil","mantel_unstarved_soil"))
# for (i in taxonomyrun[1:7]) {
#   try(make_ancom2_plots(file_path=file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level=i,var1=var1,var2=var2,var3=var3,var4 = var4,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group))
# }

##
# ANCOM on resident only flies vs fruit
##

file_path <- "mantel_starved_fruit" 

starved_fruit_tables <- make_ancom2_plots(file_path=file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = F, return_tables = T)

starved_fruit_plot <- ggplot(starved_fruit_tables[[1]][[1]], aes(x=Group, y=mean)) + 
             geom_bar(position=position_dodge(), stat="identity", fill = "yellow") +
             geom_errorbar(aes(ymin=mean+sem, ymax=mean-sem),
                           width=.2,                    # Width of the error bars
                           position=position_dodge(.9)) +
             theme(axis.text=element_text(size=14), 
                   panel.background = element_blank(),
                   axis.line = element_line(), 
                   axis.ticks=element_line(), 
                   axis.title=element_text(size=16),
                   title=element_text(size=13)) +
             labs(y="fractional abundance",x = "Sample type") +
  theme(strip.text.y = element_blank()) + 
  scale_x_discrete(labels = (c("Flies\n(resident)","Fruit")))
starved_fruit_plot + ylim(c(0,0.44))

##
# ANCOM on total microbiota flies vs fruit
##

file_path <- "mantel_unstarved_fruit" # commensalibacter

unstarved_fruit_tables <- make_ancom2_plots(file_path=file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = F, return_tables = T)

unstarved_fruit_plot <- ggplot(unstarved_fruit_tables[[1]][[1]], aes(x=Group, y=mean)) + 
             geom_bar(position=position_dodge(), stat="identity", fill = "yellow") +
             geom_errorbar(aes(ymin=mean+sem, ymax=mean-sem),
                           width=.2,                    # Width of the error bars
                           position=position_dodge(.9)) +
             theme(axis.text=element_text(size=14), 
                   panel.background = element_blank(),
                   axis.line = element_line(), 
                   axis.ticks=element_line(), 
                   axis.title=element_text(size=16),
                   title=element_text(size=13)) +
             labs(y="fractional abundance",x = "Sample type") +
  theme(strip.text.y = element_blank()) + 
  scale_x_discrete(labels = (c("Flies\n(total)","Fruit")))

unstarved_fruit_plot + ylim(c(0,0.44))


tabulate_individual_venns <- function(file_path = file_path, taxonomic_level = taxonomic_level, otu_table = otu_table, threshhold = 1, map_path = map_path) {
  starved_fruit_venn <- get_single_core_features(file_path = file_path, map_path = map_path, taxonomic_level = taxonomic_level)
  starved_fruit_venn$clustered_taxonomy = gsub("[", replacement = "", starved_fruit_venn$clustered_taxonomy, fixed=T)
  starved_fruit_venn$clustered_taxonomy = gsub("]", replacement = "", starved_fruit_venn$clustered_taxonomy, fixed=T)
  
  orchard_site_names <- otu_table %>% filter(X.SampleID %in% colnames(starved_fruit_venn)) %>% dplyr::pull(orchard_subsite) %>% table() %>% data.frame()
  
  group1name = strsplit(file_path, split = "_", fixed = T)[[1]][2]
  group2name = strsplit(file_path, split = "_", fixed = T)[[1]][3]
  
  out_df <- data.frame(orchard_subsite = character(), group1 = character(), group2 = character(), unique1 = numeric, shared = numeric(), unique2 = numeric)
  
  for (i in orchard_site_names$.) {
    sample_names <- otu_table %>% filter(orchard_subsite == i) %>% dplyr::pull(X.SampleID)
    filtered_names <- sample_names[sample_names %in% colnames(starved_fruit_venn)]
    two_way <- data.frame(starved_fruit_venn) %>% dplyr::select(clustered_taxonomy, all_of(filtered_names))
    groupinl1 = (two_way %>% filter(.[[2]] > threshhold) %>% dplyr::pull(clustered_taxonomy))
    groupinl2 = two_way %>% filter(.[[3]] > threshhold) %>% dplyr::pull(clustered_taxonomy)
    try(two_way_venns <- VennDiagram::get.venn.partitions(x = list(group1 = groupinl1, group2 = groupinl2)))
    if(class(try(two_way_venns <- VennDiagram::get.venn.partitions(x = list(group1 = groupinl1, group2 = groupinl2)),F))!="try-error") {
      two_way_venns <- VennDiagram::get.venn.partitions(x = list(group1 = groupinl1, group2 = groupinl2))
      out_df <- rbind(out_df, data.frame(orchard_subsite = as.character(i), 
                                         group1 = as.character(group1name), 
                                         group2 = as.character(group2name), 
                                         unique1 = as.numeric(two_way_venns %>% filter(group1 == T, group2 == F) %>% dplyr::pull(..count..)), 
                                         shared = as.numeric(two_way_venns %>% filter(group1 == T, group2 == T) %>% dplyr::pull(..count..)),
                                         unique2 = as.numeric(two_way_venns %>% filter(group1 == F, group2 == T) %>% dplyr::pull(..count..)))
      )
      
    }
  }
  return(out_df)
}


```

## Venn diagrams
```{r}
map_path = "-ec1and2"
otu_table = reference_table
var1 = "plotorder"
var2 = "orchard"
var3 = "wolbachia"
var4 = "opo"
var5 = NULL
Group = "plotorder"

rm(get_single_core_features)
rm(tabulate_individual_venns)


source(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/Gale_functions.R")

  
## calculate venn diagrams for sample pairs
venns_starved_fruit <- tabulate_individual_venns(file_path = "mantel_starved_fruit", mapper_file = mapper_file, taxonomic_level = "X.OTU.ID",otu_table = (reference_table %>% dplyr::rename(`X.SampleID` = `#SampleID`)), threshhold = 2, map_path = "-ec1and2")
venns_unstarved_fruit <- tabulate_individual_venns(file_path = "mantel_unstarved_fruit", mapper_file = mapper_file,  taxonomic_level = "X.OTU.ID",otu_table = reference_table %>% dplyr::rename(`X.SampleID` = `#SampleID`), threshhold = 2, map_path = "-ec1and2")

## calculate average representation in each Venn partition
venns_starved_fruit2 <- venns_starved_fruit %>%
  mutate(perc1 = unique1/(unique1+shared+unique2),
         percS = shared/(unique1+shared+unique2),
         perc2 = unique2/(unique1+shared+unique2)) 
venns_starved_fruit3 <- venns_starved_fruit2 %>%
  tidyr::gather(venn_partition, perc_ASVs, -orchard_subsite, -group1, -group2, -unique1, -shared, -unique2) %>%
  mutate(venn_partition = factor(venn_partition, levels = c("perc1","percS","perc2"), labels = c("Flies (starved) only","Shared","Fruit only")))%>%
  mutate(comparison = "starved_v_fruit")
ggplot(venns_starved_fruit3, aes(x = venn_partition, y = perc_ASVs)) + geom_jitter(width = .1)
venns_starved_fruit2 %>% 
  dplyr::summarize(perc1M = mean(perc1), sem1 = sd(perc1)/sqrt(dplyr::n()), percSM = mean(percS), semS = sd(percS)/sqrt(dplyr::n()), perc2M = mean(perc2),sem2 = sd(perc2)/sqrt(dplyr::n()))

venns_unstarved_fruit2 <- venns_unstarved_fruit %>%
  mutate(perc1 = unique1/(unique1+shared+unique2),
         percS = shared/(unique1+shared+unique2),
         perc2 = unique2/(unique1+shared+unique2))
venns_unstarved_fruit3 <- venns_unstarved_fruit2 %>%
  tidyr::gather(venn_partition, perc_ASVs, -orchard_subsite, -group1, -group2, -unique1, -shared, -unique2) %>%
  mutate(venn_partition = factor(venn_partition, levels = c("perc1","percS","perc2"), labels = c("Flies (unstarved) only","Shared","Fruit only"))) %>%
  mutate(comparison = "unstarved_v_fruit")
ggplot(venns_unstarved_fruit3, aes(x = venn_partition, y = perc_ASVs)) + geom_jitter(width = .1)
venns_unstarved_fruit2 %>% 
  dplyr::summarize(perc1M = mean(perc1), sem1 = sd(perc1)/sqrt(dplyr::n()), percSM = mean(percS), semS = sd(percS)/sqrt(dplyr::n()), perc2M = mean(perc2),sem2 = sd(perc2)/sqrt(dplyr::n()))

## test if the partitions are different between resident-only and total microbiota
wilcox.test(venns_starved_fruit2 %>% pull(perc1), venns_unstarved_fruit2 %>% pull(perc1))
wilcox.test(venns_starved_fruit2 %>% pull(percS), venns_unstarved_fruit2 %>% pull(percS))
wilcox.test(venns_starved_fruit2 %>% pull(perc2), venns_unstarved_fruit2 %>% pull(perc2))


```

### Venn diagram table
```{r}
all_starved_asvs <- get_single_core_features(file_path = "mantel_starved_fruit_S", map_path = map_path, mapper_file = mapper_file, var1 = var1, var2 = var2, var3 = var3, var4 = var4, var5 = NULL, Group = Group, taxonomic_level = "X.OTU.ID") %>%
  mutate(sum = rowSums(across(where(is.numeric)))) %>%
  dplyr::select(clustered_taxonomy, sum)
all_fruit_asvs <- get_single_core_features(file_path = "mantel_starved_fruit_F", map_path = map_path, taxonomic_level = "X.OTU.ID", mapper_file = mapper_file, var1 = var1, var2 = var2, var3 = var3, var4 = var4, var5 = NULL, Group = Group,) %>%
  mutate(sum = rowSums(across(where(is.numeric)))) %>%
  dplyr::select(clustered_taxonomy, sum)

sf_asvs <- all_starved_asvs %>% 
  full_join(all_fruit_asvs, by = "clustered_taxonomy")  

  ## replace with zeros
for(i in 1:length(sf_asvs$sum.x)) {
  sf_asvs$sum.x[i] <- ifelse(is.na(sf_asvs$sum.x[i]),0,sf_asvs$sum.x[i])
  sf_asvs$sum.y[i] <- ifelse(is.na(sf_asvs$sum.y[i]),0,sf_asvs$sum.y[i])
}

  ## add a perc column
sf_asvs$percfly <- sf_asvs$sum.x/(sf_asvs$sum.x + sf_asvs$sum.y)
sf_asvs$genus = c()

  ## get the genera names
genusname <- read.csv(paste0('taxonomy',map_path,'/taxonomy_forR.csv')) %>% 
    mutate(genusname = paste0(kingdom, phylum, class,order, family, genus)) %>% 
    dplyr::select(X.OTU.ID = Feature.ID,genusname)

i=1
  ## merge genus names to clusters
for(i in 1:length(sf_asvs$clustered_taxonomy)) {
  sf_asvs$X.OTU.ID[i] = tail(str_split(sf_asvs$clustered_taxonomy[i], pattern = "_")[[1]],1)
}

sf_asvs2 <- sf_asvs %>% left_join(genusname, by = "X.OTU.ID")

s2 <- sf_asvs2 %>%
  group_by(genusname) %>%
  dplyr::summarize(inFly = sum(sum.x), inFruit = sum(sum.y)) %>%
  ungroup() %>% 
  mutate(percfly = inFly / (inFly + inFruit)) %>%
  arrange(desc(percfly))

for(i in 1:length(s2$genusname)) {
  s2$chisq_p[i] = chisq.test(matrix(c(s2$inFly[i], s2$inFruit[i], sum(s2$inFly), sum(s2$inFruit)), nrow = 2, byrow = T))$p.value
}

s3 <- s2 %>% filter(inFly > 175 | inFruit > 175) 


all_unstarved_asvs <- get_single_core_features(file_path = "mantel_unstarved_fruit_U", map_path = map_path, taxonomic_level = "X.OTU.ID", mapper_file = mapper_file, var1 = var1, var2 = var2, var3 = var3, var4 = var4, var5 = NULL, Group = Group,) %>%
  mutate(sum = rowSums(across(where(is.numeric)))) %>%
  dplyr::select(clustered_taxonomy, sum)
all_unfruit_asvs <- get_single_core_features(file_path = "mantel_unstarved_fruit_F", map_path = map_path, taxonomic_level = "X.OTU.ID", mapper_file = mapper_file, var1 = var1, var2 = var2, var3 = var3, var4 = var4, var5 = NULL, Group = Group,) %>%
  mutate(sum = rowSums(across(where(is.numeric)))) %>%
  dplyr::select(clustered_taxonomy, sum)

uf_asvs <- all_unstarved_asvs %>% 
  full_join(all_unfruit_asvs, by = "clustered_taxonomy")  

  ## replace with zeros
for(i in 1:length(uf_asvs$sum.x)) {
  uf_asvs$sum.x[i] <- ifelse(is.na(uf_asvs$sum.x[i]),0,uf_asvs$sum.x[i])
  uf_asvs$sum.y[i] <- ifelse(is.na(uf_asvs$sum.y[i]),0,uf_asvs$sum.y[i])
}

  ## add a perc column
uf_asvs$percfly <- uf_asvs$sum.x/(uf_asvs$sum.x + uf_asvs$sum.y)
uf_asvs$genus = c()

  ## merge genus names to clusters
for(i in 1:length(uf_asvs$clustered_taxonomy)) {
  uf_asvs$X.OTU.ID[i] = tail(str_split(uf_asvs$clustered_taxonomy[i], pattern = "_")[[1]],1)
}

uf_asvs2 <- uf_asvs %>% left_join(genusname, by = "X.OTU.ID")

u2 <- uf_asvs2 %>%
  group_by(genusname) %>%
  dplyr::summarize(inFly = sum(sum.x), inFruit = sum(sum.y)) %>%
  ungroup() %>% 
#  filter(inFly > 175 | inFruit > 175) %>%
  mutate(percfly = inFly / (inFly + inFruit)) %>%
  arrange(desc(percfly))

i=5
for(i in 1:length(u2$genusname)) {
  u2$chisq_p[i] = chisq.test(matrix(c(u2$inFly[i], u2$inFruit[i], sum(u2$inFly), sum(u2$inFruit)), nrow = 2, byrow = T))$p.value
}


u2 %>% filter(inFly > 175 | inFruit>175)
s2 %>% filter(inFly > 175 | inFruit > 175) 


```

## neutral analysis
```{r}
taxmat = read.csv('taxonomy-ec1and2/taxonomy_forR.csv')
rownames(taxmat) <- taxmat$Feature.ID
taxmat <- taxmat %>% dplyr::select(-Feature.ID)
taxmat <- as.matrix(taxmat)
colnames(taxmat) <- c("Domain","Phylum","Class","Order","Family","Genus","Species")

#neutral_dataframe <- read.csv('neutral_models_starved-unstarved_soil_fruit.csv', header=T)
neutral_dataframe <- readr::read_csv(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/neutral_models_starved-unstarved_soil_fruit.csv", show_col_types = F) 

migration_coefficient <- neutral_dataframe[,c("m","m.ci","Samples","i","j")] %>%
  mutate(comparison = c("self","other","other","self","self","self","self","other","other","self","self","other","other","self","self","other","self"),comparison2 = c("self","other","other","self","self","self","self","other","other","self","self","other","other","self","self","other","self")) %>%
  tidyr::spread(key = comparison, value = m) %>%
  dplyr::rename(migrationVother = other,migrationVself = self) %>%
  tidyr::spread(key = comparison2, value = m.ci) %>%
  dplyr::rename(otherSEM = other,selfSEM = self) %>%
  mutate(group1 = c("starved","starved","unstarved","unstarved","fruit","fruit","starved","starved","fruit","fruit","unstarved","unstarved","starved","unstarved","unstarved","soil","starved")) %>%
  mutate(group2 = c("starved","unstarved","starved","unstarved","fruit","starved","fruit","starved","fruit","unstarved","fruit","unstarved","starved","starved","unstarved","soil","starved"))
migration_coefficient
# write.csv(migration_coefficient, "migration_edited.csv", quote = F)

#migration_edited <- read.csv("~/Dropbox/sequencing/2022-08-ECrun1and2/migration_edited.csv")
migration_edited <- readr::read_csv(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/migration_edited.csv", show_col_types = F) 

neutral_plot <- ggplot(migration_edited %>% mutate(group1 = factor(group1, levels = c("unstarved","starved","fruit"), labels = c("Flies (total)","Flies (resident)","Fruit"))) %>% mutate(group2 = factor(group2, levels = c("unstarved","starved","fruit"), labels = c("Flies\n(total)","Flies\n(resident)","Fruit"))) %>% mutate(migrationVother = factor(migrationVother, labels = c("Comparison","Reference"))), aes(x = migrationVother , y = AIC, group = group1, color = migrationVother, shape = migrationVother, fill = migrationVother)) + 
        geom_line() +
        geom_point(size = 4) +
        scale_fill_manual(values = c("black","grey")) +
        scale_shape_manual(values = c(20,22)) +
        scale_color_manual(values = c("black","black")) +
        #  facet_wrap(~ group1 + group2, scales="free_x", ncol = 6) +
          facet_nested_wrap(~ group1 + group2, scales="free_x", ncol = 6, nest_line = T)+
        ylim(c(-2000,0)) +
        #scale_x_discrete("") +
        theme_cowplot() +
        theme(legend.position = "none", 
              panel.grid = element_blank(),
        #      axis.text.x = element_blank(),
              axis.line.y = element_line(size = .5),
          strip.background.x = element_blank(), 
        strip.text = element_text(face = "bold"),
        axis.title = element_text(face = "bold"),
        axis.text.x = element_text(angle = 35, vjust = 1, hjust=1)
        ) +
          ggplot2::annotate("segment",x=Inf,xend=-Inf,y=Inf,yend=Inf,color="black",lwd=1) + 
          labs(x = "Reference group compared to ____")
neutral_plot

```

## map
```{r}
east_coast <- subset(map, region %in% c("maine","connecticut","massachusetts","new hampshire","rhode island","vermont","pennsylvania","maryland","district of columbia","delaware","new jersey","new york","virginia"))

east_coast$group2 <- east_coast$group
east_coast$color <- "white"

labs2 <- readr::read_csv(file = "https://raw.githubusercontent.com/johnchaston/microbiota2024/main/DataS13.csv", show_col_types = F) %>% mutate(mergecol = c("ME2", "VA","MD", "NH","ME1","MA","PA","CT"))  %>% arrange(desc(latitude))

labs2$location <- c(6,13,5,8,18,9,10,3)
fruit_map <- ggplot() + geom_polygon(data=east_coast, aes(x=long, y=lat, group=group2), color="black",fill=east_coast$color) + coord_fixed(1.3) + theme_void() +
  geom_point(data = labs2, aes(y=latitude, x=longitude, shape = location), size = 3.5, color = "black") + scale_shape_identity()

fruit_map
# Harvard, MA	42.50867	-71.57071	Westward_F
# Bowdoin, ME	44.0253	-69.94379	RockyRidge_I
# Middlefield, CT	41.49383	-72.71145	Lyman_N
# Media, PA	39.8845	-75.4125	Linvilla_D
# Churchville, MD	39.539	-76.2399	Lahr_G
# Etna, ME	44.82082	-69.11144	Conant_H
# Durham, NH 43.088790 -71.745530
# Charlottesville, VA	38.07474	-78.38155	CarterMountain_H
```

## permanova table
```{r}
file_path = "4wayTaxon"

  ## unweighted
flies_unweighted <- read.table(paste('core-metrics-results-',file_path,'/unweighted_unifrac_distance_matrix/distance-matrix.tsv',sep=""), header=T, sep="\t") %>% mutate(X=as.character(X))
flies_unweighted_dm <- as.dist(flies_unweighted[,2:dim(flies_unweighted)[2]])
flies_unifrac_table <- flies_unweighted %>% dplyr::select(X) %>% left_join(read.table(mapper_file,comment.char = "", header=T, fill=T, sep="\t"), by=c("X"="X.SampleID")) %>% droplevels()

str(flies_unifrac_table)
table(list(flies_unifrac_table$plotorder)) # 
table(list(flies_unifrac_table$orchard_subsite)) # 
table(list(flies_unifrac_table$site)) # 

make_permanova_tables(file_path = file_path, mapper_file = mapper_file, flies_unifrac_table = flies_unifrac_table, pform = "~ plotorder * site + orchard_subsite")

```

## taxon plots
```{r}
mapper_file <- "mapper_1and2_WendyJMC.tsv"
map_path <- "-ec1and2"

###
# Make taxon plots
###
file_path = "4wayTaxon"
		## taxon plot 
map_path = "-ec1and2" # taxonomy folder extension
plotvarnames = c("plotorder","orchard_subsite","site","ospo") 
mapper_file = mapper_file 
taxonomic_level="genus"
rpa_in_chart = .03
read_depth = 500
legend_position = "bottom"
sort_x_axis = "orchard_subsite"
x_cluster = "X.SampleID" #

ptp <- prep_taxon_plot(file_path = file_path, map_path = map_path, plotvarnames = plotvarnames, mapper_file = mapper_file, taxonomic_level=taxonomic_level, rpa_in_chart = rpa_in_chart, read_depth = read_depth, legend_position=legend_position,  sort_x_axis = sort_x_axis,  x_cluster = x_cluster, predefined_taxa = predefined_taxa, predefined_shortname = predefined_shortname)

ptp[[1]]$plotorder <- gsub(pattern = "flies_starved",replacement = "fljes_starved", x = ptp[[1]]$plotorder)
x_cluster <- "ospo"
plot_color <- c("black",predefined_colors[-c(1,2,4,12)])
plot_order <- ptp[[3]]$shortname[c(14,13,12,1,3,2,6,5,4,7,9,11,10,8)]
order_numbers <- c(14,13,12,1,3,2,6,5,4,7,9,11,10,8)
facet_formula <- ". ~ plotorder + site"
sort_x_axis <- "ospo"
legend_position = "bottom"
xorder = "orchard_subsite"

base_plot <- make_taxon_plot_condensed(ptp = ptp, plot_color = plot_color, facet_formula = facet_formula, sort_x_axis = sort_x_axis, x_cluster = x_cluster, plotvarnames = plotvarnames, legend_position = legend_position, read_depth = read_depth, plot_order = plot_order, xorder = xorder, column_bar_width = 0.9, relevel_row = "site", relevelrow_vec = rev(c("Etna, ME","Bowdoin, ME","Durham, NH","Harvard, MA","Middlefield, CT","Media, PA","Churchville, MD","Charlottesville, VA")))
quartet_taxon_plot <- base_plot + 
  facet_grid(cols = vars(plotorder), rows = vars(site), drop = FALSE, as.table = FALSE, space = "free", scales="free", labeller = as_labeller(c(`flies_unstarved`="Flies\n(total)",`fljes_starved`="Flies\n(resident)",`fruit_NA`="Fruit",`soil_NA`="Soil"))) + #, space = "free") + 
#  facet_wrap(site ~ plotorder, strip.position = "left") +
  theme(strip.background.x = element_blank(), axis.text.x = element_text(angle = 90)) + 
#  scale_x_discrete(labels = rev(c("Total","Resident"))) +
  labs(x = "Sampling location") + 
  coord_flip() + 
 theme(legend.position = "none", strip.text.y = element_blank(), 
       axis.text.y = element_blank()
 ) 
quartet_taxon_plot
taxon_legend <- g_legend(quartet_taxon_plot + theme(legend.position = "bottom")+ labs(fill = NULL) + guides(fill=guide_legend(nrow=6,byrow=TRUE)))
  plot(taxon_legend)
```

## distance analysis
```{r}
file_path = "4wayTaxon"
folder_path = "weighted_unifrac"
var1 = "plotorder"
mapper_file = mapper_file
comparisons = c("flies_starved_flies_starved", "flies_starved_flies_unstarved", "flies_starved_fruit_NA", "flies_starved_soil_NA", "flies_unstarved_flies_unstarved", "flies_unstarved_fruit_NA", "flies_unstarved_soil_NA", "fruit_NA_fruit_NA", "fruit_NA_soil_NA", "soil_NA_soil_NA")
title_name_add=""
the_xid="X"
plot_type = "dot"
alpha = 1


flies_unifrac_table <- read.table(paste('core-metrics-results-',file_path,'/',folder_path,'_distance_matrix/distance-matrix.tsv',sep=""), header=T, sep="\t") %>% 
    dplyr::select(X) %>% 
    left_join(read.table(mapper_file,comment.char = "", header=T, sep = "\t"), by=c("X"="X.SampleID")) %>% 
    mutate(the_xid=gsub(x = X,pattern = "-",replacement = ".")) %>%
    dplyr::select("the_xid", var1) 
  
  flies_dm <- read.table(paste('core-metrics-results-',file_path,'/',folder_path,'_distance_matrix/distance-matrix.tsv',sep=""), header=T, sep="\t")
  fs2 <- as.dist(flies_dm[,2:dim(flies_dm)[2]])
  
  fs3 <- as.matrix(fs2); fs3[lower.tri(fs3)] <- NA

    ## melt the table and get out just the right comparisons
  melt_table <- data.frame(melt(fs3)) %>% 
    filter(is.na(value)==F, value!=0) %>% 
    #	  mutate(Var1 = gsub(".","",Var1), Va2 = gsub(".","",Var2)) %>%
    inner_join(flies_unifrac_table, by=c("Var1"="the_xid")) %>% 
    inner_join(flies_unifrac_table, by=c("Var2"="the_xid")) %>%  
    mutate(gs.x = as.character(get(paste0(var1,".x"))), gs.y=as.character(get(paste0(var1,".y")))) %>% 
    mutate(gs=ifelse(gs.x < gs.y, paste(gs.x,gs.y, sep="_"), paste(gs.y, gs.x, sep="_"))) %>% 
    mutate(gs2 = as.factor(as.character(gs))) 
  
  print(table(list(melt_table$gs2)))
  
  melt_table <- melt_table %>% 
    filter(gs2%in%comparisons) %>% droplevels()
  
  ## run statistics
  def <- kruskal.test(value ~ gs2, melt_table);   #print(def$p.value)
  efg <- dunn.test(melt_table$value,melt_table$gs2, method = "bh", table = F, kw = F)
  ghi <- cldList(comparison = efg$comparisons, p.value = efg$P.adjusted, threshold = 0.05);   print(ghi)
  
  ## make the summary statistics
  melt_table_plot <- melt_table %>% 
    group_by(gs2) %>% 
    dplyr::summarize(mean=mean(value), sem=sd(value)/sqrt(length(value))) %>%
    ungroup() %>%
    mutate(plotorder.x = c("flies_starved","flies_starved","flies_starved","flies_starved","flies_unstarved","flies_unstarved","flies_unstarved","fruit_NA","fruit_NA","soil_NA"), plotorder.y = c("flies_starved","flies_unstarved","fruit_NA","soil_NA","flies_unstarved","fruit_NA","soil_NA","fruit_NA","soil_NA","soil_NA")) %>% inner_join(ghi, by = c("gs2" = "Group"))

  melt_table_plot
  ## reshuffle the table
  mt2 <- melt_table %>% filter(plotorder.x == "flies_unstarved", plotorder.y == "flies_unstarved") %>%
    rbind(melt_table %>% filter(plotorder.x == "flies_unstarved", plotorder.y == "flies_starved")) %>% 
    rbind(melt_table %>% filter(plotorder.x == "flies_unstarved", plotorder.y == "fruit_NA")) %>% 
    rbind(melt_table %>% filter(plotorder.x == "flies_unstarved", plotorder.y == "soil_NA")) %>% 
    rbind(melt_table %>% filter(plotorder.x == "flies_starved", plotorder.y == "flies_starved")) %>% 
    rbind(melt_table %>% filter(plotorder.x == "flies_starved", plotorder.y == "fruit_NA")) %>% 
    rbind(melt_table %>% filter(plotorder.x == "flies_starved", plotorder.y == "soil_NA")) %>% 
    rbind(melt_table %>% filter(plotorder.x == "fruit_NA", plotorder.y == "fruit_NA")) %>% 
    rbind(melt_table %>% filter(plotorder.x == "fruit_NA", plotorder.y == "soil_NA")) %>% 
    rbind(melt_table %>% filter(plotorder.x == "soil_NA", plotorder.y == "soil_NA")) %>% 
    rbind(melt_table %>% filter(plotorder.x == "flies_starved", plotorder.y == "flies_unstarved") %>% mutate(plotorder.x = "flies_unstarved", plotorder.y = "flies_starved")) %>%
    rbind(melt_table %>% filter(plotorder.x == "fruit_NA", plotorder.y == "flies_unstarved") %>% mutate(plotorder.x = "flies_unstarved", plotorder.y = "fruit_NA")) %>%
    rbind(melt_table %>% filter(plotorder.x == "soil_NA", plotorder.y == "flies_unstarved") %>% mutate(plotorder.x = "flies_unstarved", plotorder.y = "soil_NA")) %>%
    rbind(melt_table %>% filter(plotorder.x == "fruit_NA", plotorder.y == "flies_starved") %>% mutate(plotorder.x = "flies_starved", plotorder.y = "fruit_NA")) %>%
    rbind(melt_table %>% filter(plotorder.x == "soil_NA", plotorder.y == "flies_starved") %>% mutate(plotorder.x = "flies_starved", plotorder.y = "soil_NA")) %>%
    rbind(melt_table %>% filter(plotorder.x == "soil_NA", plotorder.y == "fruit_NA") %>% mutate(plotorder.x = "fruit_NA", plotorder.y = "soil_NA"))
        
        
        abc <- melt_table %>% mutate(newcol = paste0(plotorder.x, plotorder.y)) %>% dplyr::pull()
        def <- mt2 %>% mutate(newcol = paste0(plotorder.x, plotorder.y)) %>% dplyr::pull()
        data.frame(table(abc)) %>% full_join(data.frame(table(def)), by = c("abc" = "def"))
        
melt_table_plot$plotorder.x = factor(c("Flies (resident)","Flies (total)","Flies (resident)","Flies (resident)","Flies (total)","Flies (total)","Flies (total)","Fruit","Fruit","Soil"), levels = c("Flies (total)","Flies (resident)","Fruit","Soil"))
melt_table_plot$plotorder.y = factor(c("Flies\n(r)","Flies\n(r)","Fruit","Soil","Flies\n(t)","Fruit","Soil","Fruit","Soil","Soil"), levels = c("Flies\n(t)","Flies\n(r)","Fruit","Soil"))

set.seed(43) 
distance_plot_sufs <- ggplot(mt2 %>% mutate(plotorder.x = factor(plotorder.x, levels = c("flies_unstarved","flies_starved","fruit_NA","soil_NA"), labels = c("Flies (total)","Flies (resident)","Fruit","Soil")),plotorder.y = factor(plotorder.y, levels = c("flies_unstarved","flies_starved","fruit_NA","soil_NA"), labels = c("Flies\n(t)","Flies\n(r)","Fruit","Soil"))),
         aes(x= plotorder.y, y = value, group = factor(plotorder.x))) + 
      geom_jitter(width = 0.2, height = 0, alpha = 0.2) +
      facet_nested_wrap(~ plotorder.x + plotorder.y, scales="free_x", ncol = 13, nest_line = T)  +
    geom_crossbar(data=melt_table_plot, aes(y = mean, ymin = mean, ymax = mean), size=0.75, col="blue", width = .6) +
    theme_cowplot() +
theme(legend.position = "none", 
      panel.grid = element_blank(),
#      axis.text.x = element_blank(),
      axis.line.y = element_line(size = .5),
  strip.background.x = element_blank(), 
strip.text = element_text(face = "bold"),
axis.title = element_text(face = "bold"),
axis.text.x = element_blank()
) +
  ggplot2::annotate("segment",x=Inf,xend=-Inf,y=Inf,yend=Inf,color="black",lwd=1) +
  labs(x = NULL)  +
    geom_text(data = melt_table_plot, aes(y = 1.4, label = Letter))
# grid.text("Comparison\ngroup", x = unit(0.1, "npc"), y = unit(.85, "npc"), hjust = 1, )
# grid.text("Reference\ngroup", x = unit(0.1, "npc"), y = unit(.95, "npc"), hjust = 1, )
distance_plot_sufs
```

## bash
```{sh}
# name=ec1and2
# mapper=mapper_1and2_WendyJMC.tsv
# bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name pile_flies 845 $mapper "sample_type = 'flies' AND orchard IN ('Lyman','Conant','Westward') AND species = 'melanogaster' AND noSubSite = 'NA' AND starved = 'starved'" yes
pwd
for i in pile_flies; do
  mkdir core-metrics-results-$i
 wget -O core-metrics-results-$i/rarefied_table.txt https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-$i/rarefied_table.txt
  for j in unweighted_unifrac_distance_matrix weighted_unifrac_distance_matrix bray_curtis_distance_matrix ; do
  mkdir core-metrics-results-$i""/$j
  wget -O core-metrics-results-$i""/$j/distance-matrix.tsv https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-$i""/$j/distance-matrix.tsv
  done;
done
pwd
```

## Table S7 PERMANOVA
```{r}
file_path <- "pile_flies"
mapper_file <- "mapper_1and2_WendyJMC.tsv"

  ## unweighted
flies_unweighted <- read.table(paste('core-metrics-results-',file_path,'/unweighted_unifrac_distance_matrix/distance-matrix.tsv',sep=""), header=T, sep="\t") %>% mutate(X=as.character(X))
flies_unweighted_dm <- as.dist(flies_unweighted[,2:dim(flies_unweighted)[2]])
flies_unifrac_table <- flies_unweighted %>% dplyr::select(X) %>% left_join(read.table(mapper_file,comment.char = "", header=T, fill=T, sep="\t"), by=c("X"="X.SampleID")) %>% droplevels()

sd((table(paste0(flies_unifrac_table$fp_subsite, flies_unifrac_table$orchard))))/sqrt(6)
table(list(flies_unifrac_table$fp_subsite)) 
table(list(flies_unifrac_table$orchard)) 

make_permanova_tables(file_path = file_path, mapper_file = mapper_file, flies_unifrac_table = flies_unifrac_table, pform = "~ fp_subsite * orchard", beta_div_tests = c("unweighted_unifrac","weighted_unifrac","bray_curtis"), seed_to_set = 42)
```

## taxon plots
```{r}
file_path <- "pile_flies"
mapper_file <- "mapper_1and2_WendyJMC.tsv"

### genus
		## taxon plot 
map_path = "-ec1and2" # taxonomy folder extension
plotvarnames = c("fps_location","orchard","fp_subsite") 
mapper_file = mapper_file 
taxonomic_level="genus"
rpa_in_chart = 0.03
read_depth = 845
legend_position = "none"
sort_x_axis = "fps_location"
x_cluster = "X.SampleID" #

ptp <- prep_taxon_plot(file_path = file_path, map_path = map_path, plotvarnames = plotvarnames, mapper_file = mapper_file, taxonomic_level=taxonomic_level, rpa_in_chart = rpa_in_chart, read_depth = read_depth, legend_position=legend_position,  sort_x_axis = sort_x_axis,  x_cluster = x_cluster, predefined_taxa = predefined_taxa, predefined_shortname = predefined_shortname)

ptp[[1]]$fps_location <- factor(ptp[[1]]$fps_location, levels = rev(c("freshLyman","pileLyman","freshWestward","pileWestward","freshConant","pileConant")))
ptp[[1]]$orchard <- factor(ptp[[1]]$orchard, levels = rev(c("Lyman","Westward","Conant")))
ptp[[1]]$fp_subsite <- factor(ptp[[1]]$fp_subsite, levels = rev(c("fresh","pile")))
plot_color <- c("black","lightgreen","darkgreen","purple3","blue4","purple4","purple1","lavender","blue","pink","yellow","orange","magenta","red")
plot_order <- ptp[[3]]$shortname[c(13,12,1,3,2,5,6,4,7,9,11,10,8)]
order_numbers <- c(13,12,1,3,2,5,6,4,7,9,11,10,8)
legend_position = "right"
x_cluster = "fps_location"

compost_plot <- make_taxon_plot_condensed(ptp = ptp, plot_color = plot_color, sort_x_axis = sort_x_axis, x_cluster = x_cluster, plotvarnames = plotvarnames, legend_position = legend_position, plot_order = plot_order, read_depth = read_depth, xorder = "fp_subsite", column_bar_width = 0.9,facet_formula = "orchard ~ .", relevel_row = "orchard", relevelrow_vec = c("Conant","Westward","Lyman"), relevel_col = "fp_subsite",relevel_vec = c("pile","fresh")) + 
  scale_fill_manual(name = "Legend",  values= plot_color) + 
  scale_x_discrete(labels = rev(c("apple","compost"))) + 
  labs(x = "Fruit") +
  coord_flip() + 
  theme(strip.text.y = element_blank(), axis.text.x = element_text(angle = 90), legend.position = "none") + 
  labs(x = "Sample")
```

## ancom
```{r}

rm(melted_table, melted_table_no0, mt2)
var1 = "fp_subsite"
var2 = "orchard"
var3 = "fps_location"
var4 = "fps_location"
the_id = "X.SampleID"
newcol <- "newcol"
the_id <- "X.SampleID"

main.var="fp_subsite"
adj.formula= NULL #"orchard" #"fruit + pile"
random.formula = "~1|orchard"
multcorr="fdr"
sig=0.05
prev.cut=0.9
Group = "fp_subsite"

# taxonomyrun <- c("phylum","class","order","family","genus","species","X.OTU.ID")
# for (i in taxonomyrun[1:7]) {
#   try(make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level=i,var1=var1,var2=var2,var3=var3,var4 = var4,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group))
# }


pile_ancoms <- make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T)

pile_c <- pile_ancoms[[1]] + labs(title = NULL, x = "Site", y = "fractional\nabundance") + theme(axis.title = element_text(face = "bold"), axis.text.x = element_text(angle = 90)) + scale_x_discrete(labels = c("apple","compost")) + geom_bar(position=position_dodge(), stat="identity", fill = "purple1") + geom_errorbar(aes(ymin=mean+sem, ymax=mean-sem), width=.2,position=position_dodge(.9), color = "black") 
pile_d <- pile_ancoms[[2]] + labs(title = NULL, x = "Site", y = "fractional\nabundance") + theme(axis.title = element_text(face = "bold"), axis.text.x = element_text(angle = 90)) + scale_x_discrete(labels = c("apple","compost"))+ geom_bar(position=position_dodge(), stat="identity", fill = "lavender") + geom_errorbar(aes(ymin=mean+sem, ymax=mean-sem), width=.2,position=position_dodge(.9), color = "black") 


```

## overall figure
```{r}
i=2.1
tagsize=24
smallsize = 16

plot_this_legend <- g_legend(ec2021 + theme(legend.position = "bottom", legend.text = element_text(size = 12), legend.title = element_blank()) + guides(fill=guide_legend(ncol=2,byrow=TRUE)))

plot_this_legend <- g_legend(ec2021 + theme(legend.position = "bottom", legend.text = element_text(size = 12), legend.title = element_blank()) + guides(fill=guide_legend(ncol=2,byrow=F)) + scale_fill_manual(name = "", values = c("black",predefined_colors[-c(1,2,4,12)]), labels = c("other", bquote(italic(Pasteurellales)), bquote(italic(Enterobacteriaceae)),bquote(italic(Lactobacillales)), bquote(italic(Lactobacillaceae)), bquote(italic(Enterococcus)), bquote(italic(Leuconostoc)), bquote(italic(Weissella)), bquote(italic(Lactobacillus)), bquote(italic(Acetobacteraceae)), bquote(italic(Commensalibacter)), bquote(italic(Gluconobacter)), bquote(italic(Komagataeibacter)), bquote(italic(Acetobacter)))))

#jpeg(h=6*i, w=8*i, "figure_sufs_aug2024.jpg", units = "in", res=300)
grid.arrange(fruit_map,
             plot_this_legend,
             ggplot(labs2 %>% mutate(xval = c(1)) %>% mutate(yval = (81-c(7,20,32.5,47,59,69.5,78,81))), aes(x = xval, y = yval, shape = location)) +   geom_point(size = 2) + scale_shape_identity() +   theme_cowplot() + theme_void() +  theme(legend.position = "none")+   xlim(c(0.88,1.12)) + ylim(c(-4,84)),
             quartet_taxon_plot + theme(axis.title.y = element_blank()) + theme(strip.text.x = element_text(face = "bold"), axis.title = element_text(face = "bold")),
             neutral_plot,
             distance_plot_sufs + labs(y = "Bray-Curtis distance", x = "Comparison groups"),
             ggplot() + theme_void(), 
             ggplot() + theme_void(), 
             compost_plot + theme(axis.title = element_text(face = "bold")) + labs(x = NULL),
             pile_c + labs(tag = "G") + theme(axis.title.x = element_blank(), axis.text.x = element_blank(), plot.tag = element_text(face = "bold", size = tagsize), axis.title.y = element_text(size = smallsize)),
             pile_d + labs(tag = "H", x= "Site sampled") + theme(plot.tag = element_text(face = "bold", size = tagsize), axis.title.y = element_text(size = smallsize)),
ggplot(labs2 %>% filter(Geography %in% c("E","H","L")) %>% mutate(xval = c(1)) %>% mutate(yval = (81-c(9,(81/7*3),63))), aes(x = xval, y = yval, shape = location)) +   geom_point(size = 2) + scale_shape_identity() +   theme_cowplot() + theme_void() +  theme(legend.position = "none", axis.title.y = element_text(face = "bold", angle = 90, size = smallsize))+   xlim(c(0.88,1.12)) + ylim(c(-4,84)) + labs(y = "Sample"),
             widths = c(0.6,0.6,.1,1,.1,.75,.6),
             heights = c(3,1,.5,.5,1.4,1.6),
             layout_matrix=rbind(		
                c(1,1,3,4,7,5,5),
                c(1,1,3,4,7,6,6),
                c(2,2,3,4,7,6,6),
                c(2,2,3,4,7,6,6),
                c(2,2,3,4,12,9,10),
                c(8,8,3,4,12,9,11)
             )
             ) 
#grid.text(c("Comparison","Reference","Comparison","Reference"), x = unit(c(0.705,.705,.695,.695), "npc"), y = unit(c(.912,.965,.415,.47), "npc"), hjust = 1, vjust = 0.5,gp=gpar(fontsize=12)) 
#grid.text(c(expression(bold("A")),expression(bold("B")),expression(bold("C")),expression(bold("D"))), y = unit(c(.965,.965,.965,.47), "npc"), x = unit(c(0.03,0.35,0.64,0.64), "npc"), hjust = 1, vjust = 0.5,gp=gpar(fontsize=24)) 
grid.text(c("Comparison","Reference","Comparison","Reference"), x = unit(c(0.685,0.685,.675,.675), "npc"), y = unit(c(.97,.935,.6,.57), "npc"), hjust = 1, vjust = 0.5,gp=gpar(fontsize=12)) 
grid.text(c(expression(bold("A")),expression(bold("B")),expression(bold("C")),expression(bold("D")),expression(bold("E")),expression(bold("F"))), x = unit(c(0.014, 0.35, 0.014, 0.62,0.62,0.62), "npc"), y = unit(c(.965,.965,0.25,.965,0.6,.35), "npc"), hjust = 1, vjust = 0.5,gp=gpar(fontsize=24)) 
# dev.off()


```

# Figure 3
## survey
### bash
```{sh, eval=F}
# name=eastcoast
# mapper=mapper_all.tsv
# bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name lymanfruits 120 $mapper "noSubSite IN ('asianpear','bartlettpear','crispin','daybreakFuji','gloriapeach','goldendelicious','macintoshL','macoun','messinaPeachD','messinaPeachL')"
pwd
for i in lymanfruits; do
  mkdir core-metrics-results-$i
 wget -O core-metrics-results-$i/rarefied_table.txt https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-$i/rarefied_table.txt
  for j in unweighted_unifrac_distance_matrix weighted_unifrac_distance_matrix bray_curtis_distance_matrix ; do
  mkdir core-metrics-results-$i""/$j
  wget -O core-metrics-results-$i""/$j/distance-matrix.tsv https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-$i""/$j/distance-matrix.tsv
  done;
done

mkdir taxonomy-eastcoast
wget -O taxonomy-eastcoast/taxonomy_forR.csv https://raw.githubusercontent.com/johnchaston/Gale2024/main/taxonomy-eastcoast/taxonomy_forR.csv
pwd
```

### Table S5 (Permanova)
```{r}
file_path <- "lymanfruits"
mapper_file <- "~/Dropbox/sequencing/2022-08-ECrun1and2/mapper_1and2_WendyJMC.tsv"

  ## unweighted
flies_unweighted <- read.table(paste('core-metrics-results-',file_path,'/unweighted_unifrac_distance_matrix/distance-matrix.tsv',sep=""), header=T, sep="\t") %>% mutate(X=as.character(X))
flies_unweighted_dm <- as.dist(flies_unweighted[,2:dim(flies_unweighted)[2]])
flies_unifrac_table <- flies_unweighted %>% dplyr::select(X) %>% left_join(read.table(mapper_file,comment.char = "", header=T, fill=T, sep="\t"), by=c("X"="X.SampleID")) %>% droplevels()

str(flies_unifrac_table)
mean(table(list(flies_unifrac_table$fn)))
sd(table(list(flies_unifrac_table$fn)))/sqrt(10)
table(list(flies_unifrac_table$fn)) #
table(list(flies_unifrac_table$fruittype)) #
table(list(flies_unifrac_table$noSubSite)) #
table(list(flies_unifrac_table$wolbachia)) #

make_permanova_tables(file_path = file_path, mapper_file = mapper_file, flies_unifrac_table = flies_unifrac_table, pform = "~ fruittype/noSubSite * wolbachia")
```

### taxon plot
```{r}
file_path <- "lymanfruits"
mapper_file <- "mapper_1and2_WendyJMC.tsv"
map_path = "-eastcoast" # taxonomy folder extension
plotvarnames = c("site", "fruittype","noSubSite", "fn") 
mapper_file = mapper_file 
taxonomic_level="genus"
rpa_in_chart = .02
read_depth = 120
legend_position = "bottom"
sort_x_axis = "noSubSite"
x_cluster = "X.SampleID" #

ptp <- prep_taxon_plot(file_path = file_path, map_path = map_path, plotvarnames = plotvarnames, mapper_file = mapper_file, taxonomic_level=taxonomic_level, rpa_in_chart = rpa_in_chart, read_depth = read_depth, legend_position=legend_position,  sort_x_axis = sort_x_axis,  x_cluster = x_cluster, predefined_taxa = predefined_taxa, predefined_shortname = predefined_shortname)

table(ptp[[1]]$noSubSite)
ptp[[1]]$noSubSite <- factor(ptp[[1]]$noSubSite, levels =(c((c("crispin","daybreakFuji","goldendelicious","macintoshL","macoun")),(c("gloriapeach","messinaPeachD","messinaPeachL")),(c("asianpear","bartlettpear")))))
ptp[[1]]$noSubSite <- factor(ptp[[1]]$noSubSite, labels = (c((c("Crispin","Fuji","Golden delicious","Macintosh","Macoun")),(c("Gloria","Messina site 2","Messina site 1")),(c("Asian","Bartlett")))))
ptp[[1]]$fruittype <- factor(ptp[[1]]$fruittype, labels = c("Apple","Peach","Pear"))
#ptp[[1]]$wolbachia <- factor(ptp[[1]]$wolbachia, labels = c("*Wolbachia* negative","*Wolbachia* positive"))

x_cluster <- "fn"
plot_color <- c("black","black","lightgreen","darkgreen","blue4","purple4","purple1","lavender","blue","pink","yellow","orange","magenta","red")
plot_order <- ptp[[3]]$shortname[c(1,13,12,3,2,5,6,4,7,9,11,10,8)]
order_numbers <- c(1,13,12,3,2,5,6,4,7,9,11,10,8)
facet_formula <- ". ~ noSubSite * fruittype"
sort_x_axis <- "noSubSite"
legend_position = "bottom"

base_plot <- make_taxon_plot_condensed(ptp = ptp, plot_color = plot_color, facet_formula = facet_formula, sort_x_axis = sort_x_axis, x_cluster = x_cluster, plotvarnames = plotvarnames, legend_position = "none", read_depth = read_depth, plot_order = plot_order,xorder = "noSubSite", relevel_row = "fruittype",relevelrow_vec =  (c("Apple","Peach","Pear")), column_bar_width = 0.9, relevel_col = "noSubSite",relevel_vec = (c((c("Crispin","Fuji","Golden delicious","Macintosh","Macoun")),(c("Gloria","Messina site 2","Messina site 1")),(c("Asian","Bartlett")))), predefined_colors = predefined_colors, predefined_shortname = predefined_shortname, predefined_taxa = predefined_taxa, default_colors = T, default_order = F, order_numbers = order_numbers)
wild_fruits <- base_plot + 
  facet_grid(cols = vars(fruittype), drop = FALSE, as.table = FALSE, scales = "free", space = "free", labeller = labeller(wolbachia = c("hi","bue"))) + #bquote(italic(Wolbachia)~"+"), bquote(italic(Wolbachia)~"-")))) + 
  theme(strip.text = element_text(size = 24, face = "bold"), axis.text.x = element_text(angle = 45), axis.text.y = element_text(size=20), strip.text.y = element_text(angle = 270), strip.background = element_blank()) + 
  labs(x = "Sample") + 
  theme(strip.text = ggtext::element_markdown())
wild_fruits
wf_legend <- g_legend(wild_fruits + theme(legend.position = "bottom", legend.title = element_blank()) + guides(fill=guide_legend(ncol=3,byrow=F))+ theme(legend.text = element_text(size = 18)) + scale_fill_manual(name = "J", values = c("black","black","lightgreen","darkgreen","blue4","purple4","purple1","lavender","blue","pink","yellow","orange","magenta","red"), labels = c("other",bquote(italic(Dysgonomonas)),bquote(italic(Pasteurellales)),bquote(italic(Enterobacteriaceae)),bquote(italic(Lactobacillaceae)),bquote(italic(Enterococcus)),bquote(italic(Leuconostoc)),bquote(italic(Weissella)),bquote(italic(Lactobacillus)),bquote(italic(Acetobacteraceae)),bquote(italic(Commensalibacter)),bquote(italic(Gluconobacter)),bquote(italic(Komagataeibacter)),bquote(italic(Acetobacter)))))
plot(wf_legend)
```

### ancom - no differences with fruittype or subsite
```{r}
var1 = "fruittype"
var2 = "noSubSite"
var3 = "fruittype"
var4 = "fnw"
the_id = "X.SampleID"
newcol <- "newcol"
the_id <- "X.SampleID"

main.var="fruittype"
adj.formula= "noSubSite"# + noSubSite" #"orchard" #"fruit + pile"
random.formula = NULL #"~1|fruittype"
multcorr="fdr"
sig=0.05
prev.cut=0.9
Group = "fnw"

# taxonomyrun <- c("phylum","class","order","family","genus","species","X.OTU.ID")
# for (i in taxonomyrun[1:7]) {
#   try(make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level=i,var1=var1,var2=var2,var3=var3,var4 = var4,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group))
# }


var1 = "noSubSite"
var2 = "fruittype"
var3 = "wolbachia"
var4 = "fnw"
the_id = "X.SampleID"
newcol <- "newcol"
the_id <- "X.SampleID"

main.var="noSubSite"
adj.formula= "fruittype + wolbachia"# + noSubSite" #"orchard" #"fruit + pile"
random.formula = NULL #"~1|fruittype"
multcorr="none"
sig=0.05
prev.cut=0.9
Group = "fnw"

# taxonomyrun <- c("phylum","class","order","family","genus","species","X.OTU.ID")
# for (i in taxonomyrun[1:7]) {
#   try(make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level=i,var1=var1,var2=var2,var3=var3,var4 = var4,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group))
# }
```

## gnotobiotic fruits
### bash
```{bash}
# name=fv2
#mapper=mapper.tsv
# bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name gflies 200 mapper.tsv "Treatment = 'G' AND NOT Fruit IN ('AS', 'CL')"
pwd
for i in gflies; do
  mkdir core-metrics-results-$i
 wget -O core-metrics-results-$i/rarefied_table.txt https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-$i/rarefied_table.txt
  for j in unweighted_unifrac_distance_matrix weighted_unifrac_distance_matrix bray_curtis_distance_matrix ; do
  mkdir core-metrics-results-$i""/$j
  wget -O core-metrics-results-$i""/$j/distance-matrix.tsv https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-$i""/$j/distance-matrix.tsv
  done;
done

mkdir taxonomy-fv2
wget -O taxonomy-fv2/taxonomy_forR.csv https://raw.githubusercontent.com/johnchaston/Gale2024/main/taxonomy-fv2/taxonomy_forR.csv

wget https://raw.githubusercontent.com/johnchaston/Gale2024/main/mapper.tsv
pwd
```

### Table S6 (permanova)
```{r}
file_path <- "gflies"
mapper_file <- "mapper.tsv"

flies_unweighted <- read.table(paste('core-metrics-results-',file_path,'/unweighted_unifrac_distance_matrix/distance-matrix.tsv',sep=""), header=T, sep="\t") %>% mutate(X=as.character(X))
flies_unweighted_dm <- as.dist(flies_unweighted[,2:dim(flies_unweighted)[2]])
flies_unifrac_table <- flies_unweighted %>% dplyr::select(X) %>% left_join(read.table(mapper_file,comment.char = "", header=T, fill=T, sep="\t"), by=c("X"="X.SampleID")) %>% droplevels()

mean(table(list(flies_unifrac_table$FruitGuess)))
sd(table(list(flies_unifrac_table$FruitGuess)))/sqrt(10)

table(list(flies_unifrac_table$FruitGuess))

make_permanova_tables(file_path = file_path, mapper_file = mapper_file, flies_unifrac_table = flies_unifrac_table, pform = "~ FruitGuess", beta_div_tests = c("unweighted_unifrac","weighted_unifrac","bray_curtis"), seed_to_set = 42)
```

### Taxon plot
```{r}

# apoc dls4or6= a47c56b57bbcbc2940d672582f8a7249 - 99.59 % # k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Acetobacter_ s___a47c56b57bbcbc2940d672582f8a7249
# atrc dls5 = d5e253f9fd3d9fff16e51560b9ba1add - 100% # k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Acetobacter_ s___d5e253f9fd3d9fff16e51560b9ba1add
# lbrc dls1 = 9489650ff7365fbfdac60d72947bea37 - 98.77% but 100% match if i blast against draft genomes... # k__Bacteria_ p__Firmicutes_ c__Bacilli_ o__Lactobacillales_ f__Lactobacillaceae_ g__Lactobacillus_ s__brevis_9489650ff7365fbfdac60d72947bea37
# lplc dls2 = f9a8f7eb646316ec1a73e776ac3b8a5d - 100% k__Bacteria_ p__Firmicutes_ c__Bacilli_ o__Lactobacillales_ f__Lactobacillaceae___f9a8f7eb646316ec1a73e776ac3b8a5d
# bsub dls13 = c0836deeef20cb98d4b071c870c85fd8 - 99.59% [12] "k__Bacteria_ p__Firmicutes_ c__Bacilli_ o__Bacillales_ f__Bacillaceae_ g__Bacillus_ s___c0836deeef20cb98d4b071c870c85fd8"
# ecok dls15 = 4c56aeed6ccd9cab509a5b0b0ddcb21c - 100% [61] "k__Bacteria_ p__Proteobacteria_ c__Gammaproteobacteria_ o__Enterobacteriales_ f__Enterobacteriaceae___4c56aeed6ccd9cab509a5b0b0ddcb21c"                                 
# kmed dls31 = 56d303503337df69f247bd5d9bc8bb08 - 97.94% but 100% when blasting against the nbrc3288 genome [37] "k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Gluconacetobacter_ s__intermedius_56d303503337df69f247bd5d9bc8bb08"
# g cerinus dls58 = use 865584b1e6664cd406d2268c653d8775 [38] "k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Gluconobacter_ s___865584b1e6664cd406d2268c653d8775"
# Weissella paramesenteroides strain DmW_115  = 22e313d08b660317ae55cd4c2ff17d80 [68] "k__Bacteria_ p__Firmicutes_ c__Bacilli_ o__Lactobacillales_ f__Leuconostocaceae_ g__Weissella__22e313d08b660317ae55cd4c2ff17d80"
# Providencia rettgeri strain JGM226 (big10 recipe says JGM232) 2ac177233bbe7813e1f8b09b26919ccc  [44] "k__Bacteria_ p__Proteobacteria_ c__Gammaproteobacteria_ o__Enterobacteriales_ f__Enterobacteriaceae_ g__Providencia_ s___2ac177233bbe7813e1f8b09b26919ccc"              
# pantoea dispersa strain JGM106 =  f66f60890f29131e9620441a7986321b [65] "k__Bacteria_ p__Proteobacteria_ c__Gammaproteobacteria_ o__Enterobacteriales_ f__Enterobacteriaceae___f66f60890f29131e9620441a7986321b"
# tatumella JGM49 = ee6363a0ab008fd7bc6ada9e9c65f348 [42] "k__Bacteria_ p__Proteobacteria_ c__Gammaproteobacteria_ o__Enterobacteriales_ f__Enterobacteriaceae_ g__Enterobacter__ee6363a0ab008fd7bc6ada9e9c65f348"                 

predefined_taxa_fv <- c("k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Acetobacter_ s___a47c56b57bbcbc2940d672582f8a7249",
                     "k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Acetobacter_ s___d5e253f9fd3d9fff16e51560b9ba1add",
                     "k__Bacteria_ p__Firmicutes_ c__Bacilli_ o__Lactobacillales_ f__Lactobacillaceae_ g__Lactobacillus_ s__brevis_9489650ff7365fbfdac60d72947bea37",
                     "k__Bacteria_ p__Firmicutes_ c__Bacilli_ o__Lactobacillales_ f__Lactobacillaceae___f9a8f7eb646316ec1a73e776ac3b8a5d",
                     "k__Bacteria_ p__Firmicutes_ c__Bacilli_ o__Bacillales_ f__Bacillaceae_ g__Bacillus_ s___c0836deeef20cb98d4b071c870c85fd8",
                     "k__Bacteria_ p__Proteobacteria_ c__Gammaproteobacteria_ o__Enterobacteriales_ f__Enterobacteriaceae___4c56aeed6ccd9cab509a5b0b0ddcb21c",
                     "k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Gluconacetobacter_ s__intermedius_56d303503337df69f247bd5d9bc8bb08",
                     "k__Bacteria_ p__Proteobacteria_ c__Alphaproteobacteria_ o__Rhodospirillales_ f__Acetobacteraceae_ g__Gluconobacter_ s___865584b1e6664cd406d2268c653d8775",
                     "k__Bacteria_ p__Firmicutes_ c__Bacilli_ o__Lactobacillales_ f__Leuconostocaceae_ g__Weissella__22e313d08b660317ae55cd4c2ff17d80",
                     "k__Bacteria_ p__Proteobacteria_ c__Gammaproteobacteria_ o__Enterobacteriales_ f__Enterobacteriaceae_ g__Providencia_ s___2ac177233bbe7813e1f8b09b26919ccc",
                     "k__Bacteria_ p__Proteobacteria_ c__Gammaproteobacteria_ o__Enterobacteriales_ f__Enterobacteriaceae___f66f60890f29131e9620441a7986321b",
                     "k__Bacteria_ p__Proteobacteria_ c__Gammaproteobacteria_ o__Enterobacteriales_ f__Enterobacteriaceae_ g__Enterobacter__ee6363a0ab008fd7bc6ada9e9c65f348"
                     )

predefined_shortname_fv <- c("a47c56b57bbcbc2940d672582f8a7249",
                             "d5e253f9fd3d9fff16e51560b9ba1add",
                             "brevis9489650ff7365fbfdac60d72947bea37",
                             "Lactobacillaceae",
                             "c0836deeef20cb98d4b071c870c85fd8",
                             "Enterobacteriaceae1",
                             "intermedius56d303503337df69f247bd5d9bc8bb08",
                             "865584b1e6664cd406d2268c653d8775",
                             "Weissella",
                             "2ac177233bbe7813e1f8b09b26919ccc",
                             "Enterobacteriaceae2",
                             "Enterobacter")

map_path = "-fv2" # taxonomy folder extension
plotvarnames = c("FruitGuess") 
mapper_file = mapper_file 
taxonomic_level="ASV"
rpa_in_chart = .03
read_depth = 200
legend_position = "none"
sort_x_axis = "FruitGuess"
x_cluster = "X.SampleID" #

ptp <- prep_taxon_plot(file_path = file_path, map_path = map_path, plotvarnames = plotvarnames, mapper_file = mapper_file, taxonomic_level=taxonomic_level, rpa_in_chart = rpa_in_chart, read_depth = read_depth, legend_position=legend_position,  sort_x_axis = sort_x_axis,  x_cluster = x_cluster, predefined_taxa = predefined_taxa_fv, predefined_shortname = predefined_shortname_fv)

plot_color <- c("black","grey","darkgreen","green4","green3","green","lavender","blue4","blue","orange","magenta","red4","red")
plot_order <- ptp[[3]]$shortname[c(1,12,9,11,10,4,3,2,8,7,5,6)]
order_numbers <- c(1,12,9,11,10,4,3,2,8,7,5,6)
facet_formula <- ". ~ FruitGuess"
legend_position = "right"

# 		## taxon plot 
 x_cluster = "FruitGuess"
# ptp <- prep_taxon_plot(file_path = file_path, map_path = map_path, plotvarnames = plotvarnames, mapper_file = mapper_file, taxonomic_level=taxonomic_level, rpa_in_chart = rpa_in_chart, read_depth = read_depth, legend_position=legend_position,  sort_x_axis = sort_x_axis,  x_cluster = x_cluster)
# #ptp[[1]]$FruitGuess <- factor(ptp[[1]]$FruitGuess, levels = c("CA","BA","CO","SP","FJ","RP","PR","MA","LI","OR"))
# #ptp[[1]]$FruitGuess <- factor(ptp[[1]]$FruitGuess, labels = c("Carrot","Banana","Corn","Spaghetti\nsquash","Apple","Peach","Pear","Mango","Lime","Orange"))

ptp[[1]]$FruitGuess <- factor(ptp[[1]]$FruitGuess, levels = (c("FJ","BA","CA","CO","LI","MA","OR","RP","PR","SP")))
ptp[[1]]$FruitGuess <- factor(ptp[[1]]$FruitGuess, labels = (c("Apple","Banana","Carrot","Corn","Lime","Mango","Orange","Peach","Pear","Spaghetti\nsquash")))

fv_taxonplot_baseplot <- make_taxon_plot_condensed(ptp = ptp, plot_color = plot_color, facet_formula = facet_formula, sort_x_axis = sort_x_axis, x_cluster = x_cluster, plotvarnames = plotvarnames, legend_position = "none", plot_order = plot_order, read_depth = read_depth,xorder = "FruitGuess", column_bar_width = 0.9) + scale_fill_manual(name = "Legend",  values= plot_color) + labs(x = "Fruit")

fv_taxonplot <- fv_taxonplot_baseplot + 
  theme(axis.text.x = element_text(size=12)) +
  facet_grid(NULL)
fv_taxonplot  

fv_legend <- g_legend(fv_taxonplot_baseplot + theme(legend.position = "bottom") + guides(fill=guide_legend(ncol=3,byrow=F)) + scale_fill_manual(name = "   ", values = plot_color, labels = c("other",bquote(italic("B. subtilis")),bquote(italic("Pantoea")~"sp. JGM106"),bquote(italic("Pantoea")~"sp. JGM49"),bquote(italic("P. rettgeri")),bquote(italic("E. coli")),bquote(italic("W. paramesenteroides")),bquote(italic("L. plantarum")),bquote(italic("L. brevis")),bquote(italic("G. cerinus")),bquote(italic("K. medellinensis")),bquote(italic("A. pomorum")),bquote(italic("A. tropicalis")))) + theme(legend.text = element_text(size = 18)))
plot(fv_legend)
```

### ANCOM
```{r}
rm(melted_table, melted_table_no0, mt2)
var1 = "FruitGuess"
var2 = "Exp"
var3 = "FruitGuess"
var4 = "FruitGuess"
the_id = "X.SampleID"
newcol <- "newcol"
the_id <- "X.SampleID"

main.var="FruitGuess"
adj.formula= "FruitGuess + Exp"# + noSubSite" #"orchard" #"fruit + pile"
random.formula = NULL #"~1|fruittype"
multcorr="fdr"
sig=0.05
prev.cut=0.9
Group = "FruitGuess"

# taxonomyrun <- c("phylum","class","order","family","genus","species","X.OTU.ID")
# for (i in taxonomyrun[1:7]) {
#   try(make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level=i,var1=var1,var2=var2,var3=var3,var4 = var4,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group))
# }

for (i in "X.OTU.ID") {
  try(make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level=i,var1=var1,var2=var2,var3=var3,var4 = var4,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group))
}
```

## overall figure
```{r}
i=1.7
tagsize=40
smallsize = 22

plot1 <- wild_fruits + theme(axis.text = element_text(size = smallsize), axis.title = element_text(size=smallsize*1.2, face = "bold"))
plot3 <- fv_taxonplot +theme(axis.text.x = element_text(size=smallsize, angle=35), axis.text.y = element_text(size=smallsize), axis.title = element_text(size=smallsize*1.2, face = "bold"))

plot(wf_legend)
i=1.6
#jpeg(h=4*i*1.4, w=11*i, "dietsandLAB_july2024.jpg", units = "in", res=300)
grid.arrange(plot1 + labs(tag = "A") + theme(plot.tag = element_text(size = tagsize, face = "bold")) + theme(axis.text.x = element_text(angle = 35, hjust=1, vjust=1)),
            ggplot() + theme_void(),
             plot3 + labs(tag = "B") + theme(plot.tag = element_text(size = tagsize, face = "bold")) + theme(axis.text.x = element_text(angle = 35, hjust=1, vjust=1)),
            wf_legend,
            ggplot() + theme_void(),
            fv_legend,
             widths = c(2,.11,2),
             heights = c(1.4,.4),
             layout_matrix=rbind(		
                c(1,2,3),
                c(4,5,6)
#                c(2,2,3,3,4,4)
             )
)

#dev.off()
```

# Figure 4
## bash
```{sh}
# name=joe
# mapper=joe_mapper.tsv
# bash ~/Dropbox/sequencing/16sscripts/standard_qiime3.sh $name""_noW $name allbutS 500 $mapper "fruit IN ('a','p') AND species NOT IN ('s','suz','nothing')" yes
pwd
for i in allbutS; do
  mkdir core-metrics-results-$i
 wget -O core-metrics-results-$i/rarefied_table.txt https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-$i/rarefied_table.txt
  for j in unweighted_unifrac_distance_matrix weighted_unifrac_distance_matrix bray_curtis_distance_matrix ; do
  mkdir core-metrics-results-$i""/$j
  wget -O core-metrics-results-$i""/$j/distance-matrix.tsv https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-$i""/$j/distance-matrix.tsv
  done;
done

mkdir taxonomy-joe
wget -O taxonomy-joe/taxonomy_forR.csv https://raw.githubusercontent.com/johnchaston/Gale2024/main/taxonomy-joe/taxonomy_forR.csv

wget https://raw.githubusercontent.com/johnchaston/Gale2024/main/joe_mapper.tsv
mkdir core-metrics-results-A_allbutS
wget -O core-metrics-results-A_allbutS/rarefied_table.txt https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-A_allbutS/rarefied_table.txt
mkdir core-metrics-results-P_allbutS
 wget -O core-metrics-results-P_allbutS/rarefied_table.txt https://raw.githubusercontent.com/johnchaston/Gale2024/main/core-metrics-results-P_allbutS/rarefied_table.txt
pwd
```

## Table S8 (PERMANOVA)
```{r}
file_path <- "allbutS"
mapper_file <- "joe_mapper.tsv"

  ## unweighted
flies_unweighted <- read.table(paste('core-metrics-results-',file_path,'/unweighted_unifrac_distance_matrix/distance-matrix.tsv',sep=""), header=T, sep="\t") %>% mutate(X=as.character(X))
flies_unweighted_dm <- as.dist(flies_unweighted[,2:dim(flies_unweighted)[2]])
flies_unifrac_table <- flies_unweighted %>% dplyr::select(X) %>% left_join(read.table(mapper_file,comment.char = "", header=T, fill=T, sep="\t"), by=c("X"="X.SampleID")) %>% droplevels()

str(flies_unifrac_table)
table(list(flies_unifrac_table$fruit)) 
table(list(flies_unifrac_table$pile)) 
table(list(flies_unifrac_table$lineartime)) 
table(list(flies_unifrac_table$time)) 
table(list(flies_unifrac_table$sex)) 
table(list(flies_unifrac_table$wolbachia)) 
table(list(flies_unifrac_table$species)) 

make_permanova_tables(file_path = file_path, mapper_file = mapper_file, flies_unifrac_table = flies_unifrac_table, pform = "~ fruit/pile * lineartime + sex + wolbachia")
make_permanova_tables(file_path = file_path, mapper_file = mapper_file, flies_unifrac_table = flies_unifrac_table, pform = "~ fruit/pile * time + sex + wolbachia")

```

## taxon plots
```{r}
smallsize = 16

file_path = "allbutS"
mapper_file <- "joe_mapper.tsv"
map_path = "-joe" # taxonomy folder extension
plotvarnames = c("pile","time") 
mapper_file = mapper_file 
taxonomic_level="genus"
rpa_in_chart = .03
read_depth = 500
legend_position = "bottom"
sort_x_axis = "pile"
x_cluster = "X.SampleID" #

ptp <- prep_taxon_plot(file_path = file_path, map_path = map_path, plotvarnames = plotvarnames, mapper_file = mapper_file, taxonomic_level=taxonomic_level, rpa_in_chart = rpa_in_chart, read_depth = read_depth, legend_position=legend_position,  sort_x_axis = sort_x_axis,  x_cluster = x_cluster, predefined_taxa = predefined_taxa, predefined_shortname = predefined_shortname)

x_cluster <- "time"
plot_color <- c("black","lightgreen","darkgreen","purple4","purple1","lavender","blue","pink","yellow","orange","magenta","red")
plot_order <- ptp[[3]]$shortname[c(11,10,1,3,4,2,9,6,8,7,5)]
order_numbers <- c(11,10,1,3,4,2,9,6,8,7,5)
sort_x_axis <- "pile"
legend_position = "nonew"

time_plot <- make_taxon_plot_condensed_numericX(ptp = ptp, plot_color = plot_color, facet_formula = facet_formula, sort_x_axis = sort_x_axis, x_cluster = x_cluster, plotvarnames = plotvarnames, legend_position = legend_position, read_depth = read_depth, plot_order = plot_order)
time_plot + xlim(c(0,70))


###
# Make taxon plots - Calendar date
###

plotvarnames = c("pile","lineartime") 
x_cluster = "X.SampleID" #

ptp <- prep_taxon_plot(file_path = file_path, map_path = map_path, plotvarnames = plotvarnames, mapper_file = mapper_file, taxonomic_level=taxonomic_level, rpa_in_chart = rpa_in_chart, read_depth = read_depth, legend_position=legend_position,  sort_x_axis = sort_x_axis,  x_cluster = x_cluster, predefined_taxa = predefined_taxa, predefined_shortname = predefined_shortname)

x_cluster <- "lineartime"
plot_color <- c("black","lightgreen","darkgreen","purple4","purple1","lavender","blue","pink","yellow","orange","magenta","red")
plot_order <- ptp[[3]]$shortname[c(11,10,1,3,4,2,9,6,8,7,5)]
order_numbers <- c(11,10,1,3,4,2,9,6,8,7,5)
sort_x_axis <- "pile"
legend_position = "none"

lineartime_plot <- make_taxon_plot_condensed_numericX(ptp = ptp, plot_color = plot_color, facet_formula = facet_formula, sort_x_axis = sort_x_axis, x_cluster = x_cluster, plotvarnames = plotvarnames, legend_position = legend_position, read_depth = read_depth, plot_order = plot_order)
lineartime_plot + xlim(c(0,70))

```

## timeline plots
```{r}
timeline <- readr::read_tsv(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/joe_mapper.tsv", show_col_types = F, ) %>% dplyr::rename('X.SampleID' = `#SampleID`)

file_path <- "allbutS"
mapper_file <- "joe_mapper.tsv"

flies_unweighted <- read.table(paste('core-metrics-results-',file_path,'/unweighted_unifrac_distance_matrix/distance-matrix.tsv',sep=""), header=T, sep="\t") %>% mutate(X=as.character(X))
flies_unweighted_dm <- as.dist(flies_unweighted[,2:dim(flies_unweighted)[2]])
flies_unifrac_table <- flies_unweighted %>% dplyr::select(X) %>% left_join(read.table(mapper_file,comment.char = "", header=T, fill=T, sep="\t"), by=c("X"="X.SampleID")) %>% droplevels()

timeline2 <- flies_unifrac_table %>% 
  dplyr::select(wp = fruit, activity = pile,start_date, ) %>% 
  unique() %>% 
  filter(wp %in% c("a","p")) %>%
  mutate(wp = as.factor(wp)) %>%
  mutate(wp = recode_factor(wp, `a` = "Apples",`p` = "Peaches")) %>% 
  mutate(activity = as.factor(activity)) %>%
  mutate(activity = recode_factor(activity, `a1` = "Pile 1",`a2` = "Pile 2",`a3` = "Pile 3",`p1` = "Pile 1",`p2` = "Pile 2",`p3` = "Pile 3",`a4` = "Pile 4")) %>% 
  dplyr::mutate(end_date = "11-1-22")
timeline2$start_date <- as.Date(c("2022-08-27","2022-09-10","2022-08-27","2022-09-10", "2022-09-22","2022-09-22","2022-10-06"))-15
timeline2$end_date <- as.Date("2022-11-01")-15
timeline2 <- timeline2[c(3,4,5,7,1,2,6),]
t2 <- timeline2 %>%
  mutate(wa = paste0(wp, " ", activity))
t2$wa <- factor(t2$wa, levels = rev(c("Apples Pile 1","Apples Pile 2","Apples Pile 3","Apples Pile 4","Peaches Pile 1","Peaches Pile 2","Peaches Pile 3"))) 
t2$wa <- factor(t2$wa, labels = rev(c("A Pile 1","A Pile 2","A Pile 3","A Pile 4","P Pile 1","P Pile 2","P Pile 3")))
t2$wa <- factor(t2$wa, labels = rev(c("    A 1    ","A 2    ","A 3    ","A 4    ","P 1    ","P 2    ","P 3    ")))

timeline3 <- timeline %>% 
  dplyr::select(wp = fruit, activity = pile,date) %>% 
  unique() %>% 
  filter(wp %in% c("a","p")) %>%
  mutate(wp = as.factor(wp)) %>%
  mutate(wp = recode_factor(wp, `a` = "Apples",`p` = "Peaches")) %>% 
  mutate(activity = as.factor(activity)) %>%
  mutate(activity = recode_factor(activity, `a1` = "Pile 1",`a2` = "Pile 2",`a3` = "Pile 3",`p1` = "Pile 1",`p2` = "Pile 2",`p3` = "Pile 3",`a4` = "Pile 4")) %>% 
 # dplyr::select(-wp) %>% 
  mutate(spot_date = as.Date(date,format="%m/%d/%y")-15) %>%
  mutate(spot_type = "S")
t3 <- timeline3 %>%
  mutate(wa = paste0(wp," ",activity))
t3$wa <- factor(t3$wa, levels = rev(c("Apples Pile 1","Apples Pile 2","Apples Pile 3","Apples Pile 4","Peaches Pile 1","Peaches Pile 2","Peaches Pile 3"))) 
t3$wa <- factor(t3$wa, labels = rev(c("A Pile 1","A Pile 2","A Pile 3","A Pile 4","P Pile 1","P Pile 2","P Pile 3")))
t3$wa <- factor(t3$wa, labels = rev(c("    A 1    ","A 2    ","A 3    ","A 4    ","P 1    ","P 2    ","P 3    ")))

joepiles_gantt <- ggplot(t2, aes(x=wa, ymin = start_date-19216, ymax = end_date-19216, color = c("grey","grey","grey","grey","black","black","black"))) + geom_linerange(linewidth = 3) + coord_flip() + 
  theme_cowplot() +
  ggplot2::annotate(geom = "label",y = as.numeric(t3$spot_date-19216), x = t3$wa, label ="S", size = 3.5, alpha = 1, fontface = "bold") +
  scale_color_identity() +
  labs(y = "Time (d)", x = NULL) + 
  theme(axis.line = element_blank(), axis.ticks = element_blank()) 
joepiles_gantt  

t4 <- t2 %>% 
  inner_join(t3 %>% dplyr::select(-activity, -wp), by = "wa") %>%
  mutate(pile_date = spot_date - as.Date(start_date, format="%m/%d/%y") - 1)
t42 <- t4 %>% dplyr::select(wa, end_date,start_date) %>% unique()

joepiles_gantt2 <- ggplot(t42, aes(x=wa, ymin = 0, ymax = end_date-start_date, color = c("grey","grey","grey","grey","black","black","black"))) + geom_linerange(linewidth = 3) + coord_flip() + 
  theme_cowplot() +
  ggplot2::annotate(geom = "label",y = as.numeric(t4$pile_date), x = t4$wa, label ="S", size = 3.5, alpha = 1, fontface = "bold") +
  scale_color_identity() +
  labs(y = "Time (d)", x = NULL) + 
  theme(axis.line = element_blank(), axis.ticks = element_blank()) 

joepiles_gantt2
```

## figure SX - mantel tests
```{r}
file_path <- "allbutS"
mapper_file <- "joe_mapper.tsv"

time_WUmantel <- mantel_distance_plot(calc_mantel_microbiome(file_path,"weighted_unifrac",mapper_file,mapper_cols = c("time")),y_title = "time", point_size = 0.25)
lineartime_WUmantel <- mantel_distance_plot(calc_mantel_microbiome(file_path,"weighted_unifrac",mapper_file,mapper_cols = c("lineartime")),y_title = "lineartime", point_size = 0.25)

time_UUmantel <- mantel_distance_plot(calc_mantel_microbiome(file_path,"unweighted_unifrac",mapper_file,mapper_cols = c("time")),y_title = "time", point_size = 0.25)
lineartime_UUmantel <- mantel_distance_plot(calc_mantel_microbiome(file_path,"unweighted_unifrac",mapper_file,mapper_cols = c("lineartime")),y_title = "lineartime", point_size = 0.25)

time_BCmantel <- mantel_distance_plot(calc_mantel_microbiome(file_path,"bray_curtis",mapper_file,mapper_cols = c("time")),y_title = "time", point_size = 0.25)
lineartime_BCmantel <- mantel_distance_plot(calc_mantel_microbiome(file_path,"bray_curtis",mapper_file,mapper_cols = c("lineartime")),y_title = "lineartime", point_size = 0.25)

alphaval = .5
i=1.7
tagsize=28
smallsize = 16
time_plot
annotationsize = 4.75

#jpeg(h=8*1.1*i, w=8*i*(2/3), "mantel_tests.jpg", units = "in", res=300)
grid.arrange(time_UUmantel + labs(tag = "A", y = "Microbiota distance\n(weighted Unifrac)", x = "Difference in time (d)") + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL, axis.text = element_text(size = smallsize), axis.title = element_text(size = smallsize, face = "bold")), 
             lineartime_UUmantel + labs(tag = "B",y = "Microbiota distance\n(weighted Unifrac)", x = "Difference in time (d)") + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL, axis.text = element_text(size = smallsize), axis.title = element_text(size = smallsize, face = "bold")),
             time_WUmantel + labs(tag = "C", y = "Microbiota distance\n(weighted Unifrac)", x = "Difference in time (d)") + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL, axis.text = element_text(size = smallsize), axis.title = element_text(size = smallsize, face = "bold")), 
             lineartime_WUmantel + labs(tag = "D",y = "Microbiota distance\n(weighted Unifrac)", x = "Difference in time (d)") + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL, axis.text = element_text(size = smallsize), axis.title = element_text(size = smallsize, face = "bold")),
             time_BCmantel + labs(tag = "E", y = "Microbiota distance\n(weighted Unifrac)", x = "Difference in time (d)") + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL, axis.text = element_text(size = smallsize), axis.title = element_text(size = smallsize, face = "bold")), 
             lineartime_BCmantel + labs(tag = "F",y = "Microbiota distance\n(weighted Unifrac)", x = "Difference in time (d)") + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL, axis.text = element_text(size = smallsize), axis.title = element_text(size = smallsize, face = "bold")),
  
             widths = c(1,1),
#             heights = c(1,.5,.3,1.2,.8,.75,.8),
             heights = c(1,1,1),
             layout_matrix=rbind(		
               c(1,2),
               c(3,4),
               c(5,6)
                )
)
#dev.off()

```

## ancom - used these
```{r}
file_path = "allbutS"
rm(melted_table, melted_table_no0, mt2)
mapper_file <- "joe_mapper.tsv"
map_path = "-joe" # taxonomy folder extension
var1 = "lineartime"
var2 = "pile"
var3 = "fruit"
var4 = "sex"
var5 = "wolbachia"
the_id = "X.SampleID"
newcol <- "newcol"
the_id <- "X.SampleID"
main.var="lineartime"
adj.formula= "sex + wolbachia + fruit"
random.formula = "~1|pile"
multcorr="fdr"
sig=0.05
prev.cut=0.9
Group = "lineartime"

# taxonomyrun <- c("phylum","class","order","family","genus","species","X.OTU.ID")
# for (i in taxonomyrun[1:7]) {
#   try(make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level=i,var1=var1,var2=var2,var3=var3,var4 = var4,var5 =var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group))
# }

lineartime_ancom <- make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_tables = T)

str(lineartime_ancom)
lineartime_ancom_glucono <- ggplot(lineartime_ancom[[1]][[2]], aes(x=Group, y = mean)) + 
  geom_col(fill = "orange") + 
  theme_cowplot() +
  geom_errorbar(aes(ymin=mean+sem, ymax=mean-sem), width=.2,position=position_dodge(.9), color = "black") +
  theme(strip.text = element_text(size = 24, face = "bold"), axis.text.x = element_text(angle = 90), axis.text.y = element_text(size=20), strip.text.y = element_text(angle = 270), strip.background = element_blank()) + 
  labs(x = "Sample") + 
  theme(strip.text = ggtext::element_markdown())
lineartime_cortest_glucono <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Proteobacteria_c__Alphaproteobacteria_o__Rhodospirillales_f__Acetobacteraceae_g__Gluconacetobacter",return_table = T)
cor.test(lineartime_cortest_glucono$Group, lineartime_cortest_glucono$mean, method = "spearman")

lineartime_ancom_entero <- ggplot(lineartime_ancom[[1]][[3]], aes(x=Group, y = mean)) + 
  geom_col(fill = "darkgreen") + 
  theme_cowplot() +
  geom_errorbar(aes(ymin=mean+sem, ymax=mean-sem), width=.2,position=position_dodge(.9), color = "black") +
  theme(strip.text = element_text(size = 24, face = "bold"), axis.text.x = element_text(angle = 90), axis.text.y = element_text(size=20), strip.text.y = element_text(angle = 270), strip.background = element_blank()) + 
  labs(x = "Sample") + 
  theme(strip.text = ggtext::element_markdown())
lineartime_cortest_entero <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Proteobacteria_c__Gammaproteobacteria_o__Enterobacteriales_f__Enterobacteriaceae_NA",return_table = T)
cor.test(lineartime_cortest_entero$Group, lineartime_cortest_entero$mean, method = "spearman")

lineartime_ancom_leuconostoc <- ggplot(lineartime_ancom[[1]][[1]], aes(x=Group, y = mean)) +
  geom_col(fill = "purple1") +
  theme_cowplot() +
  geom_errorbar(aes(ymin=mean+sem, ymax=mean-sem), width=.2,position=position_dodge(.9), color = "black") +
  theme(strip.text = element_text(size = 24, face = "bold"), axis.text.x = element_text(angle = 90), axis.text.y = element_text(size=20), strip.text.y = element_text(angle = 270), strip.background = element_blank()) +
  labs(x = "Sample") +
  theme(strip.text = ggtext::element_markdown())
lineartime_cortest_leuconostoc <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Firmicutes_c__Bacilli_o__Lactobacillales_f__Leuconostocaceae_g__Leuconostoc",return_table = T)
cor.test(lineartime_cortest_leuconostoc$Group, lineartime_cortest_leuconostoc$mean, method = "spearman")

lineartime_ancom_lactobacillus <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Firmicutes_c__Bacilli_o__Lactobacillales_f__Lactobacillaceae_g__Lactobacillus")
lineartime_cortest_lactobacillus <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Firmicutes_c__Bacilli_o__Lactobacillales_f__Lactobacillaceae_g__Lactobacillus",return_table = T)
cor.test(lineartime_cortest_lactobacillus$Group, lineartime_cortest_lactobacillus$mean, method = "spearman")


var1 = "time"
main.var = "time"
Group = "time"
taxonomyrun <- c("phylum","class","order","family","genus","species","X.OTU.ID")
# for (i in taxonomyrun[1:7]) {
#   try(make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level=i,var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group))
# }

time_ancom <- make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_tables = T)
str(time_ancom)

time_ancom_glucono <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Proteobacteria_c__Alphaproteobacteria_o__Rhodospirillales_f__Acetobacteraceae_g__Gluconacetobacter")
time_cortest_glucono <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Proteobacteria_c__Alphaproteobacteria_o__Rhodospirillales_f__Acetobacteraceae_g__Gluconacetobacter",return_table = T)
cor.test(time_cortest_glucono$Group, time_cortest_glucono$mean, method = "spearman")

time_ancom_entero <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Proteobacteria_c__Gammaproteobacteria_o__Enterobacteriales_f__Enterobacteriaceae_NA")
time_cortest_entero <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Proteobacteria_c__Gammaproteobacteria_o__Enterobacteriales_f__Enterobacteriaceae_NA", return_table = T)
cor.test(time_cortest_entero$Group, time_cortest_entero$mean, method = "spearman")

time_ancom_leuconostoc <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Firmicutes_c__Bacilli_o__Lactobacillales_f__Leuconostocaceae_g__Leuconostoc")
time_cortest_leuconostoc <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Firmicutes_c__Bacilli_o__Lactobacillales_f__Leuconostocaceae_g__Leuconostoc", return_table = T)
cor.test(time_cortest_leuconostoc$Group, time_cortest_leuconostoc$mean, method = "spearman")
 
time_ancom_lactobacillus <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Firmicutes_c__Bacilli_o__Lactobacillales_f__Lactobacillaceae_g__Lactobacillus")
time_cortest_lactobacillus <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Firmicutes_c__Bacilli_o__Lactobacillales_f__Lactobacillaceae_g__Lactobacillus",return_table = T)
cor.test(time_cortest_lactobacillus$Group, time_cortest_lactobacillus$mean, method = "spearman")

###
# apples
###

file_path = "A_allbutS"
rm(melted_table, melted_table_no0, mt2)
adj.formula= "sex + wolbachia"# + noSubSite" #"orchard" #"fruit + pile"
var1 = "lineartime"
main.var="lineartime"
Group = "lineartime"

# taxonomyrun <- c("phylum","class","order","family","genus","species","X.OTU.ID")
# for (i in taxonomyrun[1:7]) {
#   try(make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level=i,var1=var1,var2=var2,var3=var3,var4 = var4,var5 =var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group))
# }

var1 = "time"
main.var = "time"
Group = "time"
# taxonomyrun <- c("phylum","class","order","family","genus","species","X.OTU.ID")
# for (i in taxonomyrun[1:7]) {
#   try(make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level=i,var1=var1,var2=var2,var3=var3,var4 = var4,var5=var5,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group))
# }

###
# peaches
###

file_path = "P_allbutS"

var1 = "lineartime"
main.var="lineartime"
Group = "lineartime"

# taxonomyrun <- c("phylum","class","order","family","genus","species","X.OTU.ID")
# for (i in taxonomyrun[1:7]) {
#   try(make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level=i,var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group))
# }

peachlineartime_ancom <- make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5=var5,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_tables = T)

peachlineartime_ancom_glucono <- ggplot(peachlineartime_ancom[[1]][[2]], aes(x=Group, y = mean)) + 
  geom_col(fill = "orange") + 
  theme_cowplot() +
  geom_errorbar(aes(ymin=mean+sem, ymax=mean-sem), width=.2,position=position_dodge(.9), color = "black") +
  theme(strip.text = element_text(size = 24, face = "bold"), axis.text.x = element_text(angle = 90), axis.text.y = element_text(size=20), strip.text.y = element_text(angle = 270), strip.background = element_blank()) + 
  labs(x = "Sample") + 
  theme(strip.text = ggtext::element_markdown())
peachlineartime_ancom_glucono
peachlineartime_cortest_glucono <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Proteobacteria_c__Alphaproteobacteria_o__Rhodospirillales_f__Acetobacteraceae_g__Gluconacetobacter",return_table = T)
cor.test(peachlineartime_cortest_glucono$Group, peachlineartime_cortest_glucono$mean, method = "spearman")

peachlineartime_ancom_lacto <- ggplot(peachlineartime_ancom[[1]][[1]], aes(x=Group, y = mean)) + 
  geom_col(fill = "blue") + 
  theme_cowplot() +
  geom_errorbar(aes(ymin=mean+sem, ymax=mean-sem), width=.2,position=position_dodge(.9), color = "black") +
  theme(strip.text = element_text(size = 24, face = "bold"), axis.text.x = element_text(angle = 90), axis.text.y = element_text(size=20), strip.text.y = element_text(angle = 270), strip.background = element_blank()) + 
  labs(x = "Sample") + 
  theme(strip.text = ggtext::element_markdown())
peachlineartime_ancom_lacto
peachlineartime_cortest_lacto <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Firmicutes_c__Bacilli_o__Lactobacillales_f__Lactobacillaceae_g__Lactobacillus",return_table = T)
cor.test(peachlineartime_cortest_lacto$Group, peachlineartime_cortest_lacto$mean, method = "spearman")




var1 = "time"
main.var = "time"
Group = "time"
# taxonomyrun <- c("phylum","class","order","family","genus","species","X.OTU.ID")
# for (i in taxonomyrun[1:7]) {
#   try(make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level=i,var1=var1,var2=var2,var3=var3,var4 = var4,var5=var5,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group))
# }

peachtime_ancom <- make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_tables = T)

peachTime_ancom_lacto <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Firmicutes_c__Bacilli_o__Lactobacillales_f__Lactobacillaceae_g__Lactobacillus")
peachtime_ancom_lacto <- ggplot(peachtime_ancom[[1]][[1]], aes(x=Group, y = mean)) + 
  geom_col(fill = "blue") + 
  theme_cowplot() +
  geom_errorbar(aes(ymin=mean+sem, ymax=mean-sem), width=.2,position=position_dodge(.9), color = "black") +
  theme(strip.text = element_text(size = 24, face = "bold"), axis.text.x = element_text(angle = 90), axis.text.y = element_text(size=20), strip.text.y = element_text(angle = 270), strip.background = element_blank()) + 
  labs(x = "Sample") + 
  theme(strip.text = ggtext::element_markdown())
peachtime_ancom_lacto
peachtime_cortest_lacto <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Firmicutes_c__Bacilli_o__Lactobacillales_f__Lactobacillaceae_g__Lactobacillus",return_table = T)
cor.test(peachtime_cortest_lacto$Group, peachtime_cortest_lacto$mean, method = "spearman")



peachTime_ancom_glucono <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Proteobacteria_c__Alphaproteobacteria_o__Rhodospirillales_f__Acetobacteraceae_g__Gluconacetobacter")
peachtime_ancom_glucono <- ggplot(peachtime_ancom[[1]][[2]], aes(x=Group, y = mean)) + 
  geom_col(fill = "orange") + 
  theme_cowplot() +
  geom_errorbar(aes(ymin=mean+sem, ymax=mean-sem), width=.2,position=position_dodge(.9), color = "black") +
  theme(strip.text = element_text(size = 24, face = "bold"), axis.text.x = element_text(angle = 90), axis.text.y = element_text(size=20), strip.text.y = element_text(angle = 270), strip.background = element_blank()) + 
  labs(x = "Sample") + 
  theme(strip.text = ggtext::element_markdown())
peachtime_ancom_glucono
peachtime_cortest_glucono <- make_abundance_plot(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="genus",var1=var1,var2=var2,var3=var3,var4 = var4,var5 = var5, newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_plots = T, return_tables = F, specified_group = "k__Bacteria_p__Proteobacteria_c__Alphaproteobacteria_o__Rhodospirillales_f__Acetobacteraceae_g__Gluconacetobacter",return_table = T)
cor.test(peachtime_cortest_glucono$Group, peachtime_cortest_glucono$mean, method = "spearman")
```

### overall figure
```{r}
file_path <- "allbutS"
mapper_file <- "joe_mapper.tsv"

pc_label <- cor.test(lineartime_cortest_glucono$Group, lineartime_cortest_glucono$mean, method = "spearman", exact = F)
pd_label <- cor.test(lineartime_cortest_entero$Group, lineartime_cortest_entero$mean, method = "spearman", exact = F)
pe_label <- cor.test(lineartime_cortest_leuconostoc$Group, lineartime_cortest_leuconostoc$mean, method = "spearman", exact = F)
pf_label <- cor.test(peachlineartime_cortest_glucono$Group, peachlineartime_cortest_glucono$mean, method = "spearman", exact = F)
pg_label <- cor.test(peachlineartime_cortest_lacto$Group, peachlineartime_cortest_lacto$mean, method = "spearman", exact = F)
pj_label <- cor.test(time_cortest_glucono$Group, time_cortest_glucono$mean, method = "spearman", exact = F)
pk_label <- cor.test(time_cortest_entero$Group, time_cortest_entero$mean, method = "spearman", exact = F)
pl_label <- cor.test(time_cortest_leuconostoc$Group, time_cortest_leuconostoc$mean, method = "spearman", exact = F)
pm_label <- cor.test(peachtime_cortest_glucono$Group, peachtime_cortest_glucono$mean, method = "spearman", exact = F)
pn_label <- cor.test(peachtime_cortest_lacto$Group, peachtime_cortest_lacto$mean, method = "spearman", exact = F)

alphaval = 0
i=1.7
tagsize=28
smallsize = 16
time_plot
annotationsize = 4.75


plot_legend_lineartime <- g_legend(ec2021 + theme(legend.position = "bottom", legend.text = element_text(size = 12), legend.title = element_blank()) + guides(fill=guide_legend(nrow=2,byrow=T)) + scale_fill_manual(name = "", values = c("black",predefined_colors[-c(1,2,4,12)]), labels = c("other", bquote(italic(Pasteurellales)), bquote(italic(Enterobacteriaceae)),bquote(italic(Lactobacillales)), bquote(italic(Lactobacillaceae)), bquote(italic(Enterococcus)), bquote(italic(Leuconostoc)), bquote(italic(Weissella)), bquote(italic(Lactobacillus)), bquote(italic(Acetobacteraceae)), bquote(italic(Commensalibacter)), bquote(italic(Gluconobacter)), bquote(italic(Komagataeibacter)), bquote(italic(Acetobacter)))))



#jpeg(h=8*1.1*i, w=8*i, "figure_joepiles_july2024.jpg", units = "in", res=300)
grid.arrange(time_plot + labs(x = "Time from pile start (d)", tag = "H", y = "Fractional abundance") + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL, axis.text.x = element_text(size = smallsize), axis.text.y = element_text(size = smallsize), axis.title = element_text(size = smallsize, face = "bold")) + theme(axis.text.x = element_blank(), axis.title.x = element_blank()),
             lineartime_plot + labs(x = "Time from experimental start (d)", tag = "A", y = "Fractional abundance") + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL, axis.text.y = element_text(size = smallsize), axis.title = element_text(size = smallsize, face = "bold"), axis.text.x = element_blank(), axis.title.x = element_blank()) + xlim(c(0,67)),
             time_WUmantel + labs(tag = "I", y = "Microbiota distance\n(weighted Unifrac)", x = "Difference in time (d)") + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL, axis.text = element_text(size = smallsize), axis.title = element_text(size = smallsize, face = "bold")), 
             lineartime_WUmantel + labs(tag = "B",y = "Microbiota distance\n(weighted Unifrac)", x = "Difference in time (d)") + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL, axis.text = element_text(size = smallsize), axis.title = element_text(size = smallsize, face = "bold")),
             lineartime_ancom_glucono  + theme(axis.title.x = element_text(face = "bold", size = smallsize), axis.text.y = element_text(size = smallsize), axis.text.x = element_text(size = smallsize, angle=0)) + xlim(c(0,70)) + stat_smooth(method = "lm", alpha = alphaval, color = "black")+ ggplot2::annotate(geom = "text", x = 5, y = 0.4, hjust=0, label = paste0("S = ",unname(round(pc_label$estimate,2)),"\np =  ", unname(round(pc_label$p.value,4))), size = annotationsize) + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL, axis.title.y = element_text(size = smallsize*.9, face = "bold"))+ labs(tag = "C", title = NULL, y = "Fractional abundance", x = "Time (d)") + geom_col(fill = "orange"),
             lineartime_ancom_entero + labs(tag = "D", title = NULL, x = "Time (d)") + theme(axis.title = element_blank(), axis.title.x = element_text(face = "bold", size = smallsize), axis.text.y = element_text(size = smallsize), axis.text.x = element_text(size = smallsize, angle=0)) + xlim(c(0,70)) + stat_smooth(method = "lm", alpha = alphaval, color = "black")+ ggplot2::annotate(geom = "text", x = 60, y = 0.32, hjust=1,  label = paste0("S = ",unname(round(pd_label$estimate,2)),"\np =  ", unname(round(pd_label$p.value,4))), size = annotationsize) + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL)+ geom_col(fill = "darkgreen"),
             peachlineartime_ancom_glucono + labs(tag = "F", title = NULL, x = "Time (d)") + theme(axis.title = element_blank(), axis.title.x = element_text(face = "bold", size = smallsize), axis.text.y = element_text(size = smallsize), axis.text.x = element_text(size = smallsize, angle=0)) + xlim(c(0,70)) + stat_smooth(method = "lm", alpha = alphaval, color = "black")+ ggplot2::annotate(geom = "text", x = 5, y = 0.5, hjust=0,  label = paste0("S = ",unname(round(pf_label$estimate,2)),"\np = ", unname(round(pf_label$p.value,4))), size = annotationsize) + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL),
             time_ancom_glucono + labs(tag = "J", title = NULL, y = "Fractional abundance", x = "Time (d)") + theme(axis.text.x = element_text(size = smallsize), axis.title.x = element_text(size = smallsize, face = "bold"), axis.title.y = element_text(size = smallsize*0.9, face = "bold"))+ geom_col(fill = "orange") + stat_smooth(method = "lm", alpha = alphaval, color = "black")+ ggplot2::annotate(geom = "text",x = 5, y = 0.5, hjust=0, label = paste0("S = ",unname(round(pj_label$estimate,2)),"\np = ", unname(round(pj_label$p.value,2))), size = annotationsize) + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL),
             time_ancom_entero + labs(tag = "K", title = NULL, x= "Time (d)") + theme(axis.title.x = element_text(face = "bold", size = smallsize), axis.title.y = element_blank(), axis.text = element_text(size = smallsize))+ geom_col(fill = "darkgreen") + stat_smooth(method = "lm", alpha = alphaval, color = "black")+ ggplot2::annotate(geom = "text", x = 60, y = 0.4, hjust=1,  label = paste0("S = ",unname(round(pk_label$estimate,2)),"\np = ", unname(round(pk_label$p.value,2))), size = annotationsize) + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL),
             peachtime_ancom_glucono + labs(tag = "M", title = NULL, x = "Time (d)") + theme(axis.title.y = element_blank(), axis.title.x = element_text(size = smallsize, face = "bold"), axis.text.y = element_text(size = smallsize), axis.text.x = element_text(size = smallsize, angle=0)) + stat_smooth(method = "lm", alpha = alphaval, color = "black") + ggplot2::annotate(geom = "text", x = 5, y = 0.66, hjust=0, label = paste0("S = ",unname(round(pm_label$estimate,2)),"\np =  ", unname(round(pm_label$p.value,2))), size = annotationsize) + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL) + xlim(c(0,70)) + ylim(c(-0.1,.78)),
             joepiles_gantt + labs(tag = "B") + theme(axis.text.x = element_text(size = smallsize), axis.text.y = element_text(size = smallsize), axis.title = element_text(size = smallsize, face = "bold"), plot.tag = element_text(size = tagsize, face = "bold", color = "white")),
             joepiles_gantt2 + labs(tag = "H", y = "Time from pile start (d)") + theme(axis.text.x = element_text(size = smallsize), axis.text.y = element_text(size = smallsize), axis.title = element_text(size = smallsize, face = "bold"), plot.tag = element_text(size = tagsize, face = "bold", color = "white")),
              plot_legend_lineartime,
             peachlineartime_ancom_lacto + labs(tag = "G", title = NULL, x = "Time (d)") + theme(axis.title = element_blank(), axis.title.x = element_text(face = "bold"), axis.text.y = element_text(size = smallsize), axis.text.x = element_text(size = smallsize, angle=0)) + xlim(c(0,70)) + stat_smooth(method = "lm", alpha = alphaval, color = "black")+ ggplot2::annotate(geom = "text", x = 5, y = 0.07, hjust=0,  label = paste0("S = ",unname(round(pg_label$estimate,2)),"\np = ", unname(round(pg_label$p.value,2))), size = annotationsize) + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL),
             peachtime_ancom_lacto + labs(tag = "N", title = NULL, x = "Time (d)") + theme(axis.title.y = element_blank(), axis.text.y = element_text(size = smallsize), axis.text.x = element_text(size = smallsize, angle=0), axis.title.x = element_text(face= "bold", size = smallsize)) + xlim(c(0,70)) + ylim(c(-0.02,.18)) + stat_smooth(method = "lm", alpha = alphaval, color = "black")+ ggplot2::annotate(geom = "text", x = 5, y = 0.15, hjust=0,  label = paste0("S = ",unname(round(pn_label$estimate,2)),"\np = ", unname(round(pn_label$p.value,4))), size = annotationsize) + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL),
             lineartime_ancom_leuconostoc + theme(axis.title.x = element_text(face = "bold", size = smallsize), axis.text.y = element_text(size = smallsize), axis.text.x = element_text(size = smallsize, angle=0)) + xlim(c(0,70)) + stat_smooth(method = "lm", alpha = alphaval, color = "black")+ ggplot2::annotate(geom = "text", x = 65, y = 0.18, hjust=1, label = paste0("S = ",unname(round(pe_label$estimate,2)),"\np =  ", unname(round(pe_label$p.value,3))), size = annotationsize) + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL, axis.title.y = element_blank())+ labs(tag = "E", title = NULL, x = "Time (d)") + geom_col(fill = "purple1"),
             time_ancom_leuconostoc + labs(tag = "L", title = NULL, x= "Time (d)") + theme(axis.title.x = element_text(face = "bold", size = smallsize), axis.title.y = element_blank(), axis.text = element_text(size = smallsize))+ geom_col(fill = "purple1") + stat_smooth(method = "lm", alpha = alphaval, color = "black")+ ggplot2::annotate(geom = "text", x = 60, y = 0.1, hjust=1,  label = paste0("S = ",unname(round(pl_label$estimate,2)),"\np = ", unname(round(pl_label$p.value,5))), size = annotationsize) + theme(plot.tag = element_text(size = tagsize, face = "bold"), plot.tag.position = NULL),


             widths = c(1,1,1,1,1,1,1,1,1,1,1,1,1,1,1),
#             heights = c(1,.5,.3,1.2,.8,.75,.8),
             heights = c(.9,.7,.8,.9,.7,.8,.3),
             layout_matrix=rbind(		
               c(rep(2,10), rep(4,5)),
               c(rep(11,10), rep(4,5)),
               c(rep(5,3), rep(6,3), rep(17,3), rep(7,3),rep(15,3)),
               c(rep(1,10), rep(3,5)),
               c(rep(12,10), rep(3,5)),
#                c(2,2,4),
#                 c(11,11,4),
#                 c(11,11,3),
#                 c(1,1,3),
# #               c(11,11,4),
               c(rep(8,3), rep(9,3), rep(18,3),rep(10,3), rep(16,3)),
c(rep(14,15))
                )
)
#dev.off()


```


# Figure 5
# Development panel B
```{r}

dev_r1 <- readr::read_csv(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/dev_r1.csv", show_col_types = F) %>% dplyr::select(-`...1`)

table(dev_r1$cage)
table(dev_r1$treatment)
table(dev_r1$replicate)
table(dev_r1$vial)
table(dev_r1$experiment)
table(dev_r1$trt)
table(dev_r1$location)

## drop axenic vials that were contaminated
dev3 <- dev_r1 %>%
  filter(!vial %in% c("P19-A-3-C","P19-A-2-C","P19-A-1-C","P20-A-3-C","P20-A-2-C","P20-A-1-C","P19-A-4-C","P20-A-4-C","TW-A-4-A","P21-A-1-A","P21-A-3-A","P21-A-4-A","T1-A-2-A","T1-A-4-A","ER-A-2-A","ER-A-3-A")) 
  
dev3$S <- survival::Surv(dev3$variable, dev3$event)
dev3$tr <- with(dev3, interaction(treatment, cage))

model_all <- coxme::coxme(S ~ treatment * trt + (1|experiment/vial),  data = dev3)
summary(model_all)
model_all <- coxme::coxme(S ~ treatment * trt + (1|vial),  data = dev3)
summary(model_all)
## use this one, lowest AIC
model_all <- coxme::coxme(S ~ treatment * trt + (1|experiment),  data = dev3)
summary(model_all)

dev3$tt <- with(dev3, interaction(treatment, trt))
dev3$S <- survival::Surv(dev3$variable, dev3$event)

model_all <- coxme::coxme(S ~ tt + (1|experiment),  data = dev3)
summary(model_all)[[1]]
l1.1 <- glht(model_all, mcp(tt = "Tukey"))
p_val1 <- try(summary(l1.1)$test$pvalues[1],T)
cld(l1.1)

#jpeg(width = 11.4, height = 10, filename = "round124_axvgn_manual.jpg", units = "in", res = 300)
plot(survfit(S ~ tt, dev3),
  xlim=c(8.5,16.8), 
  lty = c(1,1,2,2,rep(1,7),rep(2,7)),
  lwd = 3, xlab = "time (d)", ylab = "% eclosed",  col = c(rep(c("black","purple","black","purple","magenta","cyan","grey"),2)),
  fun=function(x) 1-x
)
legend(x=12, y=.5, c("c  Apples - axenic","a  Compost - axenic","a  Apples - gnotobiotic","b  Compost - gnotobiotic"), col=c("black","black","purple","purple"), lty= c(1,2,1,2),lwd=3, bty = "n")
# dev.off()
```

## Panel C
```{r Dev data bacteria-free}

dev <- readr::read_csv(file = "https://raw.githubusercontent.com/johnchaston/Gale2024/main/figure5c.csv", show_col_types = F, skip = 1) %>%
  filter(acetoCFU!="#N/A" &toupper(vial)%in%c("MB6","MT6","MB6*","MB7","MB7*","MB8","MB8*","MB9*","MT6*","MT7","MT7*","MT8","MT8*","MT9*","MX6","MX6*","MX7","MX7*","MX8","MX8*","MX9*")==F) %>% mutate(vial = as.factor(vial), treatment = as.factor(treatment), cage = as.factor(cage), bacteria=as.factor(bacteria), exp= as.factor(exp), contam=as.factor(contam))

	## filter based on density and contamination
dev_filtered_bf <- dev %>% filter(density>25 & density <80 & contam=="noC" & bacteria=="bf" & treatment!="No_add") %>%  droplevels()

	## number of vials per experiment per treatment
dv2 <- dev_filtered_bf %>% group_by(treatment,vial, exp, cage) %>% dplyr::summarize(vial2=unique(vial)) %>% mutate(exp_geno = paste0(exp,cage)) %>% group_by(treatment,exp_geno, cage,exp) %>% dplyr::summarize(dplyr::n())
	dv2a <- filter(dv2,exp=="A") %>% group_by(treatment) %>% dplyr::summarize(vialsum=sum(dplyr::n()))
	dv2b <- filter(dv2,exp=="B") %>% group_by(treatment) %>% dplyr::summarize(vialsum=sum(dplyr::n()))
	dv2c <- filter(dv2,exp=="C") %>% group_by(treatment) %>% dplyr::summarize(vialsum=sum(dplyr::n())) %>% full_join(dv2b,by="treatment")%>% full_join(dv2a, by="treatment") %>% dplyr::select(treatment,vialsum,vialsum.y,vialsum.x)
colnames(dv2c) <- c("treatment","ExpA","ExpB","ExpC")
dv2c
	## number of flies per experiment per treatment
dv2 <- dev_filtered_bf %>% group_by(treatment,vial, exp, cage) %>% mutate(exp_geno = paste0(exp,cage)) %>% group_by(treatment,exp_geno, cage,exp) %>% dplyr::summarize(dplyr::n())
	dv2a <- filter(dv2,exp=="A") %>% group_by(treatment) %>% dplyr::summarize(vialsum=sum(`dplyr::n()`))
	dv2b <- filter(dv2,exp=="B") %>% group_by(treatment) %>% dplyr::summarize(vialsum=sum(`dplyr::n()`))
	dv2d <- filter(dv2,exp=="C") %>% group_by(treatment) %>% dplyr::summarize(vialsum=sum(`dplyr::n()`)) %>% full_join(dv2b,by="treatment")%>% full_join(dv2a, by="treatment") %>% dplyr::select(treatment,vialsum,vialsum.y,vialsum.x)
colnames(dv2d) <- c("treatment","ExpA","ExpB","ExpC")
print(dv2d)
dv2d <- dv2d %>% rowwise %>% mutate(sum2 = sum(ExpA,ExpB,ExpC,na.rm=T)) %>% dplyr::select(sum2) %>% unlist() %>% unname()

dev_filtered_bf$S <- Surv(dev_filtered_bf$time,dev_filtered_bf$event)  # create the response variable, a survival model

dev_filtered_bf$treatment
dev_bf_model <- coxph(S ~ treatment,dev_filtered_bf)
survdiff(S ~ treatment, data = dev_filtered_bf)

dev_bf <- coxme::coxme(S ~ treatment + (1|exp),  data = dev_filtered_bf)
summary(dev_bf)
dev_bf_model_glht <- glht(dev_bf, mcp(treatment = "Tukey"))
dev_bf_model_cld <- cld(dev_bf_model_glht)
dev_bf_model_cld
anova(dev_bf_model)
summary(dev_bf_model)

	## make treatment-level survival plot
sum(table(list(dev_filtered_bf$treatment)))
bfs <- plot(survfit(S~treatment, dev_filtered_bf), lwd=3, col=c("black","black","black"), lty = c(1,2), xlim=c(10,16), xlab="time (d)", ylab="% eclosed", main="Bacteria free", fun=function(x) 1-x)
legend(x=10, y=1, c("Rudman At-inoculated","Rudman Lb-inoculated"), col=c("black","black","black"), lty= c(1,2),lwd=3, bty = "n")
```



# Supporting analyses
## lyman fruits
### taxon plot
```{r}

file_path <- "lymanfruits"
mapper_file <- "~/Dropbox/sequencing/2022-08-ECrun1and2/mapper_1and2_WendyJMC.tsv"

  ## unweighted
flies_unweighted <- read.table(paste('core-metrics-results-',file_path,'/unweighted_unifrac_distance_matrix/distance-matrix.tsv',sep=""), header=T, sep="\t") %>% mutate(X=as.character(X))
flies_unweighted_dm <- as.dist(flies_unweighted[,2:dim(flies_unweighted)[2]])
flies_unifrac_table <- flies_unweighted %>% dplyr::select(X) %>% left_join(read.table(mapper_file,comment.char = "", header=T, fill=T, sep="\t"), by=c("X"="X.SampleID")) %>% droplevels()

str(flies_unifrac_table)
table(list(flies_unifrac_table$noSubSite)) #
table(list(flies_unifrac_table$fruittype)) #
table(list(flies_unifrac_table$wolbachia)) #

make_permanova_tables(file_path = file_path, mapper_file = mapper_file, flies_unifrac_table = flies_unifrac_table, pform = "~ fruittype/noSubSite * wolbachia")

map_path = "-eastcoast" # taxonomy folder extension
plotvarnames = c("site", "fruittype","noSubSite", "wolbachia","fnw") 
mapper_file = mapper_file 
taxonomic_level="genus"
rpa_in_chart = .02
read_depth = 120
legend_position = "bottom"
sort_x_axis = "noSubSite"
x_cluster = "X.SampleID" #

ptp <- prep_taxon_plot(file_path = file_path, map_path = map_path, plotvarnames = plotvarnames, mapper_file = mapper_file, taxonomic_level=taxonomic_level, rpa_in_chart = rpa_in_chart, read_depth = read_depth, legend_position=legend_position,  sort_x_axis = sort_x_axis,  x_cluster = x_cluster, predefined_taxa = predefined_taxa, predefined_shortname = predefined_shortname)

table(ptp[[1]]$noSubSite)
ptp[[1]]$noSubSite <- factor(ptp[[1]]$noSubSite, levels =rev(c((c("crispin","daybreakFuji","goldendelicious","macintoshL","macoun")),(c("gloriapeach","messinaPeachD","messinaPeachL")),(c("asianpear","bartlettpear")))))
ptp[[1]]$noSubSite <- factor(ptp[[1]]$noSubSite, labels = rev(c((c("Crispin","Fuji","Golden delicious","Macintosh","Macoun")),(c("Gloria","Messina site 2","Messina site 1")),(c("Asian","Bartlett")))))
ptp[[1]]$fruittype <- factor(ptp[[1]]$fruittype, labels = c("Apple","Peach","Pear"))
ptp[[1]]$wolbachia <- factor(ptp[[1]]$wolbachia, labels = c("*Wolbachia* negative","*Wolbachia* positive"))

x_cluster <- "fnw"
plot_color <- c("black","black","lightgreen","darkgreen","blue4","purple4","purple1","lavender","blue","pink","yellow","orange","magenta","red")
plot_order <- ptp[[3]]$shortname[c(1,13,12,3,2,5,6,4,7,9,11,10,8)]
order_numbers <- c(1,13,12,3,2,5,6,4,7,9,11,10,8)
facet_formula <- ". ~ noSubSite * fruittype * wolbachia"
sort_x_axis <- "noSubSite"
legend_position = "bottom"

base_plot <- make_taxon_plot_condensed(ptp = ptp, plot_color = plot_color, facet_formula = facet_formula, sort_x_axis = sort_x_axis, x_cluster = x_cluster, plotvarnames = plotvarnames, legend_position = "none", read_depth = read_depth, plot_order = plot_order,xorder = "noSubSite", relevel_row = "fruittype",relevelrow_vec = rev(c("Apple","Peach","Pear")), column_bar_width = 0.9, relevel_col = "noSubSite",relevel_vec = rev(c((c("Crispin","Fuji","Golden delicious","Macintosh","Macoun")),rev(c("Gloria","Messina site 2","Messina site 1")),rev(c("Asian","Bartlett")))), predefined_colors = predefined_colors, predefined_shortname = predefined_shortname, predefined_taxa = predefined_taxa, default_colors = T, default_order = F, order_numbers = order_numbers)
base_plot
wild_fruits_wolb <- base_plot + 
  facet_grid(rows = vars(fruittype), cols = vars(wolbachia), drop = FALSE, as.table = FALSE, scales = "free", space = "free", labeller = labeller(wolbachia = c("hi","bue"))) + #bquote(italic(Wolbachia)~"+"), bquote(italic(Wolbachia)~"-")))) + 
  theme(strip.text = element_text(size = 24, face = "bold"), axis.text.x = element_text(angle = 90), axis.text.y = element_text(size=20), strip.text.y = element_text(angle = 270), strip.background = element_blank()) + 
  coord_flip() + 
  labs(x = "Sample") + 
  theme(strip.text = ggtext::element_markdown())
wild_fruits_wolb



```

### ANCOM
```{r}

###
# ANCOM
###
rm(melted_table, melted_table_no0, mt2)
#mapper_file <- "MO.FUN.Metadata_CJR_7.5.2022_JMC.txt"
var1 = "wolbachia"
var2 = "noSubSite"
var3 = "fruittype"
var4 = "fnw"
the_id = "X.SampleID"
newcol <- "newcol"
the_id <- "X.SampleID"

main.var="wolbachia"
adj.formula= "fruittype + noSubSite"# + noSubSite" #"orchard" #"fruit + pile"
random.formula = NULL #"~1|fruittype"
multcorr="none"
sig=0.05
prev.cut=0.9
Group = "fnw"

# taxonomyrun <- c("phylum","class","order","family","genus","species","X.OTU.ID")
# for (i in taxonomyrun[1:7]) {
#   try(make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level=i,var1=var1,var2=var2,var3=var3,var4 = var4,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group))
# }


lymanancom <- make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="family",var1=var1,var2=var2,var3=var3,var4 = var4,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_tables = T)

lymanancom[[1]][[1]]$fruittype <- c(rep("Apple",9), rep("Peach",5),rep("Pear",4))
lymanancom[[1]][[1]]$noSubSite <- c("Crispin","Crispin","Fuji","Golden delicious","Golden delicious","Macintosh","Macintosh","Macoun","Macoun","Gloria","Messina Site 1","Messina Site 1","Messina Site 2","Messina Site 2","Asian","Asian","Bartlett","Bartlett")
lymanancom[[1]][[1]]$noSubSite <- rev(factor(lymanancom[[1]][[1]]$noSubSite, levels = rev(c("Crispin","Fuji","Golden delicious","Macintosh","Macoun","Gloria","Messina Site 1","Messina Site 2","Asian","Bartlett"))))
lymanancom[[1]][[1]]$wolbachia <- gsub(pattern = ".*N$", replacement = "*Wolbachia* negative", lymanancom[[1]][[1]]$Group)
lymanancom[[1]][[1]]$wolbachia <- gsub(pattern = ".*Y$", replacement = "*Wolbachia* positive", lymanancom[[1]][[1]]$wolbachia)

lyman_ancomplot_family <- ggplot(lymanancom[[1]][[1]], aes(x=noSubSite, y = mean)) + 
  geom_col(fill = "red") + 
  theme_cowplot() +
  facet_grid(. ~ wolbachia) +
  facet_grid(cols = vars(wolbachia), rows = vars(fruittype), scales = "free",space = "free") + 
  coord_flip() + 
  geom_errorbar(aes(ymin=mean+sem, ymax=mean-sem), width=.2,position=position_dodge(.9), color = "black") +
  ylim(c(0,1)) +
  theme(strip.text = element_text(size = 24, face = "bold"), axis.text.x = element_text(angle = 90), axis.text.y = element_text(size=20), strip.text.y = element_text(angle = 270), strip.background = element_blank()) + 
  labs(x = "Sample") + 
  theme(strip.text = ggtext::element_markdown())


lyman_ancomplot_family

lymanancom <- make_ancom2_plots(file_path, map_path=map_path, mapper_file=mapper_file, taxonomic_level="X.OTU.ID",var1=var1,var2=var2,var3=var3,var4 = var4,newcol="newcol", the_id = "X.SampleID", main.var=main.var, adj.formula=adj.formula, random.formula = random.formula, repeat.var = repeat.var, multcorr=multcorr, sig=sig, prev.cut=prev.cut, Group = Group, return_tables = T)

lymanancom[[1]][[1]]$fruittype <- c(rep("Apple",9), rep("Peach",5),rep("Pear",4))
lymanancom[[1]][[1]]$noSubSite <- c("Crispin","Crispin","Fuji","Golden delicious","Golden delicious","Macintosh","Macintosh","Macoun","Macoun","Gloria","Messina Site 1","Messina Site 1","Messina Site 2","Messina Site 2","Asian","Asian","Bartlett","Bartlett")
lymanancom[[1]][[1]]$noSubSite <- rev(factor(lymanancom[[1]][[1]]$noSubSite, levels = rev(c("Crispin","Fuji","Golden delicious","Macintosh","Macoun","Gloria","Messina Site 1","Messina Site 2","Asian","Bartlett"))))
lymanancom[[1]][[1]]$wolbachia <- gsub(pattern = ".*N$", replacement = "*Wolbachia* negative", lymanancom[[1]][[1]]$Group)
lymanancom[[1]][[1]]$wolbachia <- gsub(pattern = ".*Y$", replacement = "*Wolbachia* positive", lymanancom[[1]][[1]]$wolbachia)

lyman_ancomplot_ASV <- ggplot(lymanancom[[1]][[1]], aes(x=noSubSite, y = mean)) + 
  geom_col(fill = "yellow") + 
  theme_cowplot() +
  facet_grid(. ~ wolbachia) +
  facet_grid(cols = vars(wolbachia), rows = vars(fruittype), scales = "free",space = "free") + 
  coord_flip() + 
  geom_errorbar(aes(ymin=mean+sem, ymax=mean-sem), width=.2,position=position_dodge(.9), color = "black") +
  ylim(c(0,1)) +
  theme(strip.text = element_text(size = 24, face = "bold"), axis.text.x = element_text(angle = 90), axis.text.y = element_text(size=20), strip.text.y = element_text(angle = 270), strip.background = element_blank()) + 
  labs(x = "Sample") + 
  theme(strip.text = ggtext::element_markdown())


lyman_ancomplot_ASV


```

## overall figure
```{r}

getwd()
i=1.7
tagsize=40
smallsize = 22

titlesize = 24
  tagsize = 40

plot(wf_legend)

wf_legend_wolb <- g_legend(wild_fruits + theme(legend.position = "bottom", legend.title = element_blank()) + guides(fill=guide_legend(ncol=1,byrow=F))+ theme(legend.text = element_text(size = 18)) + scale_fill_manual(name = "J", values = c("black","black","lightgreen","darkgreen","blue4","purple4","purple1","lavender","blue","pink","yellow","orange","magenta","red"), labels = c("other",bquote(italic(Dysgonomonas)),bquote(italic(Pasteurellales)),bquote(italic(Enterobacteriaceae)),bquote(italic(Lactobacillaceae)),bquote(italic(Enterococcus)),bquote(italic(Leuconostoc)),bquote(italic(Weissella)),bquote(italic(Lactobacillus)),bquote(italic(Acetobacteraceae)),bquote(italic(Commensalibacter)),bquote(italic(Gluconobacter)),bquote(italic(Komagataeibacter)),bquote(italic(Acetobacter)))))
plot(wf_legend_wolb)

i=1.6
#jpeg(h=4*i*2, w=11*i*1.5, "wolbachia_lyman_july2024.jpg", units = "in", res=300)
grid.arrange(wild_fruits_wolb + labs(tag = "A") + theme(plot.tag = element_text(size = tagsize), axis.title = element_text(size = titlesize, face = "bold")),
            lyman_ancomplot_family + labs(tag = "B", x = "Fractional abundance") + theme(plot.tag = element_text(size = tagsize), axis.title = element_text(size = titlesize, face = "bold")),
             lyman_ancomplot_ASV + labs(tag = "C", x = "Fractional abundance") + theme(plot.tag = element_text(size = tagsize), axis.title = element_text(size = titlesize, face = "bold")),
            wf_legend_wolb,
            ggplot() + theme_void(),
            ggplot() + theme_void(),
             widths = c(.5,1,.1,1),
             heights = c(1,.1,1),
             layout_matrix=rbind(		
                c(1,1,1,1),
                c(5,5,5,5),
                c(4,2,6,3)
#                c(2,2,3,3,4,4)
             )
)
#dev.off()

```


